// @nativewrappers/fivem/weapon/WeaponLiveryColor.js
var WeaponLiveryColor = /* @__PURE__ */ ((WeaponLiveryColor2) => {
  WeaponLiveryColor2[WeaponLiveryColor2["Gray"] = 0] = "Gray";
  WeaponLiveryColor2[WeaponLiveryColor2["DarkGray"] = 1] = "DarkGray";
  WeaponLiveryColor2[WeaponLiveryColor2["Black"] = 2] = "Black";
  WeaponLiveryColor2[WeaponLiveryColor2["White"] = 3] = "White";
  WeaponLiveryColor2[WeaponLiveryColor2["Blue"] = 4] = "Blue";
  WeaponLiveryColor2[WeaponLiveryColor2["Cyan"] = 5] = "Cyan";
  WeaponLiveryColor2[WeaponLiveryColor2["Aqua"] = 6] = "Aqua";
  WeaponLiveryColor2[WeaponLiveryColor2["CoolBlue"] = 7] = "CoolBlue";
  WeaponLiveryColor2[WeaponLiveryColor2["DarkBlue"] = 8] = "DarkBlue";
  WeaponLiveryColor2[WeaponLiveryColor2["RoyalBlue"] = 9] = "RoyalBlue";
  WeaponLiveryColor2[WeaponLiveryColor2["Plum"] = 10] = "Plum";
  WeaponLiveryColor2[WeaponLiveryColor2["DarkPurple"] = 11] = "DarkPurple";
  WeaponLiveryColor2[WeaponLiveryColor2["Purple"] = 12] = "Purple";
  WeaponLiveryColor2[WeaponLiveryColor2["Red"] = 13] = "Red";
  WeaponLiveryColor2[WeaponLiveryColor2["WineRed"] = 14] = "WineRed";
  WeaponLiveryColor2[WeaponLiveryColor2["Magenta"] = 15] = "Magenta";
  WeaponLiveryColor2[WeaponLiveryColor2["Pink"] = 16] = "Pink";
  WeaponLiveryColor2[WeaponLiveryColor2["Salmon"] = 17] = "Salmon";
  WeaponLiveryColor2[WeaponLiveryColor2["HotPink"] = 18] = "HotPink";
  WeaponLiveryColor2[WeaponLiveryColor2["RustOrange"] = 19] = "RustOrange";
  WeaponLiveryColor2[WeaponLiveryColor2["Brown"] = 20] = "Brown";
  WeaponLiveryColor2[WeaponLiveryColor2["Earth"] = 21] = "Earth";
  WeaponLiveryColor2[WeaponLiveryColor2["Orange"] = 22] = "Orange";
  WeaponLiveryColor2[WeaponLiveryColor2["LightOrange"] = 23] = "LightOrange";
  WeaponLiveryColor2[WeaponLiveryColor2["DarkYellow"] = 24] = "DarkYellow";
  WeaponLiveryColor2[WeaponLiveryColor2["Yellow"] = 25] = "Yellow";
  WeaponLiveryColor2[WeaponLiveryColor2["LightBrown"] = 26] = "LightBrown";
  WeaponLiveryColor2[WeaponLiveryColor2["LimeGreen"] = 27] = "LimeGreen";
  WeaponLiveryColor2[WeaponLiveryColor2["Olive"] = 28] = "Olive";
  WeaponLiveryColor2[WeaponLiveryColor2["Moss"] = 29] = "Moss";
  WeaponLiveryColor2[WeaponLiveryColor2["Turquoise"] = 30] = "Turquoise";
  WeaponLiveryColor2[WeaponLiveryColor2["DarkGreen"] = 31] = "DarkGreen";
  return WeaponLiveryColor2;
})(WeaponLiveryColor || {});
export {
  WeaponLiveryColor
};
