// @nativewrappers/fivem/weapon/WeaponTint.js
var WeaponTint = /* @__PURE__ */ ((WeaponTint2) => {
  WeaponTint2[WeaponTint2["Mk2ClassicBlack"] = 0] = "Mk2ClassicBlack";
  WeaponTint2[WeaponTint2["Mk2ClassicGray"] = 1] = "Mk2ClassicGray";
  WeaponTint2[WeaponTint2["Mk2ClassicTwoTone"] = 2] = "Mk2ClassicTwoTone";
  WeaponTint2[WeaponTint2["Mk2ClassicWhite"] = 3] = "Mk2ClassicWhite";
  WeaponTint2[WeaponTint2["Mk2ClassicBeige"] = 4] = "Mk2ClassicBeige";
  WeaponTint2[WeaponTint2["Mk2ClassicGreen"] = 5] = "Mk2ClassicGreen";
  WeaponTint2[WeaponTint2["Mk2ClassicBlue"] = 6] = "Mk2ClassicBlue";
  WeaponTint2[WeaponTint2["Mk2ClassicEarth"] = 7] = "Mk2ClassicEarth";
  WeaponTint2[WeaponTint2["Mk2ClassicBrownAndBlack"] = 8] = "Mk2ClassicBrownAndBlack";
  WeaponTint2[WeaponTint2["Mk2RedContrast"] = 9] = "Mk2RedContrast";
  WeaponTint2[WeaponTint2["Mk2BlueContrast"] = 10] = "Mk2BlueContrast";
  WeaponTint2[WeaponTint2["Mk2YellowContrast"] = 11] = "Mk2YellowContrast";
  WeaponTint2[WeaponTint2["Mk2OrangeContrast"] = 12] = "Mk2OrangeContrast";
  WeaponTint2[WeaponTint2["Mk2BoldPink"] = 13] = "Mk2BoldPink";
  WeaponTint2[WeaponTint2["Mk2BoldPurpleAndYellow"] = 14] = "Mk2BoldPurpleAndYellow";
  WeaponTint2[WeaponTint2["Mk2BoldOrange"] = 15] = "Mk2BoldOrange";
  WeaponTint2[WeaponTint2["Mk2BoldGreenAndPurple"] = 16] = "Mk2BoldGreenAndPurple";
  WeaponTint2[WeaponTint2["Mk2BoldRedFeatures"] = 17] = "Mk2BoldRedFeatures";
  WeaponTint2[WeaponTint2["Mk2BoldGreenFeatures"] = 18] = "Mk2BoldGreenFeatures";
  WeaponTint2[WeaponTint2["Mk2BoldCyanFeatures"] = 19] = "Mk2BoldCyanFeatures";
  WeaponTint2[WeaponTint2["Mk2BoldYellowFeatures"] = 20] = "Mk2BoldYellowFeatures";
  WeaponTint2[WeaponTint2["Mk2BoldRedAndWhite"] = 21] = "Mk2BoldRedAndWhite";
  WeaponTint2[WeaponTint2["Mk2BoldBlueAndWhite"] = 22] = "Mk2BoldBlueAndWhite";
  WeaponTint2[WeaponTint2["Mk2MetallicGold"] = 23] = "Mk2MetallicGold";
  WeaponTint2[WeaponTint2["Mk2MetallicPlatinum"] = 24] = "Mk2MetallicPlatinum";
  WeaponTint2[WeaponTint2["Mk2MetallicGrayAndLilac"] = 25] = "Mk2MetallicGrayAndLilac";
  WeaponTint2[WeaponTint2["Mk2MetallicPurpleAndLime"] = 26] = "Mk2MetallicPurpleAndLime";
  WeaponTint2[WeaponTint2["Mk2MetallicRed"] = 27] = "Mk2MetallicRed";
  WeaponTint2[WeaponTint2["Mk2MetallicGreen"] = 28] = "Mk2MetallicGreen";
  WeaponTint2[WeaponTint2["Mk2MetallicBlue"] = 29] = "Mk2MetallicBlue";
  WeaponTint2[WeaponTint2["Mk2MetallicWhiteAndAqua"] = 30] = "Mk2MetallicWhiteAndAqua";
  WeaponTint2[WeaponTint2["Mk2MetallicRedAndYellow"] = 31] = "Mk2MetallicRedAndYellow";
  WeaponTint2[WeaponTint2["Normal"] = 0] = "Normal";
  WeaponTint2[WeaponTint2["Green"] = 1] = "Green";
  WeaponTint2[WeaponTint2["Gold"] = 2] = "Gold";
  WeaponTint2[WeaponTint2["Pink"] = 3] = "Pink";
  WeaponTint2[WeaponTint2["Army"] = 4] = "Army";
  WeaponTint2[WeaponTint2["LSPD"] = 5] = "LSPD";
  WeaponTint2[WeaponTint2["Orange"] = 6] = "Orange";
  WeaponTint2[WeaponTint2["Platinum"] = 7] = "Platinum";
  return WeaponTint2;
})(WeaponTint || {});
export {
  WeaponTint
};
