// @nativewrappers/fivem/weapon/WeaponGroup.js
var WeaponGroup = /* @__PURE__ */ ((WeaponGroup2) => {
  WeaponGroup2[WeaponGroup2["Unarmed"] = 2685387236] = "Unarmed";
  WeaponGroup2[WeaponGroup2["Melee"] = 3566412244] = "Melee";
  WeaponGroup2[WeaponGroup2["Pistol"] = 416676503] = "Pistol";
  WeaponGroup2[WeaponGroup2["SMG"] = 3337201093] = "SMG";
  WeaponGroup2[WeaponGroup2["AssaultRifle"] = 970310034] = "AssaultRifle";
  WeaponGroup2[WeaponGroup2["DigiScanner"] = 3539449195] = "DigiScanner";
  WeaponGroup2[WeaponGroup2["FireExtinguisher"] = 4257178988] = "FireExtinguisher";
  WeaponGroup2[WeaponGroup2["MG"] = 1159398588] = "MG";
  WeaponGroup2[WeaponGroup2["NightVision"] = 3493187224] = "NightVision";
  WeaponGroup2[WeaponGroup2["Parachute"] = 431593103] = "Parachute";
  WeaponGroup2[WeaponGroup2["Shotgun"] = 860033945] = "Shotgun";
  WeaponGroup2[WeaponGroup2["Sniper"] = 3082541095] = "Sniper";
  WeaponGroup2[WeaponGroup2["Stungun"] = 690389602] = "Stungun";
  WeaponGroup2[WeaponGroup2["Heavy"] = 2725924767] = "Heavy";
  WeaponGroup2[WeaponGroup2["Thrown"] = 1548507267] = "Thrown";
  WeaponGroup2[WeaponGroup2["PetrolCan"] = 1595662460] = "PetrolCan";
  return WeaponGroup2;
})(WeaponGroup || {});
export {
  WeaponGroup
};
