// @nativewrappers/client/lib/hashes/WeaponHash.js
export var WeaponHash;
(function (WeaponHash) {
    // melee
    WeaponHash[WeaponHash["Dagger"] = -1834847097] = "Dagger";
    WeaponHash[WeaponHash["Bat"] = -1786099057] = "Bat";
    WeaponHash[WeaponHash["Bottle"] = -102323637] = "Bottle";
    WeaponHash[WeaponHash["Crowbar"] = -2067956739] = "Crowbar";
    WeaponHash[WeaponHash["Unarmed"] = -1569615261] = "Unarmed";
    WeaponHash[WeaponHash["Flashlight"] = -1951375401] = "Flashlight";
    WeaponHash[WeaponHash["GolfClub"] = 1141786504] = "GolfClub";
    WeaponHash[WeaponHash["Hammer"] = 1317494643] = "Hammer";
    WeaponHash[WeaponHash["Hatchet"] = -102973651] = "Hatchet";
    WeaponHash[WeaponHash["KnuckleDuster"] = -656458692] = "KnuckleDuster";
    WeaponHash[WeaponHash["Knife"] = -1716189206] = "Knife";
    WeaponHash[WeaponHash["Machete"] = -581044007] = "Machete";
    WeaponHash[WeaponHash["SwitchBlade"] = -538741184] = "SwitchBlade";
    WeaponHash[WeaponHash["Nightstick"] = 1737195953] = "Nightstick";
    WeaponHash[WeaponHash["Wrench"] = 419712736] = "Wrench";
    WeaponHash[WeaponHash["BattleAxe"] = -853065399] = "BattleAxe";
    WeaponHash[WeaponHash["PoolCue"] = -1810795771] = "PoolCue";
    WeaponHash[WeaponHash["StoneHatchet"] = 940833800] = "StoneHatchet";
    // handguns
    WeaponHash[WeaponHash["Pistol"] = 453432689] = "Pistol";
    WeaponHash[WeaponHash["PistolMk2"] = 3219281620] = "PistolMk2";
    WeaponHash[WeaponHash["CombatPistol"] = 1593441988] = "CombatPistol";
    WeaponHash[WeaponHash["APPistol"] = 584646201] = "APPistol";
    WeaponHash[WeaponHash["StunGun"] = 911657153] = "StunGun";
    WeaponHash[WeaponHash["StunGunMp"] = 1171102963] = "StunGunMp";
    WeaponHash[WeaponHash["Pistol50"] = -1716589765] = "Pistol50";
    WeaponHash[WeaponHash["SNSPistol"] = -1076751822] = "SNSPistol";
    WeaponHash[WeaponHash["SNSPistolMk2"] = -2009644972] = "SNSPistolMk2";
    WeaponHash[WeaponHash["HeavyPistol"] = -771403250] = "HeavyPistol";
    WeaponHash[WeaponHash["VintagePistol"] = 137902532] = "VintagePistol";
    WeaponHash[WeaponHash["FlareGun"] = 1198879012] = "FlareGun";
    WeaponHash[WeaponHash["MarksmanPistol"] = -598887786] = "MarksmanPistol";
    WeaponHash[WeaponHash["Revolver"] = -1045183535] = "Revolver";
    WeaponHash[WeaponHash["RevolverMk2"] = -879347409] = "RevolverMk2";
    WeaponHash[WeaponHash["DoubleAction"] = -1746263880] = "DoubleAction";
    WeaponHash[WeaponHash["RayPistol"] = -1355376991] = "RayPistol";
    WeaponHash[WeaponHash["CeramicPistol"] = 727643628] = "CeramicPistol";
    WeaponHash[WeaponHash["NavyRevolver"] = -1853920116] = "NavyRevolver";
    WeaponHash[WeaponHash["GadgetPistol"] = 1470379660] = "GadgetPistol";
    // sub-machine guns
    WeaponHash[WeaponHash["MicroSMG"] = 324215364] = "MicroSMG";
    WeaponHash[WeaponHash["SMG"] = 736523883] = "SMG";
    WeaponHash[WeaponHash["SMGMk2"] = 2024373456] = "SMGMk2";
    WeaponHash[WeaponHash["AssaultSMG"] = -270015777] = "AssaultSMG";
    WeaponHash[WeaponHash["CombatPDW"] = 171789620] = "CombatPDW";
    WeaponHash[WeaponHash["MachinePistol"] = -619010992] = "MachinePistol";
    WeaponHash[WeaponHash["MiniSMG"] = -1121678507] = "MiniSMG";
    WeaponHash[WeaponHash["RayCarbine"] = 1198256469] = "RayCarbine";
    // Light Machine Guns
    WeaponHash[WeaponHash["MG"] = -1660422300] = "MG";
    WeaponHash[WeaponHash["CombatMG"] = 2144741730] = "CombatMG";
    WeaponHash[WeaponHash["CombatMGMk2"] = 3686625920] = "CombatMGMk2";
    WeaponHash[WeaponHash["Gusenberg"] = 1627465347] = "Gusenberg";
    // shotguns
    WeaponHash[WeaponHash["PumpShotgun"] = 487013001] = "PumpShotgun";
    WeaponHash[WeaponHash["PumpShotgunMk2"] = 1432025498] = "PumpShotgunMk2";
    WeaponHash[WeaponHash["SawnOffShotgun"] = 2017895192] = "SawnOffShotgun";
    WeaponHash[WeaponHash["AssaultShotgun"] = -494615257] = "AssaultShotgun";
    WeaponHash[WeaponHash["BullpupShotgun"] = -1654528753] = "BullpupShotgun";
    WeaponHash[WeaponHash["Musket"] = -1466123874] = "Musket";
    WeaponHash[WeaponHash["HeavyShotgun"] = 984333226] = "HeavyShotgun";
    WeaponHash[WeaponHash["DoubleBarrelShotgun"] = -275439685] = "DoubleBarrelShotgun";
    WeaponHash[WeaponHash["SweeperShotgun"] = 317205821] = "SweeperShotgun";
    WeaponHash[WeaponHash["CombatShotgun"] = 94989220] = "CombatShotgun";
    // Assault Rifles
    WeaponHash[WeaponHash["AssaultRifle"] = -1074790547] = "AssaultRifle";
    WeaponHash[WeaponHash["AssaultRifleMk2"] = 961495388] = "AssaultRifleMk2";
    WeaponHash[WeaponHash["CarbineRifle"] = -2084633992] = "CarbineRifle";
    WeaponHash[WeaponHash["CarbineRifleMk2"] = 4208062921] = "CarbineRifleMk2";
    WeaponHash[WeaponHash["AdvancedRifle"] = -1357824103] = "AdvancedRifle";
    WeaponHash[WeaponHash["SpecialCarbine"] = -1063057011] = "SpecialCarbine";
    WeaponHash[WeaponHash["SpecialCarbineMk2"] = -1768145561] = "SpecialCarbineMk2";
    WeaponHash[WeaponHash["BullpupRifle"] = 2132975508] = "BullpupRifle";
    WeaponHash[WeaponHash["BullpupRifleMk2"] = -2066285827] = "BullpupRifleMk2";
    WeaponHash[WeaponHash["CompactRifle"] = 1649403952] = "CompactRifle";
    WeaponHash[WeaponHash["MilitaryRifle"] = -1658906650] = "MilitaryRifle";
    WeaponHash[WeaponHash["HeavyRifle"] = -947031628] = "HeavyRifle";
    // Sniper Rifles
    WeaponHash[WeaponHash["SniperRifle"] = 100416529] = "SniperRifle";
    WeaponHash[WeaponHash["HeavySniper"] = 205991906] = "HeavySniper";
    WeaponHash[WeaponHash["HeavySniperMk2"] = 177293209] = "HeavySniperMk2";
    WeaponHash[WeaponHash["MarksmanRifle"] = -952879014] = "MarksmanRifle";
    WeaponHash[WeaponHash["MarksmanRifleMk2"] = 1785463520] = "MarksmanRifleMk2";
    // Heavy Weapons
    WeaponHash[WeaponHash["RPG"] = -1312131151] = "RPG";
    WeaponHash[WeaponHash["GrenadeLauncher"] = -1568386805] = "GrenadeLauncher";
    WeaponHash[WeaponHash["GrenadeLauncherSmoke"] = 1305664598] = "GrenadeLauncherSmoke";
    WeaponHash[WeaponHash["Minigun"] = 1119849093] = "Minigun";
    WeaponHash[WeaponHash["Firework"] = 2138347493] = "Firework";
    WeaponHash[WeaponHash["Railgun"] = 1834241177] = "Railgun";
    WeaponHash[WeaponHash["HomingLauncher"] = 1672152130] = "HomingLauncher";
    WeaponHash[WeaponHash["CompactGrenadeLauncher"] = 125959754] = "CompactGrenadeLauncher";
    WeaponHash[WeaponHash["RayMinigun"] = -1238556825] = "RayMinigun";
    WeaponHash[WeaponHash["EmpLauncher"] = -618237638] = "EmpLauncher";
    // Throwables
    WeaponHash[WeaponHash["Grenade"] = -1813897027] = "Grenade";
    WeaponHash[WeaponHash["BZGas"] = -1600701090] = "BZGas";
    WeaponHash[WeaponHash["Molotov"] = 615608432] = "Molotov";
    WeaponHash[WeaponHash["StickyBomb"] = 741814745] = "StickyBomb";
    WeaponHash[WeaponHash["ProximityMine"] = -1420407917] = "ProximityMine";
    WeaponHash[WeaponHash["Snowball"] = 126349499] = "Snowball";
    WeaponHash[WeaponHash["PipeBomb"] = -1169823560] = "PipeBomb";
    WeaponHash[WeaponHash["Ball"] = 600439132] = "Ball";
    WeaponHash[WeaponHash["SmokeGrenade"] = -37975472] = "SmokeGrenade";
    WeaponHash[WeaponHash["Flare"] = 1233104067] = "Flare";
    // Miscellaneous
    WeaponHash[WeaponHash["PetrolCan"] = 883325847] = "PetrolCan";
    WeaponHash[WeaponHash["Parachute"] = -72657034] = "Parachute";
    WeaponHash[WeaponHash["FireExtinguisher"] = 101631238] = "FireExtinguisher";
    WeaponHash[WeaponHash["HazardCan"] = -1168940174] = "HazardCan";
    WeaponHash[WeaponHash["FertilizerCan"] = 406929569] = "FertilizerCan";
})(WeaponHash || (WeaponHash = {}));
export var VehicleWeaponHash;
(function (VehicleWeaponHash) {
    VehicleWeaponHash[VehicleWeaponHash["Invalid"] = -1] = "Invalid";
    VehicleWeaponHash[VehicleWeaponHash["Tank"] = 1945616459] = "Tank";
    VehicleWeaponHash[VehicleWeaponHash["SpaceRocket"] = -123497569] = "SpaceRocket";
    VehicleWeaponHash[VehicleWeaponHash["PlaneRocket"] = -821520672] = "PlaneRocket";
    VehicleWeaponHash[VehicleWeaponHash["PlayerLaser"] = -268631733] = "PlayerLaser";
    VehicleWeaponHash[VehicleWeaponHash["PlayerBullet"] = 1259576109] = "PlayerBullet";
    VehicleWeaponHash[VehicleWeaponHash["PlayerBuzzard"] = 1186503822] = "PlayerBuzzard";
    VehicleWeaponHash[VehicleWeaponHash["PlayerHunter"] = -1625648674] = "PlayerHunter";
    VehicleWeaponHash[VehicleWeaponHash["PlayerLazer"] = -494786007] = "PlayerLazer";
    VehicleWeaponHash[VehicleWeaponHash["EnemyLaser"] = 1566990507] = "EnemyLaser";
    VehicleWeaponHash[VehicleWeaponHash["SearchLight"] = -844344963] = "SearchLight";
    VehicleWeaponHash[VehicleWeaponHash["Radar"] = -764006018] = "Radar";
})(VehicleWeaponHash || (VehicleWeaponHash = {}));
// TODO: Convert this to uint instead of hash
export var AmmoType;
(function (AmmoType) {
    AmmoType[AmmoType["Melee"] = 0] = "Melee";
    AmmoType[AmmoType["FireExtinguisher"] = 1359393852] = "FireExtinguisher";
    AmmoType[AmmoType["Flare"] = 1808594799] = "Flare";
    AmmoType[AmmoType["FlareGun"] = 1173416293] = "FlareGun";
    AmmoType[AmmoType["PetrolCan"] = 3395492001] = "PetrolCan";
    AmmoType[AmmoType["Shotgun"] = 2416459067] = "Shotgun";
    AmmoType[AmmoType["Pistol"] = 1950175060] = "Pistol";
    AmmoType[AmmoType["Ball"] = 4287981158] = "Ball";
    AmmoType[AmmoType["Snowball"] = 2182627693] = "Snowball";
    AmmoType[AmmoType["Sniper"] = 1285032059] = "Sniper";
    AmmoType[AmmoType["AssaultRifle"] = 218444191] = "AssaultRifle";
    AmmoType[AmmoType["SMG"] = 1820140472] = "SMG";
    AmmoType[AmmoType["Molotov"] = 1446246869] = "Molotov";
    AmmoType[AmmoType["StunGun"] = 2955849184] = "StunGun";
    AmmoType[AmmoType["MG"] = 1788949567] = "MG";
    AmmoType[AmmoType["GrenadeLauncher"] = 1003267566] = "GrenadeLauncher";
    AmmoType[AmmoType["RPG"] = 1742569970] = "RPG";
    AmmoType[AmmoType["Minigun"] = 2680539266] = "Minigun";
    AmmoType[AmmoType["Firework"] = 2938367503] = "Firework";
    AmmoType[AmmoType["Railgun"] = 2034517757] = "Railgun";
    AmmoType[AmmoType["HomingLauncher"] = 2568293933] = "HomingLauncher";
    AmmoType[AmmoType["Grenade"] = 1003688881] = "Grenade";
    AmmoType[AmmoType["StickyBomb"] = 1411692055] = "StickyBomb";
    AmmoType[AmmoType["ProximityMine"] = 2938243239] = "ProximityMine";
    AmmoType[AmmoType["PipeBomb"] = 357983224] = "PipeBomb";
    AmmoType[AmmoType["SmokeGrenade"] = 3859679398] = "SmokeGrenade";
    AmmoType[AmmoType["BZGas"] = 2608103076] = "BZGas";
})(AmmoType || (AmmoType = {}));
