// @nativewrappers/client/lib/enums/FiringPattern.js
export var FiringPattern;
(function (FiringPattern) {
    FiringPattern[FiringPattern["Default"] = 0] = "Default";
    FiringPattern[FiringPattern["FullAuto"] = 3337513804] = "FullAuto";
    FiringPattern[FiringPattern["BurstFire"] = 3607063905] = "BurstFire";
    FiringPattern[FiringPattern["BurstInCover"] = 40051185] = "BurstInCover";
    FiringPattern[FiringPattern["BurstFireDriveby"] = 3541198322] = "BurstFireDriveby";
    FiringPattern[FiringPattern["FromGround"] = 577037782] = "FromGround";
    FiringPattern[FiringPattern["DelayFireByOneSec"] = 2055493265] = "DelayFireByOneSec";
    FiringPattern[FiringPattern["SingleShot"] = 1566631136] = "SingleShot";
    FiringPattern[FiringPattern["BurstFirePistol"] = 2685983626] = "BurstFirePistol";
    FiringPattern[FiringPattern["BurstFireSMG"] = 3507334638] = "BurstFireSMG";
    FiringPattern[FiringPattern["BurstFireRifle"] = 2624893958] = "BurstFireRifle";
    FiringPattern[FiringPattern["BurstFireMG"] = 3044263348] = "BurstFireMG";
    FiringPattern[FiringPattern["BurstFirePumpShotGun"] = 12239771] = "BurstFirePumpShotGun";
    FiringPattern[FiringPattern["BurstFireHeli"] = 2437838959] = "BurstFireHeli";
    FiringPattern[FiringPattern["BurstFireMicro"] = 1122960381] = "BurstFireMicro";
    FiringPattern[FiringPattern["BurstFireBursts"] = 1122960381] = "BurstFireBursts";
    FiringPattern[FiringPattern["BurstFireTank"] = 3804904049] = "BurstFireTank";
})(FiringPattern || (FiringPattern = {}));
