// @nativewrappers/client/lib/enums/PickupType.js
export var PickupType;
(function (PickupType) {
    PickupType[PickupType["CustomScript"] = 738282662] = "CustomScript";
    PickupType[PickupType["VehicleCustomScript"] = 2780351145] = "VehicleCustomScript";
    PickupType[PickupType["Parachute"] = 1735599485] = "Parachute";
    PickupType[PickupType["PortablePackage"] = 2158727964] = "PortablePackage";
    PickupType[PickupType["PortableCrateUnfixed"] = 1852930709] = "PortableCrateUnfixed";
    PickupType[PickupType["Health"] = 2406513688] = "Health";
    PickupType[PickupType["HealthSnack"] = 483577702] = "HealthSnack";
    PickupType[PickupType["Armour"] = 1274757841] = "Armour";
    PickupType[PickupType["MoneyCase"] = 3463437675] = "MoneyCase";
    PickupType[PickupType["MoneySecurityCase"] = 3732468094] = "MoneySecurityCase";
    PickupType[PickupType["MoneyVariable"] = 4263048111] = "MoneyVariable";
    PickupType[PickupType["MoneyMedBag"] = 341217064] = "MoneyMedBag";
    PickupType[PickupType["MoneyPurse"] = 513448440] = "MoneyPurse";
    PickupType[PickupType["MoneyDepBag"] = 545862290] = "MoneyDepBag";
    PickupType[PickupType["MoneyWallet"] = 1575005502] = "MoneyWallet";
    PickupType[PickupType["MoneyPaperBag"] = 1897726628] = "MoneyPaperBag";
    PickupType[PickupType["WeaponPistol"] = 4189041807] = "WeaponPistol";
    PickupType[PickupType["WeaponCombatPistol"] = 2305275123] = "WeaponCombatPistol";
    PickupType[PickupType["WeaponAPPistol"] = 996550793] = "WeaponAPPistol";
    PickupType[PickupType["WeaponSNSPistol"] = 3317114643] = "WeaponSNSPistol";
    PickupType[PickupType["WeaponHeavyPistol"] = 2633054488] = "WeaponHeavyPistol";
    PickupType[PickupType["WeaponMicroSMG"] = 496339155] = "WeaponMicroSMG";
    PickupType[PickupType["WeaponSMG"] = 978070226] = "WeaponSMG";
    PickupType[PickupType["WeaponMG"] = 2244651441] = "WeaponMG";
    PickupType[PickupType["WeaponCombatMG"] = 2995980820] = "WeaponCombatMG";
    PickupType[PickupType["WeaponAssaultRifle"] = 4080829360] = "WeaponAssaultRifle";
    PickupType[PickupType["WeaponCarbineRifle"] = 3748731225] = "WeaponCarbineRifle";
    PickupType[PickupType["WeaponAdvancedRifle"] = 2998219358] = "WeaponAdvancedRifle";
    PickupType[PickupType["WeaponSpecialCarbine"] = 157823901] = "WeaponSpecialCarbine";
    PickupType[PickupType["WeaponBullpupRifle"] = 2170382056] = "WeaponBullpupRifle";
    PickupType[PickupType["WeaponPumpShotgun"] = 2838846925] = "WeaponPumpShotgun";
    PickupType[PickupType["WeaponSawnoffShotgun"] = 2528383651] = "WeaponSawnoffShotgun";
    PickupType[PickupType["WeaponAssaultShotgun"] = 2459552091] = "WeaponAssaultShotgun";
    PickupType[PickupType["WeaponSniperRifle"] = 4264178988] = "WeaponSniperRifle";
    PickupType[PickupType["WeaponHeavySniper"] = 1765114797] = "WeaponHeavySniper";
    PickupType[PickupType["WeaponGrenadeLauncher"] = 779501861] = "WeaponGrenadeLauncher";
    PickupType[PickupType["WeaponRPG"] = 1295434569] = "WeaponRPG";
    PickupType[PickupType["WeaponMinigun"] = 792114228] = "WeaponMinigun";
    PickupType[PickupType["WeaponGrenade"] = 1577485217] = "WeaponGrenade";
    PickupType[PickupType["WeaponStickyBomb"] = 2081529176] = "WeaponStickyBomb";
    PickupType[PickupType["WeaponSmokeGrenade"] = 483787975] = "WeaponSmokeGrenade";
    PickupType[PickupType["WeaponMolotov"] = 768803961] = "WeaponMolotov";
    PickupType[PickupType["WeaponPetrolCan"] = 3332236287] = "WeaponPetrolCan";
    PickupType[PickupType["WeaponKnife"] = 663586612] = "WeaponKnife";
    PickupType[PickupType["WeaponNightstick"] = 1587637620] = "WeaponNightstick";
    PickupType[PickupType["WeaponBat"] = 2179883038] = "WeaponBat";
    PickupType[PickupType["WeaponCrowbar"] = 2267924616] = "WeaponCrowbar";
    PickupType[PickupType["WeaponGolfclub"] = 2297080999] = "WeaponGolfclub";
    PickupType[PickupType["WeaponBottle"] = 4199656437] = "WeaponBottle";
    PickupType[PickupType["VehicleWeaponPistol"] = 2773149623] = "VehicleWeaponPistol";
    PickupType[PickupType["VehicleWeaponCombatPistol"] = 3500855031] = "VehicleWeaponCombatPistol";
    PickupType[PickupType["VehicleWeaponAPPistol"] = 3431676165] = "VehicleWeaponAPPistol";
    PickupType[PickupType["VehicleWeaponMicroSMG"] = 3094015579] = "VehicleWeaponMicroSMG";
    PickupType[PickupType["VehicleWeaponSawnoffShotgun"] = 772217690] = "VehicleWeaponSawnoffShotgun";
    PickupType[PickupType["VehicleWeaponGrenade"] = 2803366040] = "VehicleWeaponGrenade";
    PickupType[PickupType["VehicleWeaponSmokeGrenade"] = 1705498857] = "VehicleWeaponSmokeGrenade";
    PickupType[PickupType["VehicleWeaponStickyBomb"] = 746606563] = "VehicleWeaponStickyBomb";
    PickupType[PickupType["VehicleWeaponMolotov"] = 2228647636] = "VehicleWeaponMolotov";
    PickupType[PickupType["VehicleHealth"] = 160266735] = "VehicleHealth";
    PickupType[PickupType["AmmoPistol"] = 544828034] = "AmmoPistol";
    PickupType[PickupType["AmmoSMG"] = 292537574] = "AmmoSMG";
    PickupType[PickupType["AmmoMG"] = 3730366643] = "AmmoMG";
    PickupType[PickupType["AmmoRifle"] = 3837603782] = "AmmoRifle";
    PickupType[PickupType["AmmoShotgun"] = 2012476125] = "AmmoShotgun";
    PickupType[PickupType["AmmoSniper"] = 3224170789] = "AmmoSniper";
    PickupType[PickupType["AmmoGrenadeLauncher"] = 2283450536] = "AmmoGrenadeLauncher";
    PickupType[PickupType["AmmoRPG"] = 2223210455] = "AmmoRPG";
    PickupType[PickupType["AmmoMinigun"] = 4065984953] = "AmmoMinigun";
    PickupType[PickupType["AmmoMissileMP"] = 4187887056] = "AmmoMissileMP";
    PickupType[PickupType["AmmoBulletMP"] = 1426343849] = "AmmoBulletMP";
    PickupType[PickupType["AmmoGrenadeLauncherMP"] = 2753668402] = "AmmoGrenadeLauncherMP";
})(PickupType || (PickupType = {}));
