// @nativewrappers/client/lib/weapon/WeaponLiveryColor.js
export var WeaponLiveryColor;
(function (WeaponLiveryColor) {
    WeaponLiveryColor[WeaponLiveryColor["Gray"] = 0] = "Gray";
    WeaponLiveryColor[WeaponLiveryColor["DarkGray"] = 1] = "DarkGray";
    WeaponLiveryColor[WeaponLiveryColor["Black"] = 2] = "Black";
    WeaponLiveryColor[WeaponLiveryColor["White"] = 3] = "White";
    WeaponLiveryColor[WeaponLiveryColor["Blue"] = 4] = "Blue";
    WeaponLiveryColor[WeaponLiveryColor["Cyan"] = 5] = "Cyan";
    WeaponLiveryColor[WeaponLiveryColor["Aqua"] = 6] = "Aqua";
    WeaponLiveryColor[WeaponLiveryColor["CoolBlue"] = 7] = "CoolBlue";
    WeaponLiveryColor[WeaponLiveryColor["DarkBlue"] = 8] = "DarkBlue";
    WeaponLiveryColor[WeaponLiveryColor["RoyalBlue"] = 9] = "RoyalBlue";
    WeaponLiveryColor[WeaponLiveryColor["Plum"] = 10] = "Plum";
    WeaponLiveryColor[WeaponLiveryColor["DarkPurple"] = 11] = "DarkPurple";
    WeaponLiveryColor[WeaponLiveryColor["Purple"] = 12] = "Purple";
    WeaponLiveryColor[WeaponLiveryColor["Red"] = 13] = "Red";
    WeaponLiveryColor[WeaponLiveryColor["WineRed"] = 14] = "WineRed";
    WeaponLiveryColor[WeaponLiveryColor["Magenta"] = 15] = "Magenta";
    WeaponLiveryColor[WeaponLiveryColor["Pink"] = 16] = "Pink";
    WeaponLiveryColor[WeaponLiveryColor["Salmon"] = 17] = "Salmon";
    WeaponLiveryColor[WeaponLiveryColor["HotPink"] = 18] = "HotPink";
    WeaponLiveryColor[WeaponLiveryColor["RustOrange"] = 19] = "RustOrange";
    WeaponLiveryColor[WeaponLiveryColor["Brown"] = 20] = "Brown";
    WeaponLiveryColor[WeaponLiveryColor["Earth"] = 21] = "Earth";
    WeaponLiveryColor[WeaponLiveryColor["Orange"] = 22] = "Orange";
    WeaponLiveryColor[WeaponLiveryColor["LightOrange"] = 23] = "LightOrange";
    WeaponLiveryColor[WeaponLiveryColor["DarkYellow"] = 24] = "DarkYellow";
    WeaponLiveryColor[WeaponLiveryColor["Yellow"] = 25] = "Yellow";
    WeaponLiveryColor[WeaponLiveryColor["LightBrown"] = 26] = "LightBrown";
    WeaponLiveryColor[WeaponLiveryColor["LimeGreen"] = 27] = "LimeGreen";
    WeaponLiveryColor[WeaponLiveryColor["Olive"] = 28] = "Olive";
    WeaponLiveryColor[WeaponLiveryColor["Moss"] = 29] = "Moss";
    WeaponLiveryColor[WeaponLiveryColor["Turquoise"] = 30] = "Turquoise";
    WeaponLiveryColor[WeaponLiveryColor["DarkGreen"] = 31] = "DarkGreen";
})(WeaponLiveryColor || (WeaponLiveryColor = {}));
