// @nativewrappers/client/lib/weapon/WeaponTint.js
export var WeaponTint;
(function (WeaponTint) {
    WeaponTint[WeaponTint["Mk2ClassicBlack"] = 0] = "Mk2ClassicBlack";
    WeaponTint[WeaponTint["Mk2ClassicGray"] = 1] = "Mk2ClassicGray";
    WeaponTint[WeaponTint["Mk2ClassicTwoTone"] = 2] = "Mk2ClassicTwoTone";
    WeaponTint[WeaponTint["Mk2ClassicWhite"] = 3] = "Mk2ClassicWhite";
    WeaponTint[WeaponTint["Mk2ClassicBeige"] = 4] = "Mk2ClassicBeige";
    WeaponTint[WeaponTint["Mk2ClassicGreen"] = 5] = "Mk2ClassicGreen";
    WeaponTint[WeaponTint["Mk2ClassicBlue"] = 6] = "Mk2ClassicBlue";
    WeaponTint[WeaponTint["Mk2ClassicEarth"] = 7] = "Mk2ClassicEarth";
    WeaponTint[WeaponTint["Mk2ClassicBrownAndBlack"] = 8] = "Mk2ClassicBrownAndBlack";
    WeaponTint[WeaponTint["Mk2RedContrast"] = 9] = "Mk2RedContrast";
    WeaponTint[WeaponTint["Mk2BlueContrast"] = 10] = "Mk2BlueContrast";
    WeaponTint[WeaponTint["Mk2YellowContrast"] = 11] = "Mk2YellowContrast";
    WeaponTint[WeaponTint["Mk2OrangeContrast"] = 12] = "Mk2OrangeContrast";
    WeaponTint[WeaponTint["Mk2BoldPink"] = 13] = "Mk2BoldPink";
    WeaponTint[WeaponTint["Mk2BoldPurpleAndYellow"] = 14] = "Mk2BoldPurpleAndYellow";
    WeaponTint[WeaponTint["Mk2BoldOrange"] = 15] = "Mk2BoldOrange";
    WeaponTint[WeaponTint["Mk2BoldGreenAndPurple"] = 16] = "Mk2BoldGreenAndPurple";
    WeaponTint[WeaponTint["Mk2BoldRedFeatures"] = 17] = "Mk2BoldRedFeatures";
    WeaponTint[WeaponTint["Mk2BoldGreenFeatures"] = 18] = "Mk2BoldGreenFeatures";
    WeaponTint[WeaponTint["Mk2BoldCyanFeatures"] = 19] = "Mk2BoldCyanFeatures";
    WeaponTint[WeaponTint["Mk2BoldYellowFeatures"] = 20] = "Mk2BoldYellowFeatures";
    WeaponTint[WeaponTint["Mk2BoldRedAndWhite"] = 21] = "Mk2BoldRedAndWhite";
    WeaponTint[WeaponTint["Mk2BoldBlueAndWhite"] = 22] = "Mk2BoldBlueAndWhite";
    WeaponTint[WeaponTint["Mk2MetallicGold"] = 23] = "Mk2MetallicGold";
    WeaponTint[WeaponTint["Mk2MetallicPlatinum"] = 24] = "Mk2MetallicPlatinum";
    WeaponTint[WeaponTint["Mk2MetallicGrayAndLilac"] = 25] = "Mk2MetallicGrayAndLilac";
    WeaponTint[WeaponTint["Mk2MetallicPurpleAndLime"] = 26] = "Mk2MetallicPurpleAndLime";
    WeaponTint[WeaponTint["Mk2MetallicRed"] = 27] = "Mk2MetallicRed";
    WeaponTint[WeaponTint["Mk2MetallicGreen"] = 28] = "Mk2MetallicGreen";
    WeaponTint[WeaponTint["Mk2MetallicBlue"] = 29] = "Mk2MetallicBlue";
    WeaponTint[WeaponTint["Mk2MetallicWhiteAndAqua"] = 30] = "Mk2MetallicWhiteAndAqua";
    WeaponTint[WeaponTint["Mk2MetallicRedAndYellow"] = 31] = "Mk2MetallicRedAndYellow";
    WeaponTint[WeaponTint["Normal"] = 0] = "Normal";
    WeaponTint[WeaponTint["Green"] = 1] = "Green";
    WeaponTint[WeaponTint["Gold"] = 2] = "Gold";
    WeaponTint[WeaponTint["Pink"] = 3] = "Pink";
    WeaponTint[WeaponTint["Army"] = 4] = "Army";
    WeaponTint[WeaponTint["LSPD"] = 5] = "LSPD";
    WeaponTint[WeaponTint["Orange"] = 6] = "Orange";
    WeaponTint[WeaponTint["Platinum"] = 7] = "Platinum";
})(WeaponTint || (WeaponTint = {}));
