// @nativewrappers/client/lib/weapon/WeaponGroup.js
export var WeaponGroup;
(function (WeaponGroup) {
    WeaponGroup[WeaponGroup["Unarmed"] = 2685387236] = "Unarmed";
    WeaponGroup[WeaponGroup["Melee"] = 3566412244] = "Melee";
    WeaponGroup[WeaponGroup["Pistol"] = 416676503] = "Pistol";
    WeaponGroup[WeaponGroup["SMG"] = 3337201093] = "SMG";
    WeaponGroup[WeaponGroup["AssaultRifle"] = 970310034] = "AssaultRifle";
    WeaponGroup[WeaponGroup["DigiScanner"] = 3539449195] = "DigiScanner";
    WeaponGroup[WeaponGroup["FireExtinguisher"] = 4257178988] = "FireExtinguisher";
    WeaponGroup[WeaponGroup["MG"] = 1159398588] = "MG";
    WeaponGroup[WeaponGroup["NightVision"] = 3493187224] = "NightVision";
    WeaponGroup[WeaponGroup["Parachute"] = 431593103] = "Parachute";
    WeaponGroup[WeaponGroup["Shotgun"] = 860033945] = "Shotgun";
    WeaponGroup[WeaponGroup["Sniper"] = 3082541095] = "Sniper";
    WeaponGroup[WeaponGroup["Stungun"] = 690389602] = "Stungun";
    WeaponGroup[WeaponGroup["Heavy"] = 2725924767] = "Heavy";
    WeaponGroup[WeaponGroup["Thrown"] = 1548507267] = "Thrown";
    WeaponGroup[WeaponGroup["PetrolCan"] = 1595662460] = "PetrolCan";
})(WeaponGroup || (WeaponGroup = {}));
