// @nativewrappers/fivem/enums/BadgeStyle.js
var BadgeStyle = /* @__PURE__ */ ((BadgeStyle2) => {
  BadgeStyle2[BadgeStyle2["None"] = 0] = "None";
  BadgeStyle2[BadgeStyle2["Lock"] = 1] = "Lock";
  BadgeStyle2[BadgeStyle2["Star"] = 2] = "Star";
  BadgeStyle2[BadgeStyle2["Warning"] = 3] = "Warning";
  BadgeStyle2[BadgeStyle2["Crown"] = 4] = "Crown";
  BadgeStyle2[BadgeStyle2["MedalBronze"] = 5] = "MedalBronze";
  BadgeStyle2[BadgeStyle2["MedalGold"] = 6] = "MedalGold";
  BadgeStyle2[BadgeStyle2["MedalSilver"] = 7] = "MedalSilver";
  BadgeStyle2[BadgeStyle2["Cash"] = 8] = "Cash";
  BadgeStyle2[BadgeStyle2["Coke"] = 9] = "Coke";
  BadgeStyle2[BadgeStyle2["Heroin"] = 10] = "Heroin";
  BadgeStyle2[BadgeStyle2["Meth"] = 11] = "Meth";
  BadgeStyle2[BadgeStyle2["Weed"] = 12] = "Weed";
  BadgeStyle2[BadgeStyle2["Ammo"] = 13] = "Ammo";
  BadgeStyle2[BadgeStyle2["Armor"] = 14] = "Armor";
  BadgeStyle2[BadgeStyle2["Barber"] = 15] = "Barber";
  BadgeStyle2[BadgeStyle2["Clothing"] = 16] = "Clothing";
  BadgeStyle2[BadgeStyle2["Franklin"] = 17] = "Franklin";
  BadgeStyle2[BadgeStyle2["Bike"] = 18] = "Bike";
  BadgeStyle2[BadgeStyle2["Car"] = 19] = "Car";
  BadgeStyle2[BadgeStyle2["Gun"] = 20] = "Gun";
  BadgeStyle2[BadgeStyle2["HealthHeart"] = 21] = "HealthHeart";
  BadgeStyle2[BadgeStyle2["MakeupBrush"] = 22] = "MakeupBrush";
  BadgeStyle2[BadgeStyle2["Mask"] = 23] = "Mask";
  BadgeStyle2[BadgeStyle2["Michael"] = 24] = "Michael";
  BadgeStyle2[BadgeStyle2["Tattoo"] = 25] = "Tattoo";
  BadgeStyle2[BadgeStyle2["Tick"] = 26] = "Tick";
  BadgeStyle2[BadgeStyle2["Trevor"] = 27] = "Trevor";
  BadgeStyle2[BadgeStyle2["Female"] = 28] = "Female";
  BadgeStyle2[BadgeStyle2["Male"] = 29] = "Male";
  BadgeStyle2[BadgeStyle2["LockArena"] = 30] = "LockArena";
  BadgeStyle2[BadgeStyle2["Adversary"] = 31] = "Adversary";
  BadgeStyle2[BadgeStyle2["BaseJumping"] = 32] = "BaseJumping";
  BadgeStyle2[BadgeStyle2["Briefcase"] = 33] = "Briefcase";
  BadgeStyle2[BadgeStyle2["MissionStar"] = 34] = "MissionStar";
  BadgeStyle2[BadgeStyle2["Deathmatch"] = 35] = "Deathmatch";
  BadgeStyle2[BadgeStyle2["Castle"] = 36] = "Castle";
  BadgeStyle2[BadgeStyle2["Trophy"] = 37] = "Trophy";
  BadgeStyle2[BadgeStyle2["RaceFlag"] = 38] = "RaceFlag";
  BadgeStyle2[BadgeStyle2["RaceFlagPlane"] = 39] = "RaceFlagPlane";
  BadgeStyle2[BadgeStyle2["RaceFlagBicycle"] = 40] = "RaceFlagBicycle";
  BadgeStyle2[BadgeStyle2["RaceFlagPerson"] = 41] = "RaceFlagPerson";
  BadgeStyle2[BadgeStyle2["RaceFlagCar"] = 42] = "RaceFlagCar";
  BadgeStyle2[BadgeStyle2["RaceFlagBoatAnchor"] = 43] = "RaceFlagBoatAnchor";
  BadgeStyle2[BadgeStyle2["Rockstar"] = 44] = "Rockstar";
  BadgeStyle2[BadgeStyle2["Stunt"] = 45] = "Stunt";
  BadgeStyle2[BadgeStyle2["StuntPremium"] = 46] = "StuntPremium";
  BadgeStyle2[BadgeStyle2["RaceFlagStuntJump"] = 47] = "RaceFlagStuntJump";
  BadgeStyle2[BadgeStyle2["Shield"] = 48] = "Shield";
  BadgeStyle2[BadgeStyle2["TeamDeathmatch"] = 49] = "TeamDeathmatch";
  BadgeStyle2[BadgeStyle2["VehicleDeathmatch"] = 50] = "VehicleDeathmatch";
  BadgeStyle2[BadgeStyle2["MpAmmoPickup"] = 51] = "MpAmmoPickup";
  BadgeStyle2[BadgeStyle2["MpAmmo"] = 52] = "MpAmmo";
  BadgeStyle2[BadgeStyle2["MpCash"] = 53] = "MpCash";
  BadgeStyle2[BadgeStyle2["MpRp"] = 54] = "MpRp";
  BadgeStyle2[BadgeStyle2["MpSpectating"] = 55] = "MpSpectating";
  BadgeStyle2[BadgeStyle2["Sale"] = 56] = "Sale";
  BadgeStyle2[BadgeStyle2["GlobeWhite"] = 57] = "GlobeWhite";
  BadgeStyle2[BadgeStyle2["GlobeRed"] = 58] = "GlobeRed";
  BadgeStyle2[BadgeStyle2["GlobeBlue"] = 59] = "GlobeBlue";
  BadgeStyle2[BadgeStyle2["GlobeYellow"] = 60] = "GlobeYellow";
  BadgeStyle2[BadgeStyle2["GlobeGreen"] = 61] = "GlobeGreen";
  BadgeStyle2[BadgeStyle2["GlobeOrange"] = 62] = "GlobeOrange";
  BadgeStyle2[BadgeStyle2["InvArmWrestling"] = 63] = "InvArmWrestling";
  BadgeStyle2[BadgeStyle2["InvBasejump"] = 64] = "InvBasejump";
  BadgeStyle2[BadgeStyle2["InvMission"] = 65] = "InvMission";
  BadgeStyle2[BadgeStyle2["InvDarts"] = 66] = "InvDarts";
  BadgeStyle2[BadgeStyle2["InvDeathmatch"] = 67] = "InvDeathmatch";
  BadgeStyle2[BadgeStyle2["InvDrug"] = 68] = "InvDrug";
  BadgeStyle2[BadgeStyle2["InvCastle"] = 69] = "InvCastle";
  BadgeStyle2[BadgeStyle2["InvGolf"] = 70] = "InvGolf";
  BadgeStyle2[BadgeStyle2["InvBike"] = 71] = "InvBike";
  BadgeStyle2[BadgeStyle2["InvBoat"] = 72] = "InvBoat";
  BadgeStyle2[BadgeStyle2["InvAnchor"] = 73] = "InvAnchor";
  BadgeStyle2[BadgeStyle2["InvCar"] = 74] = "InvCar";
  BadgeStyle2[BadgeStyle2["InvDollar"] = 75] = "InvDollar";
  BadgeStyle2[BadgeStyle2["InvCoke"] = 76] = "InvCoke";
  BadgeStyle2[BadgeStyle2["InvKey"] = 77] = "InvKey";
  BadgeStyle2[BadgeStyle2["InvData"] = 78] = "InvData";
  BadgeStyle2[BadgeStyle2["InvHeli"] = 79] = "InvHeli";
  BadgeStyle2[BadgeStyle2["InvHeorin"] = 80] = "InvHeorin";
  BadgeStyle2[BadgeStyle2["InvKeycard"] = 81] = "InvKeycard";
  BadgeStyle2[BadgeStyle2["InvMeth"] = 82] = "InvMeth";
  BadgeStyle2[BadgeStyle2["InvBriefcase"] = 83] = "InvBriefcase";
  BadgeStyle2[BadgeStyle2["InvLink"] = 84] = "InvLink";
  BadgeStyle2[BadgeStyle2["InvPerson"] = 85] = "InvPerson";
  BadgeStyle2[BadgeStyle2["InvPlane"] = 86] = "InvPlane";
  BadgeStyle2[BadgeStyle2["InvPlane2"] = 87] = "InvPlane2";
  BadgeStyle2[BadgeStyle2["InvQuestionmark"] = 88] = "InvQuestionmark";
  BadgeStyle2[BadgeStyle2["InvRemote"] = 89] = "InvRemote";
  BadgeStyle2[BadgeStyle2["InvSafe"] = 90] = "InvSafe";
  BadgeStyle2[BadgeStyle2["InvSteerWheel"] = 91] = "InvSteerWheel";
  BadgeStyle2[BadgeStyle2["InvWeapon"] = 92] = "InvWeapon";
  BadgeStyle2[BadgeStyle2["InvWeed"] = 93] = "InvWeed";
  BadgeStyle2[BadgeStyle2["InvRaceFlagPlane"] = 94] = "InvRaceFlagPlane";
  BadgeStyle2[BadgeStyle2["InvRaceFlagBicycle"] = 95] = "InvRaceFlagBicycle";
  BadgeStyle2[BadgeStyle2["InvRaceFlagBoatAnchor"] = 96] = "InvRaceFlagBoatAnchor";
  BadgeStyle2[BadgeStyle2["InvRaceFlagPerson"] = 97] = "InvRaceFlagPerson";
  BadgeStyle2[BadgeStyle2["InvRaceFlagCar"] = 98] = "InvRaceFlagCar";
  BadgeStyle2[BadgeStyle2["InvRaceFlagHelmet"] = 99] = "InvRaceFlagHelmet";
  BadgeStyle2[BadgeStyle2["InvShootingRange"] = 100] = "InvShootingRange";
  BadgeStyle2[BadgeStyle2["InvSurvival"] = 101] = "InvSurvival";
  BadgeStyle2[BadgeStyle2["InvTeamDeathmatch"] = 102] = "InvTeamDeathmatch";
  BadgeStyle2[BadgeStyle2["InvTennis"] = 103] = "InvTennis";
  BadgeStyle2[BadgeStyle2["InvVehicleDeathmatch"] = 104] = "InvVehicleDeathmatch";
  BadgeStyle2[BadgeStyle2["AudioMute"] = 105] = "AudioMute";
  BadgeStyle2[BadgeStyle2["AudioInactive"] = 106] = "AudioInactive";
  BadgeStyle2[BadgeStyle2["AudioVol1"] = 107] = "AudioVol1";
  BadgeStyle2[BadgeStyle2["AudioVol2"] = 108] = "AudioVol2";
  BadgeStyle2[BadgeStyle2["AudioVol3"] = 109] = "AudioVol3";
  BadgeStyle2[BadgeStyle2["CountryUsa"] = 110] = "CountryUsa";
  BadgeStyle2[BadgeStyle2["CountryUk"] = 111] = "CountryUk";
  BadgeStyle2[BadgeStyle2["CountrySweden"] = 112] = "CountrySweden";
  BadgeStyle2[BadgeStyle2["CountryKorea"] = 113] = "CountryKorea";
  BadgeStyle2[BadgeStyle2["CountryJapan"] = 114] = "CountryJapan";
  BadgeStyle2[BadgeStyle2["CountryItaly"] = 115] = "CountryItaly";
  BadgeStyle2[BadgeStyle2["CountryGermany"] = 116] = "CountryGermany";
  BadgeStyle2[BadgeStyle2["CountryFrance"] = 117] = "CountryFrance";
  BadgeStyle2[BadgeStyle2["BrandAlbany"] = 118] = "BrandAlbany";
  BadgeStyle2[BadgeStyle2["BrandAnnis"] = 119] = "BrandAnnis";
  BadgeStyle2[BadgeStyle2["BrandBanshee"] = 120] = "BrandBanshee";
  BadgeStyle2[BadgeStyle2["BrandBenefactor"] = 121] = "BrandBenefactor";
  BadgeStyle2[BadgeStyle2["BrandBf"] = 122] = "BrandBf";
  BadgeStyle2[BadgeStyle2["BrandBollokan"] = 123] = "BrandBollokan";
  BadgeStyle2[BadgeStyle2["BrandBravado"] = 124] = "BrandBravado";
  BadgeStyle2[BadgeStyle2["BrandBrute"] = 125] = "BrandBrute";
  BadgeStyle2[BadgeStyle2["BrandBuckingham"] = 126] = "BrandBuckingham";
  BadgeStyle2[BadgeStyle2["BrandCanis"] = 127] = "BrandCanis";
  BadgeStyle2[BadgeStyle2["BrandChariot"] = 128] = "BrandChariot";
  BadgeStyle2[BadgeStyle2["BrandCheval"] = 129] = "BrandCheval";
  BadgeStyle2[BadgeStyle2["BrandClassique"] = 130] = "BrandClassique";
  BadgeStyle2[BadgeStyle2["BrandCoil"] = 131] = "BrandCoil";
  BadgeStyle2[BadgeStyle2["BrandDeclasse"] = 132] = "BrandDeclasse";
  BadgeStyle2[BadgeStyle2["BrandDewbauchee"] = 133] = "BrandDewbauchee";
  BadgeStyle2[BadgeStyle2["BrandDilettante"] = 134] = "BrandDilettante";
  BadgeStyle2[BadgeStyle2["BrandDinka"] = 135] = "BrandDinka";
  BadgeStyle2[BadgeStyle2["BrandDundreary"] = 136] = "BrandDundreary";
  BadgeStyle2[BadgeStyle2["BrandEmporer"] = 137] = "BrandEmporer";
  BadgeStyle2[BadgeStyle2["BrandEnus"] = 138] = "BrandEnus";
  BadgeStyle2[BadgeStyle2["BrandFathom"] = 139] = "BrandFathom";
  BadgeStyle2[BadgeStyle2["BrandGalivanter"] = 140] = "BrandGalivanter";
  BadgeStyle2[BadgeStyle2["BrandGrotti"] = 141] = "BrandGrotti";
  BadgeStyle2[BadgeStyle2["BrandGrotti2"] = 142] = "BrandGrotti2";
  BadgeStyle2[BadgeStyle2["BrandHijak"] = 143] = "BrandHijak";
  BadgeStyle2[BadgeStyle2["BrandHvy"] = 144] = "BrandHvy";
  BadgeStyle2[BadgeStyle2["BrandImponte"] = 145] = "BrandImponte";
  BadgeStyle2[BadgeStyle2["BrandInvetero"] = 146] = "BrandInvetero";
  BadgeStyle2[BadgeStyle2["BrandJacksheepe"] = 147] = "BrandJacksheepe";
  BadgeStyle2[BadgeStyle2["BrandLcc"] = 148] = "BrandLcc";
  BadgeStyle2[BadgeStyle2["BrandJobuilt"] = 149] = "BrandJobuilt";
  BadgeStyle2[BadgeStyle2["BrandKarin"] = 150] = "BrandKarin";
  BadgeStyle2[BadgeStyle2["BrandLampadati"] = 151] = "BrandLampadati";
  BadgeStyle2[BadgeStyle2["BrandMaibatsu"] = 152] = "BrandMaibatsu";
  BadgeStyle2[BadgeStyle2["BrandMammoth"] = 153] = "BrandMammoth";
  BadgeStyle2[BadgeStyle2["BrandMtl"] = 154] = "BrandMtl";
  BadgeStyle2[BadgeStyle2["BrandNagasaki"] = 155] = "BrandNagasaki";
  BadgeStyle2[BadgeStyle2["BrandObey"] = 156] = "BrandObey";
  BadgeStyle2[BadgeStyle2["BrandOcelot"] = 157] = "BrandOcelot";
  BadgeStyle2[BadgeStyle2["BrandOverflod"] = 158] = "BrandOverflod";
  BadgeStyle2[BadgeStyle2["BrandPed"] = 159] = "BrandPed";
  BadgeStyle2[BadgeStyle2["BrandPegassi"] = 160] = "BrandPegassi";
  BadgeStyle2[BadgeStyle2["BrandPfister"] = 161] = "BrandPfister";
  BadgeStyle2[BadgeStyle2["BrandPrincipe"] = 162] = "BrandPrincipe";
  BadgeStyle2[BadgeStyle2["BrandProgen"] = 163] = "BrandProgen";
  BadgeStyle2[BadgeStyle2["BrandProgen2"] = 164] = "BrandProgen2";
  BadgeStyle2[BadgeStyle2["BrandRune"] = 165] = "BrandRune";
  BadgeStyle2[BadgeStyle2["BrandSchyster"] = 166] = "BrandSchyster";
  BadgeStyle2[BadgeStyle2["BrandShitzu"] = 167] = "BrandShitzu";
  BadgeStyle2[BadgeStyle2["BrandSpeedophile"] = 168] = "BrandSpeedophile";
  BadgeStyle2[BadgeStyle2["BrandStanley"] = 169] = "BrandStanley";
  BadgeStyle2[BadgeStyle2["BrandTruffade"] = 170] = "BrandTruffade";
  BadgeStyle2[BadgeStyle2["BrandUbermacht"] = 171] = "BrandUbermacht";
  BadgeStyle2[BadgeStyle2["BrandVapid"] = 172] = "BrandVapid";
  BadgeStyle2[BadgeStyle2["BrandVulcar"] = 173] = "BrandVulcar";
  BadgeStyle2[BadgeStyle2["BrandWeeny"] = 174] = "BrandWeeny";
  BadgeStyle2[BadgeStyle2["BrandWestern"] = 175] = "BrandWestern";
  BadgeStyle2[BadgeStyle2["BrandWesternmotorcycle"] = 176] = "BrandWesternmotorcycle";
  BadgeStyle2[BadgeStyle2["BrandWillard"] = 177] = "BrandWillard";
  BadgeStyle2[BadgeStyle2["BrandZirconium"] = 178] = "BrandZirconium";
  BadgeStyle2[BadgeStyle2["Info"] = 179] = "Info";
  return BadgeStyle2;
})(BadgeStyle || {});
export {
  BadgeStyle
};
