// @nativewrappers/fivem/enums/Blip.js
var BlipColor = /* @__PURE__ */ ((BlipColor2) => {
  BlipColor2[BlipColor2["White"] = 0] = "White";
  BlipColor2[BlipColor2["Red"] = 1] = "Red";
  BlipColor2[BlipColor2["Green"] = 2] = "Green";
  BlipColor2[BlipColor2["Blue"] = 3] = "Blue";
  BlipColor2[BlipColor2["MichaelBlue"] = 42] = "MichaelBlue";
  BlipColor2[BlipColor2["FranklinGreen"] = 43] = "FranklinGreen";
  BlipColor2[BlipColor2["TrevorOrange"] = 44] = "TrevorOrange";
  BlipColor2[BlipColor2["Yellow"] = 66] = "Yellow";
  return BlipColor2;
})(BlipColor || {});
var BlipSprite = /* @__PURE__ */ ((BlipSprite2) => {
  BlipSprite2[BlipSprite2["Standard"] = 1] = "Standard";
  BlipSprite2[BlipSprite2["BigBlip"] = 2] = "BigBlip";
  BlipSprite2[BlipSprite2["PoliceOfficer"] = 3] = "PoliceOfficer";
  BlipSprite2[BlipSprite2["PoliceArea"] = 4] = "PoliceArea";
  BlipSprite2[BlipSprite2["Square"] = 5] = "Square";
  BlipSprite2[BlipSprite2["Player"] = 6] = "Player";
  BlipSprite2[BlipSprite2["North"] = 7] = "North";
  BlipSprite2[BlipSprite2["Waypoint"] = 8] = "Waypoint";
  BlipSprite2[BlipSprite2["BigCircle"] = 9] = "BigCircle";
  BlipSprite2[BlipSprite2["BigCircleOutline"] = 10] = "BigCircleOutline";
  BlipSprite2[BlipSprite2["ArrowUpOutlined"] = 11] = "ArrowUpOutlined";
  BlipSprite2[BlipSprite2["ArrowDownOutlined"] = 12] = "ArrowDownOutlined";
  BlipSprite2[BlipSprite2["ArrowUp"] = 13] = "ArrowUp";
  BlipSprite2[BlipSprite2["ArrowDown"] = 14] = "ArrowDown";
  BlipSprite2[BlipSprite2["PoliceHelicopterAnimated"] = 15] = "PoliceHelicopterAnimated";
  BlipSprite2[BlipSprite2["Jet"] = 16] = "Jet";
  BlipSprite2[BlipSprite2["Number1"] = 17] = "Number1";
  BlipSprite2[BlipSprite2["Number2"] = 18] = "Number2";
  BlipSprite2[BlipSprite2["Number3"] = 19] = "Number3";
  BlipSprite2[BlipSprite2["Number4"] = 20] = "Number4";
  BlipSprite2[BlipSprite2["Number5"] = 21] = "Number5";
  BlipSprite2[BlipSprite2["Number6"] = 22] = "Number6";
  BlipSprite2[BlipSprite2["Number7"] = 23] = "Number7";
  BlipSprite2[BlipSprite2["Number8"] = 24] = "Number8";
  BlipSprite2[BlipSprite2["Number9"] = 25] = "Number9";
  BlipSprite2[BlipSprite2["Number10"] = 26] = "Number10";
  BlipSprite2[BlipSprite2["GTAOCrew"] = 27] = "GTAOCrew";
  BlipSprite2[BlipSprite2["GTAOFriendly"] = 28] = "GTAOFriendly";
  BlipSprite2[BlipSprite2["Lift"] = 36] = "Lift";
  BlipSprite2[BlipSprite2["RaceFinish"] = 38] = "RaceFinish";
  BlipSprite2[BlipSprite2["Safehouse"] = 40] = "Safehouse";
  BlipSprite2[BlipSprite2["PoliceOfficer2"] = 41] = "PoliceOfficer2";
  BlipSprite2[BlipSprite2["PoliceCarDot"] = 42] = "PoliceCarDot";
  BlipSprite2[BlipSprite2["PoliceHelicopter"] = 43] = "PoliceHelicopter";
  BlipSprite2[BlipSprite2["ChatBubble"] = 47] = "ChatBubble";
  BlipSprite2[BlipSprite2["Garage2"] = 50] = "Garage2";
  BlipSprite2[BlipSprite2["Drugs"] = 51] = "Drugs";
  BlipSprite2[BlipSprite2["Store"] = 52] = "Store";
  BlipSprite2[BlipSprite2["PoliceCar"] = 56] = "PoliceCar";
  BlipSprite2[BlipSprite2["PolicePlayer"] = 58] = "PolicePlayer";
  BlipSprite2[BlipSprite2["PoliceStation"] = 60] = "PoliceStation";
  BlipSprite2[BlipSprite2["Hospital"] = 61] = "Hospital";
  BlipSprite2[BlipSprite2["Helicopter"] = 64] = "Helicopter";
  BlipSprite2[BlipSprite2["StrangersAndFreaks"] = 65] = "StrangersAndFreaks";
  BlipSprite2[BlipSprite2["ArmoredTruck"] = 66] = "ArmoredTruck";
  BlipSprite2[BlipSprite2["TowTruck"] = 68] = "TowTruck";
  BlipSprite2[BlipSprite2["Barber"] = 71] = "Barber";
  BlipSprite2[BlipSprite2["LosSantosCustoms"] = 72] = "LosSantosCustoms";
  BlipSprite2[BlipSprite2["Clothes"] = 73] = "Clothes";
  BlipSprite2[BlipSprite2["TattooParlor"] = 75] = "TattooParlor";
  BlipSprite2[BlipSprite2["Simeon"] = 76] = "Simeon";
  BlipSprite2[BlipSprite2["Lester"] = 77] = "Lester";
  BlipSprite2[BlipSprite2["Michael"] = 78] = "Michael";
  BlipSprite2[BlipSprite2["Trevor"] = 79] = "Trevor";
  BlipSprite2[BlipSprite2["Rampage"] = 84] = "Rampage";
  BlipSprite2[BlipSprite2["VinewoodTours"] = 85] = "VinewoodTours";
  BlipSprite2[BlipSprite2["Lamar"] = 86] = "Lamar";
  BlipSprite2[BlipSprite2["Franklin"] = 88] = "Franklin";
  BlipSprite2[BlipSprite2["Chinese"] = 89] = "Chinese";
  BlipSprite2[BlipSprite2["Airport"] = 90] = "Airport";
  BlipSprite2[BlipSprite2["Bar"] = 93] = "Bar";
  BlipSprite2[BlipSprite2["BaseJump"] = 94] = "BaseJump";
  BlipSprite2[BlipSprite2["CarWash"] = 100] = "CarWash";
  BlipSprite2[BlipSprite2["ComedyClub"] = 102] = "ComedyClub";
  BlipSprite2[BlipSprite2["Dart"] = 103] = "Dart";
  BlipSprite2[BlipSprite2["FIB"] = 106] = "FIB";
  BlipSprite2[BlipSprite2["DollarSign"] = 108] = "DollarSign";
  BlipSprite2[BlipSprite2["Golf"] = 109] = "Golf";
  BlipSprite2[BlipSprite2["AmmuNation"] = 110] = "AmmuNation";
  BlipSprite2[BlipSprite2["Exile"] = 112] = "Exile";
  BlipSprite2[BlipSprite2["ShootingRange"] = 119] = "ShootingRange";
  BlipSprite2[BlipSprite2["Solomon"] = 120] = "Solomon";
  BlipSprite2[BlipSprite2["StripClub"] = 121] = "StripClub";
  BlipSprite2[BlipSprite2["Tennis"] = 122] = "Tennis";
  BlipSprite2[BlipSprite2["Triathlon"] = 126] = "Triathlon";
  BlipSprite2[BlipSprite2["OffRoadRaceFinish"] = 127] = "OffRoadRaceFinish";
  BlipSprite2[BlipSprite2["Key"] = 134] = "Key";
  BlipSprite2[BlipSprite2["MovieTheater"] = 135] = "MovieTheater";
  BlipSprite2[BlipSprite2["Music"] = 136] = "Music";
  BlipSprite2[BlipSprite2["Marijuana"] = 140] = "Marijuana";
  BlipSprite2[BlipSprite2["Hunting"] = 141] = "Hunting";
  BlipSprite2[BlipSprite2["ArmsTraffickingGround"] = 147] = "ArmsTraffickingGround";
  BlipSprite2[BlipSprite2["Nigel"] = 149] = "Nigel";
  BlipSprite2[BlipSprite2["AssaultRifle"] = 150] = "AssaultRifle";
  BlipSprite2[BlipSprite2["Bat"] = 151] = "Bat";
  BlipSprite2[BlipSprite2["Grenade"] = 152] = "Grenade";
  BlipSprite2[BlipSprite2["Health"] = 153] = "Health";
  BlipSprite2[BlipSprite2["Knife"] = 154] = "Knife";
  BlipSprite2[BlipSprite2["Molotov"] = 155] = "Molotov";
  BlipSprite2[BlipSprite2["Pistol"] = 156] = "Pistol";
  BlipSprite2[BlipSprite2["RPG"] = 157] = "RPG";
  BlipSprite2[BlipSprite2["Shotgun"] = 158] = "Shotgun";
  BlipSprite2[BlipSprite2["SMG"] = 159] = "SMG";
  BlipSprite2[BlipSprite2["Sniper"] = 160] = "Sniper";
  BlipSprite2[BlipSprite2["SonicWave"] = 161] = "SonicWave";
  BlipSprite2[BlipSprite2["PointOfInterest"] = 162] = "PointOfInterest";
  BlipSprite2[BlipSprite2["GTAOPassive"] = 163] = "GTAOPassive";
  BlipSprite2[BlipSprite2["GTAOUsingMenu"] = 164] = "GTAOUsingMenu";
  BlipSprite2[BlipSprite2["Link"] = 171] = "Link";
  BlipSprite2[BlipSprite2["Minigun"] = 173] = "Minigun";
  BlipSprite2[BlipSprite2["GrenadeLauncher"] = 174] = "GrenadeLauncher";
  BlipSprite2[BlipSprite2["Armor"] = 175] = "Armor";
  BlipSprite2[BlipSprite2["Castle"] = 176] = "Castle";
  BlipSprite2[BlipSprite2["Camera"] = 184] = "Camera";
  BlipSprite2[BlipSprite2["Handcuffs"] = 188] = "Handcuffs";
  BlipSprite2[BlipSprite2["Yoga"] = 197] = "Yoga";
  BlipSprite2[BlipSprite2["Cab"] = 198] = "Cab";
  BlipSprite2[BlipSprite2["Number11"] = 199] = "Number11";
  BlipSprite2[BlipSprite2["Number12"] = 200] = "Number12";
  BlipSprite2[BlipSprite2["Number13"] = 201] = "Number13";
  BlipSprite2[BlipSprite2["Number14"] = 202] = "Number14";
  BlipSprite2[BlipSprite2["Number15"] = 203] = "Number15";
  BlipSprite2[BlipSprite2["Number16"] = 204] = "Number16";
  BlipSprite2[BlipSprite2["Shrink"] = 205] = "Shrink";
  BlipSprite2[BlipSprite2["Epsilon"] = 206] = "Epsilon";
  BlipSprite2[BlipSprite2["PersonalVehicleCar"] = 225] = "PersonalVehicleCar";
  BlipSprite2[BlipSprite2["PersonalVehicleBike"] = 226] = "PersonalVehicleBike";
  BlipSprite2[BlipSprite2["Custody"] = 237] = "Custody";
  BlipSprite2[BlipSprite2["ArmsTraffickingAir"] = 251] = "ArmsTraffickingAir";
  BlipSprite2[BlipSprite2["Fairground"] = 266] = "Fairground";
  BlipSprite2[BlipSprite2["PropertyManagement"] = 267] = "PropertyManagement";
  BlipSprite2[BlipSprite2["Altruist"] = 269] = "Altruist";
  BlipSprite2[BlipSprite2["Enemy"] = 270] = "Enemy";
  BlipSprite2[BlipSprite2["Chop"] = 273] = "Chop";
  BlipSprite2[BlipSprite2["Dead"] = 274] = "Dead";
  BlipSprite2[BlipSprite2["Hooker"] = 279] = "Hooker";
  BlipSprite2[BlipSprite2["Friend"] = 280] = "Friend";
  BlipSprite2[BlipSprite2["BountyHit"] = 303] = "BountyHit";
  BlipSprite2[BlipSprite2["GTAOMission"] = 304] = "GTAOMission";
  BlipSprite2[BlipSprite2["GTAOSurvival"] = 305] = "GTAOSurvival";
  BlipSprite2[BlipSprite2["CrateDrop"] = 306] = "CrateDrop";
  BlipSprite2[BlipSprite2["PlaneDrop"] = 307] = "PlaneDrop";
  BlipSprite2[BlipSprite2["Sub"] = 308] = "Sub";
  BlipSprite2[BlipSprite2["Race"] = 309] = "Race";
  BlipSprite2[BlipSprite2["Deathmatch"] = 310] = "Deathmatch";
  BlipSprite2[BlipSprite2["ArmWrestling"] = 311] = "ArmWrestling";
  BlipSprite2[BlipSprite2["AmmuNationShootingRange"] = 313] = "AmmuNationShootingRange";
  BlipSprite2[BlipSprite2["RaceAir"] = 314] = "RaceAir";
  BlipSprite2[BlipSprite2["RaceCar"] = 315] = "RaceCar";
  BlipSprite2[BlipSprite2["RaceSea"] = 316] = "RaceSea";
  BlipSprite2[BlipSprite2["GarbageTruck"] = 318] = "GarbageTruck";
  BlipSprite2[BlipSprite2["SafehouseForSale"] = 350] = "SafehouseForSale";
  BlipSprite2[BlipSprite2["Package"] = 351] = "Package";
  BlipSprite2[BlipSprite2["MartinMadrazo"] = 352] = "MartinMadrazo";
  BlipSprite2[BlipSprite2["EnemyHelicopter"] = 353] = "EnemyHelicopter";
  BlipSprite2[BlipSprite2["Boost"] = 354] = "Boost";
  BlipSprite2[BlipSprite2["Devin"] = 355] = "Devin";
  BlipSprite2[BlipSprite2["Marina"] = 356] = "Marina";
  BlipSprite2[BlipSprite2["Garage"] = 357] = "Garage";
  BlipSprite2[BlipSprite2["GolfFlag"] = 358] = "GolfFlag";
  BlipSprite2[BlipSprite2["Hangar"] = 359] = "Hangar";
  BlipSprite2[BlipSprite2["Helipad"] = 360] = "Helipad";
  BlipSprite2[BlipSprite2["JerryCan"] = 361] = "JerryCan";
  BlipSprite2[BlipSprite2["Masks"] = 362] = "Masks";
  BlipSprite2[BlipSprite2["HeistSetup"] = 363] = "HeistSetup";
  BlipSprite2[BlipSprite2["Incapacitated"] = 364] = "Incapacitated";
  BlipSprite2[BlipSprite2["PickupSpawn"] = 365] = "PickupSpawn";
  BlipSprite2[BlipSprite2["BoilerSuit"] = 366] = "BoilerSuit";
  BlipSprite2[BlipSprite2["Completed"] = 367] = "Completed";
  BlipSprite2[BlipSprite2["Rockets"] = 368] = "Rockets";
  BlipSprite2[BlipSprite2["GarageForSale"] = 369] = "GarageForSale";
  BlipSprite2[BlipSprite2["HelipadForSale"] = 370] = "HelipadForSale";
  BlipSprite2[BlipSprite2["MarinaForSale"] = 371] = "MarinaForSale";
  BlipSprite2[BlipSprite2["HangarForSale"] = 372] = "HangarForSale";
  BlipSprite2[BlipSprite2["Business"] = 374] = "Business";
  BlipSprite2[BlipSprite2["BusinessForSale"] = 375] = "BusinessForSale";
  BlipSprite2[BlipSprite2["RaceBike"] = 376] = "RaceBike";
  BlipSprite2[BlipSprite2["Parachute"] = 377] = "Parachute";
  BlipSprite2[BlipSprite2["TeamDeathmatch"] = 378] = "TeamDeathmatch";
  BlipSprite2[BlipSprite2["RaceFoot"] = 379] = "RaceFoot";
  BlipSprite2[BlipSprite2["VehicleDeathmatch"] = 380] = "VehicleDeathmatch";
  BlipSprite2[BlipSprite2["Barry"] = 381] = "Barry";
  BlipSprite2[BlipSprite2["Dom"] = 382] = "Dom";
  BlipSprite2[BlipSprite2["MaryAnn"] = 383] = "MaryAnn";
  BlipSprite2[BlipSprite2["Cletus"] = 384] = "Cletus";
  BlipSprite2[BlipSprite2["Josh"] = 385] = "Josh";
  BlipSprite2[BlipSprite2["Minute"] = 386] = "Minute";
  BlipSprite2[BlipSprite2["Omega"] = 387] = "Omega";
  BlipSprite2[BlipSprite2["Tonya"] = 388] = "Tonya";
  BlipSprite2[BlipSprite2["Paparazzo"] = 389] = "Paparazzo";
  BlipSprite2[BlipSprite2["Crosshair"] = 390] = "Crosshair";
  BlipSprite2[BlipSprite2["Creator"] = 398] = "Creator";
  BlipSprite2[BlipSprite2["CreatorDirection"] = 399] = "CreatorDirection";
  BlipSprite2[BlipSprite2["Abigail"] = 400] = "Abigail";
  BlipSprite2[BlipSprite2["Blimp"] = 401] = "Blimp";
  BlipSprite2[BlipSprite2["Repair"] = 402] = "Repair";
  BlipSprite2[BlipSprite2["Testosterone"] = 403] = "Testosterone";
  BlipSprite2[BlipSprite2["Dinghy"] = 404] = "Dinghy";
  BlipSprite2[BlipSprite2["Fanatic"] = 405] = "Fanatic";
  BlipSprite2[BlipSprite2["Information"] = 407] = "Information";
  BlipSprite2[BlipSprite2["CaptureBriefcase"] = 408] = "CaptureBriefcase";
  BlipSprite2[BlipSprite2["LastTeamStanding"] = 409] = "LastTeamStanding";
  BlipSprite2[BlipSprite2["Boat"] = 410] = "Boat";
  BlipSprite2[BlipSprite2["CaptureHouse"] = 411] = "CaptureHouse";
  BlipSprite2[BlipSprite2["JerryCan2"] = 415] = "JerryCan2";
  BlipSprite2[BlipSprite2["RP"] = 416] = "RP";
  BlipSprite2[BlipSprite2["GTAOPlayerSafehouse"] = 417] = "GTAOPlayerSafehouse";
  BlipSprite2[BlipSprite2["GTAOPlayerSafehouseDead"] = 418] = "GTAOPlayerSafehouseDead";
  BlipSprite2[BlipSprite2["CaptureAmericanFlag"] = 419] = "CaptureAmericanFlag";
  BlipSprite2[BlipSprite2["CaptureFlag"] = 420] = "CaptureFlag";
  BlipSprite2[BlipSprite2["Tank"] = 421] = "Tank";
  BlipSprite2[BlipSprite2["HelicopterAnimated"] = 422] = "HelicopterAnimated";
  BlipSprite2[BlipSprite2["Plane"] = 423] = "Plane";
  BlipSprite2[BlipSprite2["PlayerNoColor"] = 425] = "PlayerNoColor";
  BlipSprite2[BlipSprite2["GunCar"] = 426] = "GunCar";
  BlipSprite2[BlipSprite2["Speedboat"] = 427] = "Speedboat";
  BlipSprite2[BlipSprite2["Heist"] = 428] = "Heist";
  BlipSprite2[BlipSprite2["Stopwatch"] = 430] = "Stopwatch";
  BlipSprite2[BlipSprite2["DollarSignCircled"] = 431] = "DollarSignCircled";
  BlipSprite2[BlipSprite2["Crosshair2"] = 432] = "Crosshair2";
  BlipSprite2[BlipSprite2["DollarSignSquared"] = 434] = "DollarSignSquared";
  return BlipSprite2;
})(BlipSprite || {});
export {
  BlipColor,
  BlipSprite
};
