// @nativewrappers/fivem/enums/CloudHat.js
var CloudHat = /* @__PURE__ */ ((CloudHat2) => {
  CloudHat2[CloudHat2["Unknown"] = 1] = "Unknown";
  CloudHat2[CloudHat2["Altostratus"] = 2] = "Altostratus";
  CloudHat2[CloudHat2["Cirrus"] = 3] = "Cirrus";
  CloudHat2[CloudHat2["Cirrocumulus"] = 4] = "Cirrocumulus";
  CloudHat2[CloudHat2["Clear"] = 5] = "Clear";
  CloudHat2[CloudHat2["Cloudy"] = 6] = "Cloudy";
  CloudHat2[CloudHat2["Contrails"] = 7] = "Contrails";
  CloudHat2[CloudHat2["Horizon"] = 8] = "Horizon";
  CloudHat2[CloudHat2["HorizonBand1"] = 9] = "HorizonBand1";
  CloudHat2[CloudHat2["HorizonBand2"] = 10] = "HorizonBand2";
  CloudHat2[CloudHat2["HorizonBand3"] = 11] = "HorizonBand3";
  CloudHat2[CloudHat2["Horsey"] = 12] = "Horsey";
  CloudHat2[CloudHat2["Nimbus"] = 13] = "Nimbus";
  CloudHat2[CloudHat2["Puffs"] = 14] = "Puffs";
  CloudHat2[CloudHat2["Rain"] = 15] = "Rain";
  CloudHat2[CloudHat2["Snowy"] = 16] = "Snowy";
  CloudHat2[CloudHat2["Stormy"] = 17] = "Stormy";
  CloudHat2[CloudHat2["Stratoscumulus"] = 18] = "Stratoscumulus";
  CloudHat2[CloudHat2["Stripey"] = 19] = "Stripey";
  CloudHat2[CloudHat2["Shower"] = 20] = "Shower";
  CloudHat2[CloudHat2["Wispy"] = 21] = "Wispy";
  return CloudHat2;
})(CloudHat || {});
export {
  CloudHat
};
