// @nativewrappers/fivem/enums/ExplosionType.js
var ExplosionType = /* @__PURE__ */ ((ExplosionType2) => {
  ExplosionType2[ExplosionType2["Grenade"] = 0] = "Grenade";
  ExplosionType2[ExplosionType2["GrenadeL"] = 1] = "GrenadeL";
  ExplosionType2[ExplosionType2["StickyBomb"] = 2] = "StickyBomb";
  ExplosionType2[ExplosionType2["Molotov1"] = 3] = "Molotov1";
  ExplosionType2[ExplosionType2["Rocket"] = 4] = "Rocket";
  ExplosionType2[ExplosionType2["TankShell"] = 5] = "TankShell";
  ExplosionType2[ExplosionType2["HiOctane"] = 6] = "HiOctane";
  ExplosionType2[ExplosionType2["Car"] = 7] = "Car";
  ExplosionType2[ExplosionType2["Plane"] = 8] = "Plane";
  ExplosionType2[ExplosionType2["PetrolPump"] = 9] = "PetrolPump";
  ExplosionType2[ExplosionType2["Bike"] = 10] = "Bike";
  ExplosionType2[ExplosionType2["Steam"] = 11] = "Steam";
  ExplosionType2[ExplosionType2["Flame"] = 12] = "Flame";
  ExplosionType2[ExplosionType2["WaterHydrant"] = 13] = "WaterHydrant";
  ExplosionType2[ExplosionType2["GasCanister"] = 14] = "GasCanister";
  ExplosionType2[ExplosionType2["Boat"] = 15] = "Boat";
  ExplosionType2[ExplosionType2["ShipDestroy"] = 16] = "ShipDestroy";
  ExplosionType2[ExplosionType2["Truck"] = 17] = "Truck";
  ExplosionType2[ExplosionType2["Bullet"] = 18] = "Bullet";
  ExplosionType2[ExplosionType2["SmokeGL"] = 19] = "SmokeGL";
  ExplosionType2[ExplosionType2["SmokeG"] = 20] = "SmokeG";
  ExplosionType2[ExplosionType2["BZGas"] = 21] = "BZGas";
  ExplosionType2[ExplosionType2["Flare"] = 22] = "Flare";
  ExplosionType2[ExplosionType2["GasCanister2"] = 23] = "GasCanister2";
  ExplosionType2[ExplosionType2["Extinguisher"] = 24] = "Extinguisher";
  ExplosionType2[ExplosionType2["ProgramAR"] = 25] = "ProgramAR";
  ExplosionType2[ExplosionType2["Train"] = 26] = "Train";
  ExplosionType2[ExplosionType2["Barrel"] = 27] = "Barrel";
  ExplosionType2[ExplosionType2["Propane"] = 28] = "Propane";
  ExplosionType2[ExplosionType2["Blimp"] = 29] = "Blimp";
  ExplosionType2[ExplosionType2["FlameExplode"] = 30] = "FlameExplode";
  ExplosionType2[ExplosionType2["Tanker"] = 31] = "Tanker";
  ExplosionType2[ExplosionType2["PlaneRocket"] = 32] = "PlaneRocket";
  ExplosionType2[ExplosionType2["VehicleBullet"] = 33] = "VehicleBullet";
  ExplosionType2[ExplosionType2["GasTank"] = 34] = "GasTank";
  ExplosionType2[ExplosionType2["FireWork"] = 35] = "FireWork";
  ExplosionType2[ExplosionType2["SnowBall"] = 36] = "SnowBall";
  ExplosionType2[ExplosionType2["ProxMine"] = 37] = "ProxMine";
  ExplosionType2[ExplosionType2["Valkyrie"] = 38] = "Valkyrie";
  return ExplosionType2;
})(ExplosionType || {});
export {
  ExplosionType
};
