// @nativewrappers/fivem/enums/FiringPattern.js
var FiringPattern = /* @__PURE__ */ ((FiringPattern2) => {
  FiringPattern2[FiringPattern2["Default"] = 0] = "Default";
  FiringPattern2[FiringPattern2["FullAuto"] = 3337513804] = "FullAuto";
  FiringPattern2[FiringPattern2["BurstFire"] = 3607063905] = "BurstFire";
  FiringPattern2[FiringPattern2["BurstInCover"] = 40051185] = "BurstInCover";
  FiringPattern2[FiringPattern2["BurstFireDriveby"] = 3541198322] = "BurstFireDriveby";
  FiringPattern2[FiringPattern2["FromGround"] = 577037782] = "FromGround";
  FiringPattern2[FiringPattern2["DelayFireByOneSec"] = 2055493265] = "DelayFireByOneSec";
  FiringPattern2[FiringPattern2["SingleShot"] = 1566631136] = "SingleShot";
  FiringPattern2[FiringPattern2["BurstFirePistol"] = 2685983626] = "BurstFirePistol";
  FiringPattern2[FiringPattern2["BurstFireSMG"] = 3507334638] = "BurstFireSMG";
  FiringPattern2[FiringPattern2["BurstFireRifle"] = 2624893958] = "BurstFireRifle";
  FiringPattern2[FiringPattern2["BurstFireMG"] = 3044263348] = "BurstFireMG";
  FiringPattern2[FiringPattern2["BurstFirePumpShotGun"] = 12239771] = "BurstFirePumpShotGun";
  FiringPattern2[FiringPattern2["BurstFireHeli"] = 2437838959] = "BurstFireHeli";
  FiringPattern2[FiringPattern2["BurstFireMicro"] = 1122960381] = "BurstFireMicro";
  FiringPattern2[FiringPattern2["BurstFireBursts"] = 445831135] = "BurstFireBursts";
  FiringPattern2[FiringPattern2["BurstFireTank"] = 3804904049] = "BurstFireTank";
  return FiringPattern2;
})(FiringPattern || {});
export {
  FiringPattern
};
