// @nativewrappers/fivem/enums/HudComponent.js
var HudComponent = /* @__PURE__ */ ((HudComponent2) => {
  HudComponent2[HudComponent2["WantedStars"] = 1] = "WantedStars";
  HudComponent2[HudComponent2["WeaponIcon"] = 2] = "WeaponIcon";
  HudComponent2[HudComponent2["Cash"] = 3] = "Cash";
  HudComponent2[HudComponent2["MpCash"] = 4] = "MpCash";
  HudComponent2[HudComponent2["MpMessage"] = 5] = "MpMessage";
  HudComponent2[HudComponent2["VehicleName"] = 6] = "VehicleName";
  HudComponent2[HudComponent2["AreaName"] = 7] = "AreaName";
  HudComponent2[HudComponent2["Unused"] = 8] = "Unused";
  HudComponent2[HudComponent2["StreetName"] = 9] = "StreetName";
  HudComponent2[HudComponent2["HelpText"] = 10] = "HelpText";
  HudComponent2[HudComponent2["FloatingHelpText1"] = 11] = "FloatingHelpText1";
  HudComponent2[HudComponent2["FloatingHelpText2"] = 12] = "FloatingHelpText2";
  HudComponent2[HudComponent2["CashChange"] = 13] = "CashChange";
  HudComponent2[HudComponent2["Reticle"] = 14] = "Reticle";
  HudComponent2[HudComponent2["SubtitleText"] = 15] = "SubtitleText";
  HudComponent2[HudComponent2["RadioStationsWheel"] = 16] = "RadioStationsWheel";
  HudComponent2[HudComponent2["Saving"] = 17] = "Saving";
  HudComponent2[HudComponent2["GamingStreamUnusde"] = 18] = "GamingStreamUnusde";
  HudComponent2[HudComponent2["WeaponWheel"] = 19] = "WeaponWheel";
  HudComponent2[HudComponent2["WeaponWheelStats"] = 20] = "WeaponWheelStats";
  HudComponent2[HudComponent2["DrugsPurse01"] = 21] = "DrugsPurse01";
  HudComponent2[HudComponent2["DrugsPurse02"] = 22] = "DrugsPurse02";
  HudComponent2[HudComponent2["DrugsPurse03"] = 23] = "DrugsPurse03";
  HudComponent2[HudComponent2["DrugsPurse04"] = 24] = "DrugsPurse04";
  HudComponent2[HudComponent2["MpTagCashFromBank"] = 25] = "MpTagCashFromBank";
  HudComponent2[HudComponent2["MpTagPackages"] = 26] = "MpTagPackages";
  HudComponent2[HudComponent2["MpTagCuffKeys"] = 27] = "MpTagCuffKeys";
  HudComponent2[HudComponent2["MpTagDownloadData"] = 28] = "MpTagDownloadData";
  HudComponent2[HudComponent2["MpTagIfPedFollowing"] = 29] = "MpTagIfPedFollowing";
  HudComponent2[HudComponent2["MpTagKeyCard"] = 30] = "MpTagKeyCard";
  HudComponent2[HudComponent2["MpTagRandomObject"] = 31] = "MpTagRandomObject";
  HudComponent2[HudComponent2["MpTagRemoteControl"] = 32] = "MpTagRemoteControl";
  HudComponent2[HudComponent2["MpTagCashFromSafe"] = 33] = "MpTagCashFromSafe";
  HudComponent2[HudComponent2["MpTagWeaponsPackage"] = 34] = "MpTagWeaponsPackage";
  HudComponent2[HudComponent2["MpTagKeys"] = 35] = "MpTagKeys";
  HudComponent2[HudComponent2["MpVehicle"] = 36] = "MpVehicle";
  HudComponent2[HudComponent2["MpVehicleHeli"] = 37] = "MpVehicleHeli";
  HudComponent2[HudComponent2["MpVehiclePlane"] = 38] = "MpVehiclePlane";
  HudComponent2[HudComponent2["PlayerSwitchAlert"] = 39] = "PlayerSwitchAlert";
  HudComponent2[HudComponent2["MpRankBar"] = 40] = "MpRankBar";
  HudComponent2[HudComponent2["DirectorMode"] = 41] = "DirectorMode";
  HudComponent2[HudComponent2["ReplayController"] = 42] = "ReplayController";
  HudComponent2[HudComponent2["ReplayMouse"] = 43] = "ReplayMouse";
  HudComponent2[HudComponent2["ReplayHeader"] = 44] = "ReplayHeader";
  HudComponent2[HudComponent2["ReplayOptions"] = 45] = "ReplayOptions";
  HudComponent2[HudComponent2["ReplayHelpText"] = 46] = "ReplayHelpText";
  HudComponent2[HudComponent2["ReplayMiscText"] = 47] = "ReplayMiscText";
  HudComponent2[HudComponent2["ReplayTopLine"] = 48] = "ReplayTopLine";
  HudComponent2[HudComponent2["ReplayBottomLine"] = 49] = "ReplayBottomLine";
  HudComponent2[HudComponent2["ReplayLeftBar"] = 50] = "ReplayLeftBar";
  HudComponent2[HudComponent2["ReplayTimer"] = 51] = "ReplayTimer";
  return HudComponent2;
})(HudComponent || {});
export {
  HudComponent
};
