// @nativewrappers/fivem/enums/MarkerType.js
var MarkerType = /* @__PURE__ */ ((MarkerType2) => {
  MarkerType2[MarkerType2["UpsideDownCone"] = 0] = "UpsideDownCone";
  MarkerType2[MarkerType2["VerticalCylinder"] = 1] = "VerticalCylinder";
  MarkerType2[MarkerType2["ThickChevronUp"] = 2] = "ThickChevronUp";
  MarkerType2[MarkerType2["ThinChevronUp"] = 3] = "ThinChevronUp";
  MarkerType2[MarkerType2["CheckeredFlagRect"] = 4] = "CheckeredFlagRect";
  MarkerType2[MarkerType2["CheckeredFlagCircle"] = 5] = "CheckeredFlagCircle";
  MarkerType2[MarkerType2["VerticleCircle"] = 6] = "VerticleCircle";
  MarkerType2[MarkerType2["PlaneModel"] = 7] = "PlaneModel";
  MarkerType2[MarkerType2["LostMCDark"] = 8] = "LostMCDark";
  MarkerType2[MarkerType2["LostMCLight"] = 9] = "LostMCLight";
  MarkerType2[MarkerType2["Number0"] = 10] = "Number0";
  MarkerType2[MarkerType2["Number1"] = 11] = "Number1";
  MarkerType2[MarkerType2["Number2"] = 12] = "Number2";
  MarkerType2[MarkerType2["Number3"] = 13] = "Number3";
  MarkerType2[MarkerType2["Number4"] = 14] = "Number4";
  MarkerType2[MarkerType2["Number5"] = 15] = "Number5";
  MarkerType2[MarkerType2["Number6"] = 16] = "Number6";
  MarkerType2[MarkerType2["Number7"] = 17] = "Number7";
  MarkerType2[MarkerType2["Number8"] = 18] = "Number8";
  MarkerType2[MarkerType2["Number9"] = 19] = "Number9";
  MarkerType2[MarkerType2["ChevronUpx1"] = 20] = "ChevronUpx1";
  MarkerType2[MarkerType2["ChevronUpx2"] = 21] = "ChevronUpx2";
  MarkerType2[MarkerType2["ChevronUpx3"] = 22] = "ChevronUpx3";
  MarkerType2[MarkerType2["HorizontalCircleFat"] = 23] = "HorizontalCircleFat";
  MarkerType2[MarkerType2["ReplayIcon"] = 24] = "ReplayIcon";
  MarkerType2[MarkerType2["HorizontalCircleSkinny"] = 25] = "HorizontalCircleSkinny";
  MarkerType2[MarkerType2["HorizontalCircleSkinnyArrow"] = 26] = "HorizontalCircleSkinnyArrow";
  MarkerType2[MarkerType2["HorizontalSplitArrowCircle"] = 27] = "HorizontalSplitArrowCircle";
  MarkerType2[MarkerType2["DebugSphere"] = 28] = "DebugSphere";
  MarkerType2[MarkerType2["DollarSign"] = 29] = "DollarSign";
  MarkerType2[MarkerType2["HorizontalBars"] = 30] = "HorizontalBars";
  MarkerType2[MarkerType2["WolfHead"] = 31] = "WolfHead";
  MarkerType2[MarkerType2["QuestionMark"] = 32] = "QuestionMark";
  MarkerType2[MarkerType2["PlaneSymbol"] = 33] = "PlaneSymbol";
  MarkerType2[MarkerType2["HelicopterSymbol"] = 34] = "HelicopterSymbol";
  MarkerType2[MarkerType2["BoatSymbol"] = 35] = "BoatSymbol";
  MarkerType2[MarkerType2["CarSymbol"] = 36] = "CarSymbol";
  MarkerType2[MarkerType2["MotorcycleSymbol"] = 37] = "MotorcycleSymbol";
  MarkerType2[MarkerType2["BikeSymbol"] = 38] = "BikeSymbol";
  MarkerType2[MarkerType2["TruckSymbol"] = 39] = "TruckSymbol";
  MarkerType2[MarkerType2["ParachuteSymbol"] = 40] = "ParachuteSymbol";
  MarkerType2[MarkerType2["SawbladeSymbol"] = 41] = "SawbladeSymbol";
  return MarkerType2;
})(MarkerType || {});
export {
  MarkerType
};
