// @nativewrappers/fivem/enums/PickupType.js
var PickupType = /* @__PURE__ */ ((PickupType2) => {
  PickupType2[PickupType2["CustomScript"] = 738282662] = "CustomScript";
  PickupType2[PickupType2["VehicleCustomScript"] = 2780351145] = "VehicleCustomScript";
  PickupType2[PickupType2["Parachute"] = 1735599485] = "Parachute";
  PickupType2[PickupType2["PortablePackage"] = 2158727964] = "PortablePackage";
  PickupType2[PickupType2["PortableCrateUnfixed"] = 1852930709] = "PortableCrateUnfixed";
  PickupType2[PickupType2["Health"] = 2406513688] = "Health";
  PickupType2[PickupType2["HealthSnack"] = 483577702] = "HealthSnack";
  PickupType2[PickupType2["Armour"] = 1274757841] = "Armour";
  PickupType2[PickupType2["MoneyCase"] = 3463437675] = "MoneyCase";
  PickupType2[PickupType2["MoneySecurityCase"] = 3732468094] = "MoneySecurityCase";
  PickupType2[PickupType2["MoneyVariable"] = 4263048111] = "MoneyVariable";
  PickupType2[PickupType2["MoneyMedBag"] = 341217064] = "MoneyMedBag";
  PickupType2[PickupType2["MoneyPurse"] = 513448440] = "MoneyPurse";
  PickupType2[PickupType2["MoneyDepBag"] = 545862290] = "MoneyDepBag";
  PickupType2[PickupType2["MoneyWallet"] = 1575005502] = "MoneyWallet";
  PickupType2[PickupType2["MoneyPaperBag"] = 1897726628] = "MoneyPaperBag";
  PickupType2[PickupType2["WeaponPistol"] = 4189041807] = "WeaponPistol";
  PickupType2[PickupType2["WeaponCombatPistol"] = 2305275123] = "WeaponCombatPistol";
  PickupType2[PickupType2["WeaponAPPistol"] = 996550793] = "WeaponAPPistol";
  PickupType2[PickupType2["WeaponSNSPistol"] = 3317114643] = "WeaponSNSPistol";
  PickupType2[PickupType2["WeaponHeavyPistol"] = 2633054488] = "WeaponHeavyPistol";
  PickupType2[PickupType2["WeaponMicroSMG"] = 496339155] = "WeaponMicroSMG";
  PickupType2[PickupType2["WeaponSMG"] = 978070226] = "WeaponSMG";
  PickupType2[PickupType2["WeaponMG"] = 2244651441] = "WeaponMG";
  PickupType2[PickupType2["WeaponCombatMG"] = 2995980820] = "WeaponCombatMG";
  PickupType2[PickupType2["WeaponAssaultRifle"] = 4080829360] = "WeaponAssaultRifle";
  PickupType2[PickupType2["WeaponCarbineRifle"] = 3748731225] = "WeaponCarbineRifle";
  PickupType2[PickupType2["WeaponAdvancedRifle"] = 2998219358] = "WeaponAdvancedRifle";
  PickupType2[PickupType2["WeaponSpecialCarbine"] = 157823901] = "WeaponSpecialCarbine";
  PickupType2[PickupType2["WeaponBullpupRifle"] = 2170382056] = "WeaponBullpupRifle";
  PickupType2[PickupType2["WeaponPumpShotgun"] = 2838846925] = "WeaponPumpShotgun";
  PickupType2[PickupType2["WeaponSawnoffShotgun"] = 2528383651] = "WeaponSawnoffShotgun";
  PickupType2[PickupType2["WeaponAssaultShotgun"] = 2459552091] = "WeaponAssaultShotgun";
  PickupType2[PickupType2["WeaponSniperRifle"] = 4264178988] = "WeaponSniperRifle";
  PickupType2[PickupType2["WeaponHeavySniper"] = 1765114797] = "WeaponHeavySniper";
  PickupType2[PickupType2["WeaponGrenadeLauncher"] = 779501861] = "WeaponGrenadeLauncher";
  PickupType2[PickupType2["WeaponRPG"] = 1295434569] = "WeaponRPG";
  PickupType2[PickupType2["WeaponMinigun"] = 792114228] = "WeaponMinigun";
  PickupType2[PickupType2["WeaponGrenade"] = 1577485217] = "WeaponGrenade";
  PickupType2[PickupType2["WeaponStickyBomb"] = 2081529176] = "WeaponStickyBomb";
  PickupType2[PickupType2["WeaponSmokeGrenade"] = 483787975] = "WeaponSmokeGrenade";
  PickupType2[PickupType2["WeaponMolotov"] = 768803961] = "WeaponMolotov";
  PickupType2[PickupType2["WeaponPetrolCan"] = 3332236287] = "WeaponPetrolCan";
  PickupType2[PickupType2["WeaponKnife"] = 663586612] = "WeaponKnife";
  PickupType2[PickupType2["WeaponNightstick"] = 1587637620] = "WeaponNightstick";
  PickupType2[PickupType2["WeaponBat"] = 2179883038] = "WeaponBat";
  PickupType2[PickupType2["WeaponCrowbar"] = 2267924616] = "WeaponCrowbar";
  PickupType2[PickupType2["WeaponGolfclub"] = 2297080999] = "WeaponGolfclub";
  PickupType2[PickupType2["WeaponBottle"] = 4199656437] = "WeaponBottle";
  PickupType2[PickupType2["VehicleWeaponPistol"] = 2773149623] = "VehicleWeaponPistol";
  PickupType2[PickupType2["VehicleWeaponCombatPistol"] = 3500855031] = "VehicleWeaponCombatPistol";
  PickupType2[PickupType2["VehicleWeaponAPPistol"] = 3431676165] = "VehicleWeaponAPPistol";
  PickupType2[PickupType2["VehicleWeaponMicroSMG"] = 3094015579] = "VehicleWeaponMicroSMG";
  PickupType2[PickupType2["VehicleWeaponSawnoffShotgun"] = 772217690] = "VehicleWeaponSawnoffShotgun";
  PickupType2[PickupType2["VehicleWeaponGrenade"] = 2803366040] = "VehicleWeaponGrenade";
  PickupType2[PickupType2["VehicleWeaponSmokeGrenade"] = 1705498857] = "VehicleWeaponSmokeGrenade";
  PickupType2[PickupType2["VehicleWeaponStickyBomb"] = 746606563] = "VehicleWeaponStickyBomb";
  PickupType2[PickupType2["VehicleWeaponMolotov"] = 2228647636] = "VehicleWeaponMolotov";
  PickupType2[PickupType2["VehicleHealth"] = 160266735] = "VehicleHealth";
  PickupType2[PickupType2["AmmoPistol"] = 544828034] = "AmmoPistol";
  PickupType2[PickupType2["AmmoSMG"] = 292537574] = "AmmoSMG";
  PickupType2[PickupType2["AmmoMG"] = 3730366643] = "AmmoMG";
  PickupType2[PickupType2["AmmoRifle"] = 3837603782] = "AmmoRifle";
  PickupType2[PickupType2["AmmoShotgun"] = 2012476125] = "AmmoShotgun";
  PickupType2[PickupType2["AmmoSniper"] = 3224170789] = "AmmoSniper";
  PickupType2[PickupType2["AmmoGrenadeLauncher"] = 2283450536] = "AmmoGrenadeLauncher";
  PickupType2[PickupType2["AmmoRPG"] = 2223210455] = "AmmoRPG";
  PickupType2[PickupType2["AmmoMinigun"] = 4065984953] = "AmmoMinigun";
  PickupType2[PickupType2["AmmoMissileMP"] = 4187887056] = "AmmoMissileMP";
  PickupType2[PickupType2["AmmoBulletMP"] = 1426343849] = "AmmoBulletMP";
  PickupType2[PickupType2["AmmoGrenadeLauncherMP"] = 2753668402] = "AmmoGrenadeLauncherMP";
  return PickupType2;
})(PickupType || {});
export {
  PickupType
};
