// @nativewrappers/fivem/enums/RadioStation.js
var RadioStation = /* @__PURE__ */ ((RadioStation2) => {
  RadioStation2["LosSantosRockRadio"] = "RADIO_01_CLASS_ROCK";
  RadioStation2["NonStopPopFM"] = "RADIO_02_POP";
  RadioStation2["RadioLosSantos"] = "RADIO_03_HIPHOP_NEW";
  RadioStation2["ChannelX"] = "RADIO_04_PUNK";
  RadioStation2["WestCoastTalkRadio"] = "RADIO_05_TALK_01";
  RadioStation2["RebelRadio"] = "RADIO_06_COUNTRY";
  RadioStation2["SoulwaxFM"] = "RADIO_07_DANCE_01";
  RadioStation2["EastLosFM"] = "RADIO_08_MEXICAN";
  RadioStation2["WestCoastClassics"] = "RADIO_09_HIPHOP_OLD";
  RadioStation2["BlaineCountyRadio"] = "RADIO_11_TALK_02";
  RadioStation2["TheBlueArk"] = "RADIO_12_REGGAE";
  RadioStation2["WorldWideFM"] = "RADIO_13_JAZZ";
  RadioStation2["FlyloFM"] = "RADIO_14_DANCE_02";
  RadioStation2["TheLowdown"] = "RADIO_15_MOTOWN";
  RadioStation2["RadioMirrorPark"] = "RADIO_16_SILVERLAKE";
  RadioStation2["Space"] = "RADIO_17_FUNK";
  RadioStation2["VinewoodBoulevardRadio"] = "RADIO_18_90S_ROCK";
  RadioStation2["SelfRadio"] = "RADIO_19_USER";
  RadioStation2["TheLab"] = "RADIO_20_THELAB";
  RadioStation2["BlondedLosSantos"] = "RADIO_21_DLC_XM17";
  RadioStation2["LosSantosUndergroundRadio"] = "RADIO_22_DLC_BATTLE_MIX1_RADIO";
  RadioStation2["RadioOff"] = "OFF";
  return RadioStation2;
})(RadioStation || {});
export {
  RadioStation
};
