// @nativewrappers/fivem/enums/ScreenEffect.js
var ScreenEffect = /* @__PURE__ */ ((ScreenEffect2) => {
  ScreenEffect2[ScreenEffect2["SwitchHudIn"] = 0] = "SwitchHudIn";
  ScreenEffect2[ScreenEffect2["SwitchHudOut"] = 1] = "SwitchHudOut";
  ScreenEffect2[ScreenEffect2["FocusIn"] = 2] = "FocusIn";
  ScreenEffect2[ScreenEffect2["FocusOut"] = 3] = "FocusOut";
  ScreenEffect2[ScreenEffect2["MinigameEndNeutral"] = 4] = "MinigameEndNeutral";
  ScreenEffect2[ScreenEffect2["MinigameEndTrevor"] = 5] = "MinigameEndTrevor";
  ScreenEffect2[ScreenEffect2["MinigameEndFranklin"] = 6] = "MinigameEndFranklin";
  ScreenEffect2[ScreenEffect2["MinigameEndMichael"] = 7] = "MinigameEndMichael";
  ScreenEffect2[ScreenEffect2["MinigameTransitionOut"] = 8] = "MinigameTransitionOut";
  ScreenEffect2[ScreenEffect2["MinigameTransitionIn"] = 9] = "MinigameTransitionIn";
  ScreenEffect2[ScreenEffect2["SwitchShortNeutralIn"] = 10] = "SwitchShortNeutralIn";
  ScreenEffect2[ScreenEffect2["SwitchShortFranklinIn"] = 11] = "SwitchShortFranklinIn";
  ScreenEffect2[ScreenEffect2["SwitchShortTrevorIn"] = 12] = "SwitchShortTrevorIn";
  ScreenEffect2[ScreenEffect2["SwitchShortMichaelIn"] = 13] = "SwitchShortMichaelIn";
  ScreenEffect2[ScreenEffect2["SwitchOpenMichaelIn"] = 14] = "SwitchOpenMichaelIn";
  ScreenEffect2[ScreenEffect2["SwitchOpenFranklinIn"] = 15] = "SwitchOpenFranklinIn";
  ScreenEffect2[ScreenEffect2["SwitchOpenTrevorIn"] = 16] = "SwitchOpenTrevorIn";
  ScreenEffect2[ScreenEffect2["SwitchHudMichaelOut"] = 17] = "SwitchHudMichaelOut";
  ScreenEffect2[ScreenEffect2["SwitchHudFranklinOut"] = 18] = "SwitchHudFranklinOut";
  ScreenEffect2[ScreenEffect2["SwitchHudTrevorOut"] = 19] = "SwitchHudTrevorOut";
  ScreenEffect2[ScreenEffect2["SwitchShortFranklinMid"] = 20] = "SwitchShortFranklinMid";
  ScreenEffect2[ScreenEffect2["SwitchShortMichaelMid"] = 21] = "SwitchShortMichaelMid";
  ScreenEffect2[ScreenEffect2["SwitchShortTrevorMid"] = 22] = "SwitchShortTrevorMid";
  ScreenEffect2[ScreenEffect2["DeathFailOut"] = 23] = "DeathFailOut";
  ScreenEffect2[ScreenEffect2["CamPushInNeutral"] = 24] = "CamPushInNeutral";
  ScreenEffect2[ScreenEffect2["CamPushInFranklin"] = 25] = "CamPushInFranklin";
  ScreenEffect2[ScreenEffect2["CamPushInMichael"] = 26] = "CamPushInMichael";
  ScreenEffect2[ScreenEffect2["CamPushInTrevor"] = 27] = "CamPushInTrevor";
  ScreenEffect2[ScreenEffect2["SwitchSceneFranklin"] = 28] = "SwitchSceneFranklin";
  ScreenEffect2[ScreenEffect2["SwitchSceneTrevor"] = 29] = "SwitchSceneTrevor";
  ScreenEffect2[ScreenEffect2["SwitchSceneMichael"] = 30] = "SwitchSceneMichael";
  ScreenEffect2[ScreenEffect2["SwitchSceneNeutral"] = 31] = "SwitchSceneNeutral";
  ScreenEffect2[ScreenEffect2["MpCelebWin"] = 32] = "MpCelebWin";
  ScreenEffect2[ScreenEffect2["MpCelebWinOut"] = 33] = "MpCelebWinOut";
  ScreenEffect2[ScreenEffect2["MpCelebLose"] = 34] = "MpCelebLose";
  ScreenEffect2[ScreenEffect2["MpCelebLoseOut"] = 35] = "MpCelebLoseOut";
  ScreenEffect2[ScreenEffect2["DeathFailNeutralIn"] = 36] = "DeathFailNeutralIn";
  ScreenEffect2[ScreenEffect2["DeathFailMpDark"] = 37] = "DeathFailMpDark";
  ScreenEffect2[ScreenEffect2["DeathFailMpIn"] = 38] = "DeathFailMpIn";
  ScreenEffect2[ScreenEffect2["MpCelebPreloadFade"] = 39] = "MpCelebPreloadFade";
  ScreenEffect2[ScreenEffect2["PeyoteEndOut"] = 40] = "PeyoteEndOut";
  ScreenEffect2[ScreenEffect2["PeyoteEndIn"] = 41] = "PeyoteEndIn";
  ScreenEffect2[ScreenEffect2["PeyoteIn"] = 42] = "PeyoteIn";
  ScreenEffect2[ScreenEffect2["PeyoteOut"] = 43] = "PeyoteOut";
  ScreenEffect2[ScreenEffect2["MpRaceCrash"] = 44] = "MpRaceCrash";
  ScreenEffect2[ScreenEffect2["SuccessFranklin"] = 45] = "SuccessFranklin";
  ScreenEffect2[ScreenEffect2["SuccessTrevor"] = 46] = "SuccessTrevor";
  ScreenEffect2[ScreenEffect2["SuccessMichael"] = 47] = "SuccessMichael";
  ScreenEffect2[ScreenEffect2["DrugsMichaelAliensFightIn"] = 48] = "DrugsMichaelAliensFightIn";
  ScreenEffect2[ScreenEffect2["DrugsMichaelAliensFight"] = 49] = "DrugsMichaelAliensFight";
  ScreenEffect2[ScreenEffect2["DrugsMichaelAliensFightOut"] = 50] = "DrugsMichaelAliensFightOut";
  ScreenEffect2[ScreenEffect2["DrugsTrevorClownsFightIn"] = 51] = "DrugsTrevorClownsFightIn";
  ScreenEffect2[ScreenEffect2["DrugsTrevorClownsFight"] = 52] = "DrugsTrevorClownsFight";
  ScreenEffect2[ScreenEffect2["DrugsTrevorClownsFightOut"] = 53] = "DrugsTrevorClownsFightOut";
  ScreenEffect2[ScreenEffect2["HeistCelebPass"] = 54] = "HeistCelebPass";
  ScreenEffect2[ScreenEffect2["HeistCelebPassBw"] = 55] = "HeistCelebPassBw";
  ScreenEffect2[ScreenEffect2["HeistCelebEnd"] = 56] = "HeistCelebEnd";
  ScreenEffect2[ScreenEffect2["HeistCelebToast"] = 57] = "HeistCelebToast";
  ScreenEffect2[ScreenEffect2["MenuMgHeistIn"] = 58] = "MenuMgHeistIn";
  ScreenEffect2[ScreenEffect2["MenuMgTournamentIn"] = 59] = "MenuMgTournamentIn";
  ScreenEffect2[ScreenEffect2["MenuMgSelectionIn"] = 60] = "MenuMgSelectionIn";
  ScreenEffect2[ScreenEffect2["ChopVision"] = 61] = "ChopVision";
  ScreenEffect2[ScreenEffect2["DmtFlightIntro"] = 62] = "DmtFlightIntro";
  ScreenEffect2[ScreenEffect2["DmtFlight"] = 63] = "DmtFlight";
  ScreenEffect2[ScreenEffect2["DrugsDrivingIn"] = 64] = "DrugsDrivingIn";
  ScreenEffect2[ScreenEffect2["DrugsDrivingOut"] = 65] = "DrugsDrivingOut";
  ScreenEffect2[ScreenEffect2["SwitchOpenNeutralFib5"] = 66] = "SwitchOpenNeutralFib5";
  ScreenEffect2[ScreenEffect2["HeistLocate"] = 67] = "HeistLocate";
  ScreenEffect2[ScreenEffect2["MpJobLoad"] = 68] = "MpJobLoad";
  ScreenEffect2[ScreenEffect2["RaceTurbo"] = 69] = "RaceTurbo";
  ScreenEffect2[ScreenEffect2["MpIntroLogo"] = 70] = "MpIntroLogo";
  ScreenEffect2[ScreenEffect2["HeistTripSkipFade"] = 71] = "HeistTripSkipFade";
  ScreenEffect2[ScreenEffect2["MenuMgHeistOut"] = 72] = "MenuMgHeistOut";
  ScreenEffect2[ScreenEffect2["MpCoronaSwitch"] = 73] = "MpCoronaSwitch";
  ScreenEffect2[ScreenEffect2["MenuMgSelectionTint"] = 74] = "MenuMgSelectionTint";
  ScreenEffect2[ScreenEffect2["SuccessNeutral"] = 75] = "SuccessNeutral";
  ScreenEffect2[ScreenEffect2["ExplosionJosh3"] = 76] = "ExplosionJosh3";
  ScreenEffect2[ScreenEffect2["SniperOverlay"] = 77] = "SniperOverlay";
  ScreenEffect2[ScreenEffect2["RampageOut"] = 78] = "RampageOut";
  ScreenEffect2[ScreenEffect2["Rampage"] = 79] = "Rampage";
  ScreenEffect2[ScreenEffect2["DontTazemeBro"] = 80] = "DontTazemeBro";
  return ScreenEffect2;
})(ScreenEffect || {});
export {
  ScreenEffect
};
