// @nativewrappers/fivem/enums/SpeechModifier.js
var SpeechModifier = /* @__PURE__ */ ((SpeechModifier2) => {
  SpeechModifier2[SpeechModifier2["Standard"] = 0] = "Standard";
  SpeechModifier2[SpeechModifier2["AllowRepeat"] = 1] = "AllowRepeat";
  SpeechModifier2[SpeechModifier2["Beat"] = 2] = "Beat";
  SpeechModifier2[SpeechModifier2["Force"] = 3] = "Force";
  SpeechModifier2[SpeechModifier2["ForceFrontend"] = 4] = "ForceFrontend";
  SpeechModifier2[SpeechModifier2["ForceNoRepeatFrontend"] = 5] = "ForceNoRepeatFrontend";
  SpeechModifier2[SpeechModifier2["ForceNormal"] = 6] = "ForceNormal";
  SpeechModifier2[SpeechModifier2["ForceNormalClear"] = 7] = "ForceNormalClear";
  SpeechModifier2[SpeechModifier2["ForceNormalCritical"] = 8] = "ForceNormalCritical";
  SpeechModifier2[SpeechModifier2["ForceShouted"] = 9] = "ForceShouted";
  SpeechModifier2[SpeechModifier2["ForceShoutedClear"] = 10] = "ForceShoutedClear";
  SpeechModifier2[SpeechModifier2["ForceShoutedCritical"] = 11] = "ForceShoutedCritical";
  SpeechModifier2[SpeechModifier2["ForcePreloadOnly"] = 12] = "ForcePreloadOnly";
  SpeechModifier2[SpeechModifier2["Megaphone"] = 13] = "Megaphone";
  SpeechModifier2[SpeechModifier2["Helicopter"] = 14] = "Helicopter";
  SpeechModifier2[SpeechModifier2["ForceMegaphone"] = 15] = "ForceMegaphone";
  SpeechModifier2[SpeechModifier2["ForceHelicopter"] = 16] = "ForceHelicopter";
  SpeechModifier2[SpeechModifier2["Interrupt"] = 17] = "Interrupt";
  SpeechModifier2[SpeechModifier2["InterruptShouted"] = 18] = "InterruptShouted";
  SpeechModifier2[SpeechModifier2["InterruptShoutedClear"] = 19] = "InterruptShoutedClear";
  SpeechModifier2[SpeechModifier2["InterruptShoutedCritical"] = 20] = "InterruptShoutedCritical";
  SpeechModifier2[SpeechModifier2["InterruptNoForce"] = 21] = "InterruptNoForce";
  SpeechModifier2[SpeechModifier2["InterruptFrontend"] = 22] = "InterruptFrontend";
  SpeechModifier2[SpeechModifier2["InterruptNoForceFrontend"] = 23] = "InterruptNoForceFrontend";
  SpeechModifier2[SpeechModifier2["AddBlip"] = 24] = "AddBlip";
  SpeechModifier2[SpeechModifier2["AddBlipAllowRepeat"] = 25] = "AddBlipAllowRepeat";
  SpeechModifier2[SpeechModifier2["AddBlipForce"] = 26] = "AddBlipForce";
  SpeechModifier2[SpeechModifier2["AddBlipShouted"] = 27] = "AddBlipShouted";
  SpeechModifier2[SpeechModifier2["AddBlipShoutedForce"] = 28] = "AddBlipShoutedForce";
  SpeechModifier2[SpeechModifier2["AddBlipInterrupt"] = 29] = "AddBlipInterrupt";
  SpeechModifier2[SpeechModifier2["AddBlipInterruptForce"] = 30] = "AddBlipInterruptForce";
  SpeechModifier2[SpeechModifier2["ForcePreloadOnlyShouted"] = 31] = "ForcePreloadOnlyShouted";
  SpeechModifier2[SpeechModifier2["ForcePreloadOnlyShoutedClear"] = 32] = "ForcePreloadOnlyShoutedClear";
  SpeechModifier2[SpeechModifier2["ForcePreloadOnlyShoutedCritical"] = 33] = "ForcePreloadOnlyShoutedCritical";
  SpeechModifier2[SpeechModifier2["Shouted"] = 34] = "Shouted";
  SpeechModifier2[SpeechModifier2["ShoutedClear"] = 35] = "ShoutedClear";
  SpeechModifier2[SpeechModifier2["ShoutedCritical"] = 36] = "ShoutedCritical";
  return SpeechModifier2;
})(SpeechModifier || {});
export {
  SpeechModifier
};
