// @nativewrappers/fivem/enums/Vehicle.js
var CargobobHook = /* @__PURE__ */ ((CargobobHook2) => {
  CargobobHook2[CargobobHook2["Hook"] = 0] = "Hook";
  CargobobHook2[CargobobHook2["Magnet"] = 1] = "Magnet";
  return CargobobHook2;
})(CargobobHook || {});
var LicensePlateStyle = /* @__PURE__ */ ((LicensePlateStyle2) => {
  LicensePlateStyle2[LicensePlateStyle2["BlueOnWhite1"] = 3] = "BlueOnWhite1";
  LicensePlateStyle2[LicensePlateStyle2["BlueOnWhite2"] = 0] = "BlueOnWhite2";
  LicensePlateStyle2[LicensePlateStyle2["BlueOnWhite3"] = 4] = "BlueOnWhite3";
  LicensePlateStyle2[LicensePlateStyle2["YellowOnBlack"] = 1] = "YellowOnBlack";
  LicensePlateStyle2[LicensePlateStyle2["YellowOnBlue"] = 2] = "YellowOnBlue";
  LicensePlateStyle2[LicensePlateStyle2["NorthYankton"] = 5] = "NorthYankton";
  return LicensePlateStyle2;
})(LicensePlateStyle || {});
var LicensePlateType = /* @__PURE__ */ ((LicensePlateType2) => {
  LicensePlateType2[LicensePlateType2["FrontAndRearPlates"] = 0] = "FrontAndRearPlates";
  LicensePlateType2[LicensePlateType2["FrontPlate"] = 1] = "FrontPlate";
  LicensePlateType2[LicensePlateType2["RearPlate"] = 2] = "RearPlate";
  LicensePlateType2[LicensePlateType2["None"] = 3] = "None";
  return LicensePlateType2;
})(LicensePlateType || {});
var VehicleClass = /* @__PURE__ */ ((VehicleClass2) => {
  VehicleClass2[VehicleClass2["Compacts"] = 0] = "Compacts";
  VehicleClass2[VehicleClass2["Sedans"] = 1] = "Sedans";
  VehicleClass2[VehicleClass2["SUVs"] = 2] = "SUVs";
  VehicleClass2[VehicleClass2["Coupes"] = 3] = "Coupes";
  VehicleClass2[VehicleClass2["Muscle"] = 4] = "Muscle";
  VehicleClass2[VehicleClass2["SportsClassics"] = 5] = "SportsClassics";
  VehicleClass2[VehicleClass2["Sports"] = 6] = "Sports";
  VehicleClass2[VehicleClass2["Super"] = 7] = "Super";
  VehicleClass2[VehicleClass2["Motorcycles"] = 8] = "Motorcycles";
  VehicleClass2[VehicleClass2["OffRoad"] = 9] = "OffRoad";
  VehicleClass2[VehicleClass2["Industrial"] = 10] = "Industrial";
  VehicleClass2[VehicleClass2["Utility"] = 11] = "Utility";
  VehicleClass2[VehicleClass2["Vans"] = 12] = "Vans";
  VehicleClass2[VehicleClass2["Cycles"] = 13] = "Cycles";
  VehicleClass2[VehicleClass2["Boats"] = 14] = "Boats";
  VehicleClass2[VehicleClass2["Helicopters"] = 15] = "Helicopters";
  VehicleClass2[VehicleClass2["Planes"] = 16] = "Planes";
  VehicleClass2[VehicleClass2["Service"] = 17] = "Service";
  VehicleClass2[VehicleClass2["Emergency"] = 18] = "Emergency";
  VehicleClass2[VehicleClass2["Military"] = 19] = "Military";
  VehicleClass2[VehicleClass2["Commercial"] = 20] = "Commercial";
  VehicleClass2[VehicleClass2["Trains"] = 21] = "Trains";
  return VehicleClass2;
})(VehicleClass || {});
var VehicleColor = /* @__PURE__ */ ((VehicleColor2) => {
  VehicleColor2[VehicleColor2["MetallicBlack"] = 0] = "MetallicBlack";
  VehicleColor2[VehicleColor2["MetallicGraphiteBlack"] = 1] = "MetallicGraphiteBlack";
  VehicleColor2[VehicleColor2["MetallicBlackSteel"] = 2] = "MetallicBlackSteel";
  VehicleColor2[VehicleColor2["MetallicDarkSilver"] = 3] = "MetallicDarkSilver";
  VehicleColor2[VehicleColor2["MetallicSilver"] = 4] = "MetallicSilver";
  VehicleColor2[VehicleColor2["MetallicBlueSilver"] = 5] = "MetallicBlueSilver";
  VehicleColor2[VehicleColor2["MetallicSteelGray"] = 6] = "MetallicSteelGray";
  VehicleColor2[VehicleColor2["MetallicShadowSilver"] = 7] = "MetallicShadowSilver";
  VehicleColor2[VehicleColor2["MetallicStoneSilver"] = 8] = "MetallicStoneSilver";
  VehicleColor2[VehicleColor2["MetallicMidnightSilver"] = 9] = "MetallicMidnightSilver";
  VehicleColor2[VehicleColor2["MetallicGunMetal"] = 10] = "MetallicGunMetal";
  VehicleColor2[VehicleColor2["MetallicAnthraciteGray"] = 11] = "MetallicAnthraciteGray";
  VehicleColor2[VehicleColor2["MatteBlack"] = 12] = "MatteBlack";
  VehicleColor2[VehicleColor2["MatteGray"] = 13] = "MatteGray";
  VehicleColor2[VehicleColor2["MatteLightGray"] = 14] = "MatteLightGray";
  VehicleColor2[VehicleColor2["UtilBlack"] = 15] = "UtilBlack";
  VehicleColor2[VehicleColor2["UtilBlackPoly"] = 16] = "UtilBlackPoly";
  VehicleColor2[VehicleColor2["UtilDarksilver"] = 17] = "UtilDarksilver";
  VehicleColor2[VehicleColor2["UtilSilver"] = 18] = "UtilSilver";
  VehicleColor2[VehicleColor2["UtilGunMetal"] = 19] = "UtilGunMetal";
  VehicleColor2[VehicleColor2["UtilShadowSilver"] = 20] = "UtilShadowSilver";
  VehicleColor2[VehicleColor2["WornBlack"] = 21] = "WornBlack";
  VehicleColor2[VehicleColor2["WornGraphite"] = 22] = "WornGraphite";
  VehicleColor2[VehicleColor2["WornSilverGray"] = 23] = "WornSilverGray";
  VehicleColor2[VehicleColor2["WornSilver"] = 24] = "WornSilver";
  VehicleColor2[VehicleColor2["WornBlueSilver"] = 25] = "WornBlueSilver";
  VehicleColor2[VehicleColor2["WornShadowSilver"] = 26] = "WornShadowSilver";
  VehicleColor2[VehicleColor2["MetallicRed"] = 27] = "MetallicRed";
  VehicleColor2[VehicleColor2["MetallicTorinoRed"] = 28] = "MetallicTorinoRed";
  VehicleColor2[VehicleColor2["MetallicFormulaRed"] = 29] = "MetallicFormulaRed";
  VehicleColor2[VehicleColor2["MetallicBlazeRed"] = 30] = "MetallicBlazeRed";
  VehicleColor2[VehicleColor2["MetallicGracefulRed"] = 31] = "MetallicGracefulRed";
  VehicleColor2[VehicleColor2["MetallicGarnetRed"] = 32] = "MetallicGarnetRed";
  VehicleColor2[VehicleColor2["MetallicDesertRed"] = 33] = "MetallicDesertRed";
  VehicleColor2[VehicleColor2["MetallicCabernetRed"] = 34] = "MetallicCabernetRed";
  VehicleColor2[VehicleColor2["MetallicCandyRed"] = 35] = "MetallicCandyRed";
  VehicleColor2[VehicleColor2["MetallicSunriseOrange"] = 36] = "MetallicSunriseOrange";
  VehicleColor2[VehicleColor2["MetallicClassicGold"] = 37] = "MetallicClassicGold";
  VehicleColor2[VehicleColor2["MetallicOrange"] = 38] = "MetallicOrange";
  VehicleColor2[VehicleColor2["MatteRed"] = 39] = "MatteRed";
  VehicleColor2[VehicleColor2["MatteDarkRed"] = 40] = "MatteDarkRed";
  VehicleColor2[VehicleColor2["MatteOrange"] = 41] = "MatteOrange";
  VehicleColor2[VehicleColor2["MatteYellow"] = 42] = "MatteYellow";
  VehicleColor2[VehicleColor2["UtilRed"] = 43] = "UtilRed";
  VehicleColor2[VehicleColor2["UtilBrightRed"] = 44] = "UtilBrightRed";
  VehicleColor2[VehicleColor2["UtilGarnetRed"] = 45] = "UtilGarnetRed";
  VehicleColor2[VehicleColor2["WornRed"] = 46] = "WornRed";
  VehicleColor2[VehicleColor2["WornGoldenRed"] = 47] = "WornGoldenRed";
  VehicleColor2[VehicleColor2["WornDarkRed"] = 48] = "WornDarkRed";
  VehicleColor2[VehicleColor2["MetallicDarkGreen"] = 49] = "MetallicDarkGreen";
  VehicleColor2[VehicleColor2["MetallicRacingGreen"] = 50] = "MetallicRacingGreen";
  VehicleColor2[VehicleColor2["MetallicSeaGreen"] = 51] = "MetallicSeaGreen";
  VehicleColor2[VehicleColor2["MetallicOliveGreen"] = 52] = "MetallicOliveGreen";
  VehicleColor2[VehicleColor2["MetallicGreen"] = 53] = "MetallicGreen";
  VehicleColor2[VehicleColor2["MetallicGasolineBlueGreen"] = 54] = "MetallicGasolineBlueGreen";
  VehicleColor2[VehicleColor2["MatteLimeGreen"] = 55] = "MatteLimeGreen";
  VehicleColor2[VehicleColor2["UtilDarkGreen"] = 56] = "UtilDarkGreen";
  VehicleColor2[VehicleColor2["UtilGreen"] = 57] = "UtilGreen";
  VehicleColor2[VehicleColor2["WornDarkGreen"] = 58] = "WornDarkGreen";
  VehicleColor2[VehicleColor2["WornGreen"] = 59] = "WornGreen";
  VehicleColor2[VehicleColor2["WornSeaWash"] = 60] = "WornSeaWash";
  VehicleColor2[VehicleColor2["MetallicMidnightBlue"] = 61] = "MetallicMidnightBlue";
  VehicleColor2[VehicleColor2["MetallicDarkBlue"] = 62] = "MetallicDarkBlue";
  VehicleColor2[VehicleColor2["MetallicSaxonyBlue"] = 63] = "MetallicSaxonyBlue";
  VehicleColor2[VehicleColor2["MetallicBlue"] = 64] = "MetallicBlue";
  VehicleColor2[VehicleColor2["MetallicMarinerBlue"] = 65] = "MetallicMarinerBlue";
  VehicleColor2[VehicleColor2["MetallicHarborBlue"] = 66] = "MetallicHarborBlue";
  VehicleColor2[VehicleColor2["MetallicDiamondBlue"] = 67] = "MetallicDiamondBlue";
  VehicleColor2[VehicleColor2["MetallicSurfBlue"] = 68] = "MetallicSurfBlue";
  VehicleColor2[VehicleColor2["MetallicNauticalBlue"] = 69] = "MetallicNauticalBlue";
  VehicleColor2[VehicleColor2["MetallicBrightBlue"] = 70] = "MetallicBrightBlue";
  VehicleColor2[VehicleColor2["MetallicPurpleBlue"] = 71] = "MetallicPurpleBlue";
  VehicleColor2[VehicleColor2["MetallicSpinnakerBlue"] = 72] = "MetallicSpinnakerBlue";
  VehicleColor2[VehicleColor2["MetallicUltraBlue"] = 73] = "MetallicUltraBlue";
  VehicleColor2[VehicleColor2["UtilDarkBlue"] = 75] = "UtilDarkBlue";
  VehicleColor2[VehicleColor2["UtilMidnightBlue"] = 76] = "UtilMidnightBlue";
  VehicleColor2[VehicleColor2["UtilBlue"] = 77] = "UtilBlue";
  VehicleColor2[VehicleColor2["UtilSeaFoamBlue"] = 78] = "UtilSeaFoamBlue";
  VehicleColor2[VehicleColor2["UtilLightningBlue"] = 79] = "UtilLightningBlue";
  VehicleColor2[VehicleColor2["UtilMauiBluePoly"] = 80] = "UtilMauiBluePoly";
  VehicleColor2[VehicleColor2["UtilBrightBlue"] = 81] = "UtilBrightBlue";
  VehicleColor2[VehicleColor2["MatteDarkBlue"] = 82] = "MatteDarkBlue";
  VehicleColor2[VehicleColor2["MatteBlue"] = 83] = "MatteBlue";
  VehicleColor2[VehicleColor2["MatteMidnightBlue"] = 84] = "MatteMidnightBlue";
  VehicleColor2[VehicleColor2["WornDarkBlue"] = 85] = "WornDarkBlue";
  VehicleColor2[VehicleColor2["WornBlue"] = 86] = "WornBlue";
  VehicleColor2[VehicleColor2["WornLightBlue"] = 87] = "WornLightBlue";
  VehicleColor2[VehicleColor2["MetallicTaxiYellow"] = 88] = "MetallicTaxiYellow";
  VehicleColor2[VehicleColor2["MetallicRaceYellow"] = 89] = "MetallicRaceYellow";
  VehicleColor2[VehicleColor2["MetallicBronze"] = 90] = "MetallicBronze";
  VehicleColor2[VehicleColor2["MetallicYellowBird"] = 91] = "MetallicYellowBird";
  VehicleColor2[VehicleColor2["MetallicLime"] = 92] = "MetallicLime";
  VehicleColor2[VehicleColor2["MetallicChampagne"] = 93] = "MetallicChampagne";
  VehicleColor2[VehicleColor2["MetallicPuebloBeige"] = 94] = "MetallicPuebloBeige";
  VehicleColor2[VehicleColor2["MetallicDarkIvory"] = 95] = "MetallicDarkIvory";
  VehicleColor2[VehicleColor2["MetallicChocoBrown"] = 96] = "MetallicChocoBrown";
  VehicleColor2[VehicleColor2["MetallicGoldenBrown"] = 97] = "MetallicGoldenBrown";
  VehicleColor2[VehicleColor2["MetallicLightBrown"] = 98] = "MetallicLightBrown";
  VehicleColor2[VehicleColor2["MetallicStrawBeige"] = 99] = "MetallicStrawBeige";
  VehicleColor2[VehicleColor2["MetallicMossBrown"] = 100] = "MetallicMossBrown";
  VehicleColor2[VehicleColor2["MetallicBistonBrown"] = 101] = "MetallicBistonBrown";
  VehicleColor2[VehicleColor2["MetallicBeechwood"] = 102] = "MetallicBeechwood";
  VehicleColor2[VehicleColor2["MetallicDarkBeechwood"] = 103] = "MetallicDarkBeechwood";
  VehicleColor2[VehicleColor2["MetallicChocoOrange"] = 104] = "MetallicChocoOrange";
  VehicleColor2[VehicleColor2["MetallicBeachSand"] = 105] = "MetallicBeachSand";
  VehicleColor2[VehicleColor2["MetallicSunBleechedSand"] = 106] = "MetallicSunBleechedSand";
  VehicleColor2[VehicleColor2["MetallicCream"] = 107] = "MetallicCream";
  VehicleColor2[VehicleColor2["UtilBrown"] = 108] = "UtilBrown";
  VehicleColor2[VehicleColor2["UtilMediumBrown"] = 109] = "UtilMediumBrown";
  VehicleColor2[VehicleColor2["UtilLightBrown"] = 110] = "UtilLightBrown";
  VehicleColor2[VehicleColor2["MetallicWhite"] = 111] = "MetallicWhite";
  VehicleColor2[VehicleColor2["MetallicFrostWhite"] = 112] = "MetallicFrostWhite";
  VehicleColor2[VehicleColor2["WornHoneyBeige"] = 113] = "WornHoneyBeige";
  VehicleColor2[VehicleColor2["WornBrown"] = 114] = "WornBrown";
  VehicleColor2[VehicleColor2["WornDarkBrown"] = 115] = "WornDarkBrown";
  VehicleColor2[VehicleColor2["WornStrawBeige"] = 116] = "WornStrawBeige";
  VehicleColor2[VehicleColor2["BrushedSteel"] = 117] = "BrushedSteel";
  VehicleColor2[VehicleColor2["BrushedBlackSteel"] = 118] = "BrushedBlackSteel";
  VehicleColor2[VehicleColor2["BrushedAluminium"] = 119] = "BrushedAluminium";
  VehicleColor2[VehicleColor2["Chrome"] = 120] = "Chrome";
  VehicleColor2[VehicleColor2["WornOffWhite"] = 121] = "WornOffWhite";
  VehicleColor2[VehicleColor2["UtilOffWhite"] = 122] = "UtilOffWhite";
  VehicleColor2[VehicleColor2["WornOrange"] = 123] = "WornOrange";
  VehicleColor2[VehicleColor2["WornLightOrange"] = 124] = "WornLightOrange";
  VehicleColor2[VehicleColor2["MetallicSecuricorGreen"] = 125] = "MetallicSecuricorGreen";
  VehicleColor2[VehicleColor2["WornTaxiYellow"] = 126] = "WornTaxiYellow";
  VehicleColor2[VehicleColor2["PoliceCarBlue"] = 127] = "PoliceCarBlue";
  VehicleColor2[VehicleColor2["MatteGreen"] = 128] = "MatteGreen";
  VehicleColor2[VehicleColor2["MatteBrown"] = 129] = "MatteBrown";
  VehicleColor2[VehicleColor2["MatteWhite"] = 131] = "MatteWhite";
  VehicleColor2[VehicleColor2["WornWhite"] = 132] = "WornWhite";
  VehicleColor2[VehicleColor2["WornOliveArmyGreen"] = 133] = "WornOliveArmyGreen";
  VehicleColor2[VehicleColor2["PureWhite"] = 134] = "PureWhite";
  VehicleColor2[VehicleColor2["HotPink"] = 135] = "HotPink";
  VehicleColor2[VehicleColor2["Salmonpink"] = 136] = "Salmonpink";
  VehicleColor2[VehicleColor2["MetallicVermillionPink"] = 137] = "MetallicVermillionPink";
  VehicleColor2[VehicleColor2["Orange"] = 138] = "Orange";
  VehicleColor2[VehicleColor2["Green"] = 139] = "Green";
  VehicleColor2[VehicleColor2["Blue"] = 140] = "Blue";
  VehicleColor2[VehicleColor2["MettalicBlackBlue"] = 141] = "MettalicBlackBlue";
  VehicleColor2[VehicleColor2["MetallicBlackPurple"] = 142] = "MetallicBlackPurple";
  VehicleColor2[VehicleColor2["MetallicBlackRed"] = 143] = "MetallicBlackRed";
  VehicleColor2[VehicleColor2["HunterGreen"] = 144] = "HunterGreen";
  VehicleColor2[VehicleColor2["MetallicPurple"] = 145] = "MetallicPurple";
  VehicleColor2[VehicleColor2["MetaillicVDarkBlue"] = 146] = "MetaillicVDarkBlue";
  VehicleColor2[VehicleColor2["ModshopBlack1"] = 147] = "ModshopBlack1";
  VehicleColor2[VehicleColor2["MattePurple"] = 148] = "MattePurple";
  VehicleColor2[VehicleColor2["MatteDarkPurple"] = 149] = "MatteDarkPurple";
  VehicleColor2[VehicleColor2["MetallicLavaRed"] = 150] = "MetallicLavaRed";
  VehicleColor2[VehicleColor2["MatteForestGreen"] = 151] = "MatteForestGreen";
  VehicleColor2[VehicleColor2["MatteOliveDrab"] = 152] = "MatteOliveDrab";
  VehicleColor2[VehicleColor2["MatteDesertBrown"] = 153] = "MatteDesertBrown";
  VehicleColor2[VehicleColor2["MatteDesertTan"] = 154] = "MatteDesertTan";
  VehicleColor2[VehicleColor2["MatteFoliageGreen"] = 155] = "MatteFoliageGreen";
  VehicleColor2[VehicleColor2["DefaultAlloyColor"] = 156] = "DefaultAlloyColor";
  VehicleColor2[VehicleColor2["EpsilonBlue"] = 157] = "EpsilonBlue";
  VehicleColor2[VehicleColor2["PureGold"] = 158] = "PureGold";
  VehicleColor2[VehicleColor2["BrushedGold"] = 159] = "BrushedGold";
  return VehicleColor2;
})(VehicleColor || {});
var VehicleLandingGearState = /* @__PURE__ */ ((VehicleLandingGearState2) => {
  VehicleLandingGearState2[VehicleLandingGearState2["Deployed"] = 0] = "Deployed";
  VehicleLandingGearState2[VehicleLandingGearState2["Closing"] = 1] = "Closing";
  VehicleLandingGearState2[VehicleLandingGearState2["Opening"] = 2] = "Opening";
  VehicleLandingGearState2[VehicleLandingGearState2["Retracted"] = 3] = "Retracted";
  return VehicleLandingGearState2;
})(VehicleLandingGearState || {});
var VehicleLockStatus = /* @__PURE__ */ ((VehicleLockStatus2) => {
  VehicleLockStatus2[VehicleLockStatus2["None"] = 0] = "None";
  VehicleLockStatus2[VehicleLockStatus2["Unlocked"] = 1] = "Unlocked";
  VehicleLockStatus2[VehicleLockStatus2["Locked"] = 2] = "Locked";
  VehicleLockStatus2[VehicleLockStatus2["LockedForPlayer"] = 3] = "LockedForPlayer";
  VehicleLockStatus2[VehicleLockStatus2["StickPlayerInside"] = 4] = "StickPlayerInside";
  VehicleLockStatus2[VehicleLockStatus2["CanBeBrokenInto"] = 7] = "CanBeBrokenInto";
  VehicleLockStatus2[VehicleLockStatus2["CanBeBrokenIntoPersist"] = 8] = "CanBeBrokenIntoPersist";
  VehicleLockStatus2[VehicleLockStatus2["CannotBeTriedToEnter"] = 10] = "CannotBeTriedToEnter";
  return VehicleLockStatus2;
})(VehicleLockStatus || {});
var VehicleNeonLight = /* @__PURE__ */ ((VehicleNeonLight2) => {
  VehicleNeonLight2[VehicleNeonLight2["Left"] = 0] = "Left";
  VehicleNeonLight2[VehicleNeonLight2["Right"] = 1] = "Right";
  VehicleNeonLight2[VehicleNeonLight2["Front"] = 2] = "Front";
  VehicleNeonLight2[VehicleNeonLight2["Back"] = 3] = "Back";
  return VehicleNeonLight2;
})(VehicleNeonLight || {});
var VehicleRoofState = /* @__PURE__ */ ((VehicleRoofState2) => {
  VehicleRoofState2[VehicleRoofState2["Closed"] = 0] = "Closed";
  VehicleRoofState2[VehicleRoofState2["Opening"] = 1] = "Opening";
  VehicleRoofState2[VehicleRoofState2["Opened"] = 2] = "Opened";
  VehicleRoofState2[VehicleRoofState2["Closing"] = 3] = "Closing";
  return VehicleRoofState2;
})(VehicleRoofState || {});
var VehicleSeat = /* @__PURE__ */ ((VehicleSeat2) => {
  VehicleSeat2[VehicleSeat2["None"] = -3] = "None";
  VehicleSeat2[VehicleSeat2["Any"] = -2] = "Any";
  VehicleSeat2[VehicleSeat2["Driver"] = -1] = "Driver";
  VehicleSeat2[VehicleSeat2["Passenger"] = 0] = "Passenger";
  VehicleSeat2[VehicleSeat2["LeftFront"] = -1] = "LeftFront";
  VehicleSeat2[VehicleSeat2["RightFront"] = 0] = "RightFront";
  VehicleSeat2[VehicleSeat2["LeftRear"] = 1] = "LeftRear";
  VehicleSeat2[VehicleSeat2["RightRear"] = 2] = "RightRear";
  VehicleSeat2[VehicleSeat2["ExtraSeat1"] = 3] = "ExtraSeat1";
  VehicleSeat2[VehicleSeat2["ExtraSeat2"] = 4] = "ExtraSeat2";
  VehicleSeat2[VehicleSeat2["ExtraSeat3"] = 5] = "ExtraSeat3";
  VehicleSeat2[VehicleSeat2["ExtraSeat4"] = 6] = "ExtraSeat4";
  VehicleSeat2[VehicleSeat2["ExtraSeat5"] = 7] = "ExtraSeat5";
  VehicleSeat2[VehicleSeat2["ExtraSeat6"] = 8] = "ExtraSeat6";
  VehicleSeat2[VehicleSeat2["ExtraSeat7"] = 9] = "ExtraSeat7";
  VehicleSeat2[VehicleSeat2["ExtraSeat8"] = 10] = "ExtraSeat8";
  VehicleSeat2[VehicleSeat2["ExtraSeat9"] = 11] = "ExtraSeat9";
  VehicleSeat2[VehicleSeat2["ExtraSeat10"] = 12] = "ExtraSeat10";
  VehicleSeat2[VehicleSeat2["ExtraSeat11"] = 13] = "ExtraSeat11";
  VehicleSeat2[VehicleSeat2["ExtraSeat12"] = 14] = "ExtraSeat12";
  return VehicleSeat2;
})(VehicleSeat || {});
var VehicleWindowTint = /* @__PURE__ */ ((VehicleWindowTint2) => {
  VehicleWindowTint2[VehicleWindowTint2["None"] = 0] = "None";
  VehicleWindowTint2[VehicleWindowTint2["PureBlack"] = 1] = "PureBlack";
  VehicleWindowTint2[VehicleWindowTint2["DarkSmoke"] = 2] = "DarkSmoke";
  VehicleWindowTint2[VehicleWindowTint2["LightSmoke"] = 3] = "LightSmoke";
  VehicleWindowTint2[VehicleWindowTint2["Stock"] = 4] = "Stock";
  VehicleWindowTint2[VehicleWindowTint2["Limo"] = 5] = "Limo";
  VehicleWindowTint2[VehicleWindowTint2["Green"] = 6] = "Green";
  return VehicleWindowTint2;
})(VehicleWindowTint || {});
var VehicleWindowIndex = /* @__PURE__ */ ((VehicleWindowIndex2) => {
  VehicleWindowIndex2[VehicleWindowIndex2["FrontRightWindow"] = 1] = "FrontRightWindow";
  VehicleWindowIndex2[VehicleWindowIndex2["FrontLeftWindow"] = 0] = "FrontLeftWindow";
  VehicleWindowIndex2[VehicleWindowIndex2["BackRightWindow"] = 3] = "BackRightWindow";
  VehicleWindowIndex2[VehicleWindowIndex2["BackLeftWindow"] = 2] = "BackLeftWindow";
  VehicleWindowIndex2[VehicleWindowIndex2["ExtraWindow1"] = 4] = "ExtraWindow1";
  VehicleWindowIndex2[VehicleWindowIndex2["ExtraWindow2"] = 5] = "ExtraWindow2";
  VehicleWindowIndex2[VehicleWindowIndex2["ExtraWindow3"] = 6] = "ExtraWindow3";
  VehicleWindowIndex2[VehicleWindowIndex2["ExtraWindow4"] = 7] = "ExtraWindow4";
  return VehicleWindowIndex2;
})(VehicleWindowIndex || {});
var VehicleModType = /* @__PURE__ */ ((VehicleModType2) => {
  VehicleModType2[VehicleModType2["Spoilers"] = 0] = "Spoilers";
  VehicleModType2[VehicleModType2["FrontBumper"] = 1] = "FrontBumper";
  VehicleModType2[VehicleModType2["RearBumper"] = 2] = "RearBumper";
  VehicleModType2[VehicleModType2["SideSkirt"] = 3] = "SideSkirt";
  VehicleModType2[VehicleModType2["Exhaust"] = 4] = "Exhaust";
  VehicleModType2[VehicleModType2["Frame"] = 5] = "Frame";
  VehicleModType2[VehicleModType2["Grille"] = 6] = "Grille";
  VehicleModType2[VehicleModType2["Hood"] = 7] = "Hood";
  VehicleModType2[VehicleModType2["Fender"] = 8] = "Fender";
  VehicleModType2[VehicleModType2["RightFender"] = 9] = "RightFender";
  VehicleModType2[VehicleModType2["Roof"] = 10] = "Roof";
  VehicleModType2[VehicleModType2["Engine"] = 11] = "Engine";
  VehicleModType2[VehicleModType2["Brakes"] = 12] = "Brakes";
  VehicleModType2[VehicleModType2["Transmission"] = 13] = "Transmission";
  VehicleModType2[VehicleModType2["Horns"] = 14] = "Horns";
  VehicleModType2[VehicleModType2["Suspension"] = 15] = "Suspension";
  VehicleModType2[VehicleModType2["Armor"] = 16] = "Armor";
  VehicleModType2[VehicleModType2["FrontWheel"] = 23] = "FrontWheel";
  VehicleModType2[VehicleModType2["RearWheel"] = 24] = "RearWheel";
  VehicleModType2[VehicleModType2["PlateHolder"] = 25] = "PlateHolder";
  VehicleModType2[VehicleModType2["VanityPlates"] = 26] = "VanityPlates";
  VehicleModType2[VehicleModType2["TrimDesign"] = 27] = "TrimDesign";
  VehicleModType2[VehicleModType2["Ornaments"] = 28] = "Ornaments";
  VehicleModType2[VehicleModType2["Dashboard"] = 29] = "Dashboard";
  VehicleModType2[VehicleModType2["DialDesign"] = 30] = "DialDesign";
  VehicleModType2[VehicleModType2["DoorSpeakers"] = 31] = "DoorSpeakers";
  VehicleModType2[VehicleModType2["Seats"] = 32] = "Seats";
  VehicleModType2[VehicleModType2["SteeringWheels"] = 33] = "SteeringWheels";
  VehicleModType2[VehicleModType2["ColumnShifterLevers"] = 34] = "ColumnShifterLevers";
  VehicleModType2[VehicleModType2["Plaques"] = 35] = "Plaques";
  VehicleModType2[VehicleModType2["Speakers"] = 36] = "Speakers";
  VehicleModType2[VehicleModType2["Trunk"] = 37] = "Trunk";
  VehicleModType2[VehicleModType2["Hydraulics"] = 38] = "Hydraulics";
  VehicleModType2[VehicleModType2["EngineBlock"] = 39] = "EngineBlock";
  VehicleModType2[VehicleModType2["AirFilter"] = 40] = "AirFilter";
  VehicleModType2[VehicleModType2["Struts"] = 41] = "Struts";
  VehicleModType2[VehicleModType2["ArchCover"] = 42] = "ArchCover";
  VehicleModType2[VehicleModType2["Aerials"] = 43] = "Aerials";
  VehicleModType2[VehicleModType2["Trim"] = 44] = "Trim";
  VehicleModType2[VehicleModType2["Tank"] = 45] = "Tank";
  VehicleModType2[VehicleModType2["Windows"] = 46] = "Windows";
  VehicleModType2[VehicleModType2["Livery"] = 48] = "Livery";
  return VehicleModType2;
})(VehicleModType || {});
var VehicleToggleModType = /* @__PURE__ */ ((VehicleToggleModType2) => {
  VehicleToggleModType2[VehicleToggleModType2["Turbo"] = 18] = "Turbo";
  VehicleToggleModType2[VehicleToggleModType2["TireSmoke"] = 20] = "TireSmoke";
  VehicleToggleModType2[VehicleToggleModType2["XenonHeadlights"] = 22] = "XenonHeadlights";
  return VehicleToggleModType2;
})(VehicleToggleModType || {});
var VehiclePaintType = /* @__PURE__ */ ((VehiclePaintType2) => {
  VehiclePaintType2[VehiclePaintType2["Normal"] = 0] = "Normal";
  VehiclePaintType2[VehiclePaintType2["Metallic"] = 1] = "Metallic";
  VehiclePaintType2[VehiclePaintType2["Pearl"] = 2] = "Pearl";
  VehiclePaintType2[VehiclePaintType2["Matte"] = 3] = "Matte";
  VehiclePaintType2[VehiclePaintType2["Metal"] = 4] = "Metal";
  VehiclePaintType2[VehiclePaintType2["Chrome"] = 5] = "Chrome";
  return VehiclePaintType2;
})(VehiclePaintType || {});
var VehicleDoorIndex = /* @__PURE__ */ ((VehicleDoorIndex2) => {
  VehicleDoorIndex2[VehicleDoorIndex2["FrontRightDoor"] = 1] = "FrontRightDoor";
  VehicleDoorIndex2[VehicleDoorIndex2["FrontLeftDoor"] = 0] = "FrontLeftDoor";
  VehicleDoorIndex2[VehicleDoorIndex2["BackRightDoor"] = 3] = "BackRightDoor";
  VehicleDoorIndex2[VehicleDoorIndex2["BackLeftDoor"] = 2] = "BackLeftDoor";
  VehicleDoorIndex2[VehicleDoorIndex2["Hood"] = 4] = "Hood";
  VehicleDoorIndex2[VehicleDoorIndex2["Trunk"] = 5] = "Trunk";
  return VehicleDoorIndex2;
})(VehicleDoorIndex || {});
var VehicleWheelType = /* @__PURE__ */ ((VehicleWheelType2) => {
  VehicleWheelType2[VehicleWheelType2["Sport"] = 0] = "Sport";
  VehicleWheelType2[VehicleWheelType2["Muscle"] = 1] = "Muscle";
  VehicleWheelType2[VehicleWheelType2["Lowrider"] = 2] = "Lowrider";
  VehicleWheelType2[VehicleWheelType2["SUV"] = 3] = "SUV";
  VehicleWheelType2[VehicleWheelType2["Offroad"] = 4] = "Offroad";
  VehicleWheelType2[VehicleWheelType2["Tuner"] = 5] = "Tuner";
  VehicleWheelType2[VehicleWheelType2["BikeWheels"] = 6] = "BikeWheels";
  VehicleWheelType2[VehicleWheelType2["HighEnd"] = 7] = "HighEnd";
  VehicleWheelType2[VehicleWheelType2["BennysOriginals"] = 8] = "BennysOriginals";
  VehicleWheelType2[VehicleWheelType2["BennysBespoke"] = 9] = "BennysBespoke";
  return VehicleWheelType2;
})(VehicleWheelType || {});
var VehicleWheelIndex = /* @__PURE__ */ ((VehicleWheelIndex2) => {
  VehicleWheelIndex2[VehicleWheelIndex2["FrontLeftWheel"] = 0] = "FrontLeftWheel";
  VehicleWheelIndex2[VehicleWheelIndex2["FrontRightWheel"] = 1] = "FrontRightWheel";
  VehicleWheelIndex2[VehicleWheelIndex2["MidLeftWheel"] = 2] = "MidLeftWheel";
  VehicleWheelIndex2[VehicleWheelIndex2["MidRightWheel"] = 3] = "MidRightWheel";
  VehicleWheelIndex2[VehicleWheelIndex2["RearLeftWheel"] = 4] = "RearLeftWheel";
  VehicleWheelIndex2[VehicleWheelIndex2["RearRightWheel"] = 5] = "RearRightWheel";
  VehicleWheelIndex2[VehicleWheelIndex2["TrailerMidLeftWheel"] = 45] = "TrailerMidLeftWheel";
  VehicleWheelIndex2[VehicleWheelIndex2["TrailerMidRightWheel"] = 47] = "TrailerMidRightWheel";
  return VehicleWheelIndex2;
})(VehicleWheelIndex || {});
export {
  CargobobHook,
  LicensePlateStyle,
  LicensePlateType,
  VehicleClass,
  VehicleColor,
  VehicleDoorIndex,
  VehicleLandingGearState,
  VehicleLockStatus,
  VehicleModType,
  VehicleNeonLight,
  VehiclePaintType,
  VehicleRoofState,
  VehicleSeat,
  VehicleToggleModType,
  VehicleWheelIndex,
  VehicleWheelType,
  VehicleWindowIndex,
  VehicleWindowTint
};
