// @nativewrappers/fivem/enums/ZoneID.js
var ZoneID = /* @__PURE__ */ ((ZoneID2) => {
  ZoneID2[ZoneID2["AIRP"] = 0] = "AIRP";
  ZoneID2[ZoneID2["ALAMO"] = 1] = "ALAMO";
  ZoneID2[ZoneID2["ALTA"] = 2] = "ALTA";
  ZoneID2[ZoneID2["ARMYB"] = 3] = "ARMYB";
  ZoneID2[ZoneID2["BANHAMC"] = 4] = "BANHAMC";
  ZoneID2[ZoneID2["BANNING"] = 5] = "BANNING";
  ZoneID2[ZoneID2["BEACH"] = 6] = "BEACH";
  ZoneID2[ZoneID2["BHAMCA"] = 7] = "BHAMCA";
  ZoneID2[ZoneID2["BRADP"] = 8] = "BRADP";
  ZoneID2[ZoneID2["BRADT"] = 9] = "BRADT";
  ZoneID2[ZoneID2["BURTON"] = 10] = "BURTON";
  ZoneID2[ZoneID2["CALAFB"] = 11] = "CALAFB";
  ZoneID2[ZoneID2["CANNY"] = 12] = "CANNY";
  ZoneID2[ZoneID2["CCREAK"] = 13] = "CCREAK";
  ZoneID2[ZoneID2["CHAMH"] = 14] = "CHAMH";
  ZoneID2[ZoneID2["CHIL"] = 15] = "CHIL";
  ZoneID2[ZoneID2["CHU"] = 16] = "CHU";
  ZoneID2[ZoneID2["CMSW"] = 17] = "CMSW";
  ZoneID2[ZoneID2["CYPRE"] = 18] = "CYPRE";
  ZoneID2[ZoneID2["DAVIS"] = 19] = "DAVIS";
  ZoneID2[ZoneID2["DELBE"] = 20] = "DELBE";
  ZoneID2[ZoneID2["DELPE"] = 21] = "DELPE";
  ZoneID2[ZoneID2["DELSOL"] = 22] = "DELSOL";
  ZoneID2[ZoneID2["DESRT"] = 23] = "DESRT";
  ZoneID2[ZoneID2["DOWNT"] = 24] = "DOWNT";
  ZoneID2[ZoneID2["DTVINE"] = 25] = "DTVINE";
  ZoneID2[ZoneID2["EAST_V"] = 26] = "EAST_V";
  ZoneID2[ZoneID2["EBURO"] = 27] = "EBURO";
  ZoneID2[ZoneID2["ELGORL"] = 28] = "ELGORL";
  ZoneID2[ZoneID2["ELYSIAN"] = 29] = "ELYSIAN";
  ZoneID2[ZoneID2["GALFISH"] = 30] = "GALFISH";
  ZoneID2[ZoneID2["golf"] = 31] = "golf";
  ZoneID2[ZoneID2["GRAPES"] = 32] = "GRAPES";
  ZoneID2[ZoneID2["GREATC"] = 33] = "GREATC";
  ZoneID2[ZoneID2["HARMO"] = 34] = "HARMO";
  ZoneID2[ZoneID2["HAWICK"] = 35] = "HAWICK";
  ZoneID2[ZoneID2["HORS"] = 36] = "HORS";
  ZoneID2[ZoneID2["HUMLAB"] = 37] = "HUMLAB";
  ZoneID2[ZoneID2["JAIL"] = 38] = "JAIL";
  ZoneID2[ZoneID2["KOREAT"] = 39] = "KOREAT";
  ZoneID2[ZoneID2["LACT"] = 40] = "LACT";
  ZoneID2[ZoneID2["LAGO"] = 41] = "LAGO";
  ZoneID2[ZoneID2["LDAM"] = 42] = "LDAM";
  ZoneID2[ZoneID2["LEGSQU"] = 43] = "LEGSQU";
  ZoneID2[ZoneID2["LMESA"] = 44] = "LMESA";
  ZoneID2[ZoneID2["LOSPUER"] = 45] = "LOSPUER";
  ZoneID2[ZoneID2["MIRR"] = 46] = "MIRR";
  ZoneID2[ZoneID2["MORN"] = 47] = "MORN";
  ZoneID2[ZoneID2["MOVIE"] = 48] = "MOVIE";
  ZoneID2[ZoneID2["MTCHIL"] = 49] = "MTCHIL";
  ZoneID2[ZoneID2["MTGORDO"] = 50] = "MTGORDO";
  ZoneID2[ZoneID2["MTJOSE"] = 51] = "MTJOSE";
  ZoneID2[ZoneID2["MURRI"] = 52] = "MURRI";
  ZoneID2[ZoneID2["NCHU"] = 53] = "NCHU";
  ZoneID2[ZoneID2["NOOSE"] = 54] = "NOOSE";
  ZoneID2[ZoneID2["OCEANA"] = 55] = "OCEANA";
  ZoneID2[ZoneID2["PALCOV"] = 56] = "PALCOV";
  ZoneID2[ZoneID2["PALETO"] = 57] = "PALETO";
  ZoneID2[ZoneID2["PALFOR"] = 58] = "PALFOR";
  ZoneID2[ZoneID2["PALHIGH"] = 59] = "PALHIGH";
  ZoneID2[ZoneID2["PALMPOW"] = 60] = "PALMPOW";
  ZoneID2[ZoneID2["PBLUFF"] = 61] = "PBLUFF";
  ZoneID2[ZoneID2["PBOX"] = 62] = "PBOX";
  ZoneID2[ZoneID2["PROCOB"] = 63] = "PROCOB";
  ZoneID2[ZoneID2["RANCHO"] = 64] = "RANCHO";
  ZoneID2[ZoneID2["RGLEN"] = 65] = "RGLEN";
  ZoneID2[ZoneID2["RICHM"] = 66] = "RICHM";
  ZoneID2[ZoneID2["ROCKF"] = 67] = "ROCKF";
  ZoneID2[ZoneID2["RTRAK"] = 68] = "RTRAK";
  ZoneID2[ZoneID2["SanAnd"] = 69] = "SanAnd";
  ZoneID2[ZoneID2["SANCHIA"] = 70] = "SANCHIA";
  ZoneID2[ZoneID2["SANDY"] = 71] = "SANDY";
  ZoneID2[ZoneID2["SKID"] = 72] = "SKID";
  ZoneID2[ZoneID2["SLAB"] = 73] = "SLAB";
  ZoneID2[ZoneID2["STAD"] = 74] = "STAD";
  ZoneID2[ZoneID2["STRAW"] = 75] = "STRAW";
  ZoneID2[ZoneID2["TATAMO"] = 76] = "TATAMO";
  ZoneID2[ZoneID2["TERMINA"] = 77] = "TERMINA";
  ZoneID2[ZoneID2["TEXTI"] = 78] = "TEXTI";
  ZoneID2[ZoneID2["TONGVAH"] = 79] = "TONGVAH";
  ZoneID2[ZoneID2["TONGVAV"] = 80] = "TONGVAV";
  ZoneID2[ZoneID2["VCANA"] = 81] = "VCANA";
  ZoneID2[ZoneID2["VESP"] = 82] = "VESP";
  ZoneID2[ZoneID2["VINE"] = 83] = "VINE";
  ZoneID2[ZoneID2["WINDF"] = 84] = "WINDF";
  ZoneID2[ZoneID2["WVINE"] = 85] = "WVINE";
  ZoneID2[ZoneID2["ZANCUDO"] = 86] = "ZANCUDO";
  ZoneID2[ZoneID2["ZP_ORT"] = 87] = "ZP_ORT";
  ZoneID2[ZoneID2["ZQ_UAR"] = 88] = "ZQ_UAR";
  return ZoneID2;
})(ZoneID || {});
export {
  ZoneID
};
