// @nativewrappers/client/lib/weaponComponent/ComponentAttachmentPoint.js
export var ComponentAttachmentPoint;
(function (ComponentAttachmentPoint) {
    ComponentAttachmentPoint[ComponentAttachmentPoint["Invalid"] = 4294967295] = "Invalid";
    ComponentAttachmentPoint[ComponentAttachmentPoint["Clip"] = 3723347892] = "Clip";
    ComponentAttachmentPoint[ComponentAttachmentPoint["Clip2"] = 291640902] = "Clip2";
    ComponentAttachmentPoint[ComponentAttachmentPoint["FlashLaser"] = 679107254] = "FlashLaser";
    ComponentAttachmentPoint[ComponentAttachmentPoint["FlashLaser2"] = 2722126698] = "FlashLaser2";
    ComponentAttachmentPoint[ComponentAttachmentPoint["Supp"] = 1863181664] = "Supp";
    ComponentAttachmentPoint[ComponentAttachmentPoint["Supp2"] = 945598191] = "Supp2";
    ComponentAttachmentPoint[ComponentAttachmentPoint["GunRoot"] = 962500902] = "GunRoot";
    ComponentAttachmentPoint[ComponentAttachmentPoint["Scope"] = 196630833] = "Scope";
    ComponentAttachmentPoint[ComponentAttachmentPoint["Scope2"] = 1684637069] = "Scope2";
    ComponentAttachmentPoint[ComponentAttachmentPoint["Grip"] = 2972950469] = "Grip";
    ComponentAttachmentPoint[ComponentAttachmentPoint["Grip2"] = 3748215485] = "Grip2";
    ComponentAttachmentPoint[ComponentAttachmentPoint["TorchBulb"] = 421673795] = "TorchBulb";
    ComponentAttachmentPoint[ComponentAttachmentPoint["Rail"] = 2451679629] = "Rail";
    ComponentAttachmentPoint[ComponentAttachmentPoint["Rail2"] = 497110245] = "Rail2";
    ComponentAttachmentPoint[ComponentAttachmentPoint["Barrel"] = 2982890265] = "Barrel";
})(ComponentAttachmentPoint || (ComponentAttachmentPoint = {}));
