// @nativewrappers/fivem/hashes/MaterialHash.js
var MaterialHash = /* @__PURE__ */ ((MaterialHash2) => {
  MaterialHash2[MaterialHash2["None"] = 0] = "None";
  MaterialHash2[MaterialHash2["Unk"] = 2519482235] = "Unk";
  MaterialHash2[MaterialHash2["Concrete"] = 1187676648] = "Concrete";
  MaterialHash2[MaterialHash2["ConcretePothole"] = 359120722] = "ConcretePothole";
  MaterialHash2[MaterialHash2["ConcreteDusty"] = 3210327185] = "ConcreteDusty";
  MaterialHash2[MaterialHash2["Tarmac"] = 282940568] = "Tarmac";
  MaterialHash2[MaterialHash2["TarmacPainted"] = 2993614768] = "TarmacPainted";
  MaterialHash2[MaterialHash2["TarmacPothole"] = 1886546517] = "TarmacPothole";
  MaterialHash2[MaterialHash2["RumbleStrip"] = 4044799021] = "RumbleStrip";
  MaterialHash2[MaterialHash2["BreezeBlock"] = 3340854742] = "BreezeBlock";
  MaterialHash2[MaterialHash2["Rock"] = 3454750755] = "Rock";
  MaterialHash2[MaterialHash2["RockMossy"] = 4170197704] = "RockMossy";
  MaterialHash2[MaterialHash2["Stone"] = 765206029] = "Stone";
  MaterialHash2[MaterialHash2["Cobblestone"] = 576169331] = "Cobblestone";
  MaterialHash2[MaterialHash2["Brick"] = 1639053622] = "Brick";
  MaterialHash2[MaterialHash2["Marble"] = 1945073303] = "Marble";
  MaterialHash2[MaterialHash2["PavingSlab"] = 1907048430] = "PavingSlab";
  MaterialHash2[MaterialHash2["SandstoneSolid"] = 592446772] = "SandstoneSolid";
  MaterialHash2[MaterialHash2["SandstoneBrittle"] = 1913209870] = "SandstoneBrittle";
  MaterialHash2[MaterialHash2["SandLoose"] = 2699818980] = "SandLoose";
  MaterialHash2[MaterialHash2["SandCompact"] = 510490462] = "SandCompact";
  MaterialHash2[MaterialHash2["SandWet"] = 909950165] = "SandWet";
  MaterialHash2[MaterialHash2["SandTrack"] = 2387446527] = "SandTrack";
  MaterialHash2[MaterialHash2["SandUnderwater"] = 3158909604] = "SandUnderwater";
  MaterialHash2[MaterialHash2["SandDryDeep"] = 509508168] = "SandDryDeep";
  MaterialHash2[MaterialHash2["SandWetDeep"] = 1288448767] = "SandWetDeep";
  MaterialHash2[MaterialHash2["Ice"] = 3508906581] = "Ice";
  MaterialHash2[MaterialHash2["IceTarmac"] = 2363942873] = "IceTarmac";
  MaterialHash2[MaterialHash2["SnowLoose"] = 2357397706] = "SnowLoose";
  MaterialHash2[MaterialHash2["SnowCompact"] = 3416406407] = "SnowCompact";
  MaterialHash2[MaterialHash2["SnowDeep"] = 1619704960] = "SnowDeep";
  MaterialHash2[MaterialHash2["SnowTarmac"] = 1550304810] = "SnowTarmac";
  MaterialHash2[MaterialHash2["GravelSmall"] = 951832588] = "GravelSmall";
  MaterialHash2[MaterialHash2["GravelLarge"] = 2128369009] = "GravelLarge";
  MaterialHash2[MaterialHash2["GravelDeep"] = 3938260814] = "GravelDeep";
  MaterialHash2[MaterialHash2["GravelTrainTrack"] = 1925605558] = "GravelTrainTrack";
  MaterialHash2[MaterialHash2["DirtTrack"] = 2409420175] = "DirtTrack";
  MaterialHash2[MaterialHash2["MudHard"] = 2352068586] = "MudHard";
  MaterialHash2[MaterialHash2["MudPothole"] = 312396330] = "MudPothole";
  MaterialHash2[MaterialHash2["MudSoft"] = 1635937914] = "MudSoft";
  MaterialHash2[MaterialHash2["MudUnderwater"] = 4021477129] = "MudUnderwater";
  MaterialHash2[MaterialHash2["MudDeep"] = 1109728704] = "MudDeep";
  MaterialHash2[MaterialHash2["Marsh"] = 223086562] = "Marsh";
  MaterialHash2[MaterialHash2["MarshDeep"] = 1584636462] = "MarshDeep";
  MaterialHash2[MaterialHash2["Soil"] = 3594309083] = "Soil";
  MaterialHash2[MaterialHash2["ClayHard"] = 1144315879] = "ClayHard";
  MaterialHash2[MaterialHash2["ClaySoft"] = 560985072] = "ClaySoft";
  MaterialHash2[MaterialHash2["GrassLong"] = 3833216577] = "GrassLong";
  MaterialHash2[MaterialHash2["Grass"] = 1333033863] = "Grass";
  MaterialHash2[MaterialHash2["GrassShort"] = 3008270349] = "GrassShort";
  MaterialHash2[MaterialHash2["Hay"] = 2461440131] = "Hay";
  MaterialHash2[MaterialHash2["Bushes"] = 581794674] = "Bushes";
  MaterialHash2[MaterialHash2["Twigs"] = 3381615457] = "Twigs";
  MaterialHash2[MaterialHash2["Leaves"] = 2253637325] = "Leaves";
  MaterialHash2[MaterialHash2["Woodchips"] = 3985845843] = "Woodchips";
  MaterialHash2[MaterialHash2["TreeBark"] = 2379541433] = "TreeBark";
  MaterialHash2[MaterialHash2["MetalSolidSmall"] = 2847687191] = "MetalSolidSmall";
  MaterialHash2[MaterialHash2["MetalSolidMedium"] = 3929336056] = "MetalSolidMedium";
  MaterialHash2[MaterialHash2["MetalSolidLarge"] = 752131025] = "MetalSolidLarge";
  MaterialHash2[MaterialHash2["MetalHollowSmall"] = 15972667] = "MetalHollowSmall";
  MaterialHash2[MaterialHash2["MetalHollowMedium"] = 1849540536] = "MetalHollowMedium";
  MaterialHash2[MaterialHash2["MetalHollowLarge"] = 3711753465] = "MetalHollowLarge";
  MaterialHash2[MaterialHash2["MetalChainlinkSmall"] = 762193613] = "MetalChainlinkSmall";
  MaterialHash2[MaterialHash2["MetalChainlinkLarge"] = 125958708] = "MetalChainlinkLarge";
  MaterialHash2[MaterialHash2["MetalCorrugatedIron"] = 834144982] = "MetalCorrugatedIron";
  MaterialHash2[MaterialHash2["MetalGrille"] = 3868849285] = "MetalGrille";
  MaterialHash2[MaterialHash2["MetalRailing"] = 2100727187] = "MetalRailing";
  MaterialHash2[MaterialHash2["MetalDuct"] = 1761524221] = "MetalDuct";
  MaterialHash2[MaterialHash2["MetalGarageDoor"] = 4063706601] = "MetalGarageDoor";
  MaterialHash2[MaterialHash2["MetalManhole"] = 3539969597] = "MetalManhole";
  MaterialHash2[MaterialHash2["WoodSolidSmall"] = 3895095068] = "WoodSolidSmall";
  MaterialHash2[MaterialHash2["WoodSolidMedium"] = 555004797] = "WoodSolidMedium";
  MaterialHash2[MaterialHash2["WoodSolidLarge"] = 815762359] = "WoodSolidLarge";
  MaterialHash2[MaterialHash2["WoodSolidPolished"] = 126470059] = "WoodSolidPolished";
  MaterialHash2[MaterialHash2["WoodFloorDusty"] = 3545514974] = "WoodFloorDusty";
  MaterialHash2[MaterialHash2["WoodHollowSmall"] = 1993976879] = "WoodHollowSmall";
  MaterialHash2[MaterialHash2["WoodHollowMedium"] = 3929491133] = "WoodHollowMedium";
  MaterialHash2[MaterialHash2["WoodHollowLarge"] = 3369548007] = "WoodHollowLarge";
  MaterialHash2[MaterialHash2["WoodChipboard"] = 1176309403] = "WoodChipboard";
  MaterialHash2[MaterialHash2["WoodOldCreaky"] = 722686013] = "WoodOldCreaky";
  MaterialHash2[MaterialHash2["WoodHighDensity"] = 2552123904] = "WoodHighDensity";
  MaterialHash2[MaterialHash2["WoodLattice"] = 2011204130] = "WoodLattice";
  MaterialHash2[MaterialHash2["Ceramic"] = 3108646581] = "Ceramic";
  MaterialHash2[MaterialHash2["RoofTile"] = 1755188853] = "RoofTile";
  MaterialHash2[MaterialHash2["RoofFelt"] = 2877802565] = "RoofFelt";
  MaterialHash2[MaterialHash2["Fibreglass"] = 1354180827] = "Fibreglass";
  MaterialHash2[MaterialHash2["Tarpaulin"] = 3652308448] = "Tarpaulin";
  MaterialHash2[MaterialHash2["Plastic"] = 2221655295] = "Plastic";
  MaterialHash2[MaterialHash2["PlasticHollow"] = 627123e3] = "PlasticHollow";
  MaterialHash2[MaterialHash2["PlasticHighDensity"] = 2668971817] = "PlasticHighDensity";
  MaterialHash2[MaterialHash2["PlasticClear"] = 2435246283] = "PlasticClear";
  MaterialHash2[MaterialHash2["PlasticHollowClear"] = 772722531] = "PlasticHollowClear";
  MaterialHash2[MaterialHash2["PlasticHighDensityClear"] = 2956494126] = "PlasticHighDensityClear";
  MaterialHash2[MaterialHash2["FibreglassHollow"] = 3528912198] = "FibreglassHollow";
  MaterialHash2[MaterialHash2["Rubber"] = 4149231379] = "Rubber";
  MaterialHash2[MaterialHash2["RubberHollow"] = 3511032624] = "RubberHollow";
  MaterialHash2[MaterialHash2["Linoleum"] = 289630530] = "Linoleum";
  MaterialHash2[MaterialHash2["Laminate"] = 1845676458] = "Laminate";
  MaterialHash2[MaterialHash2["CarpetSolid"] = 669292054] = "CarpetSolid";
  MaterialHash2[MaterialHash2["CarpetSolidDusty"] = 158576196] = "CarpetSolidDusty";
  MaterialHash2[MaterialHash2["CarpetFloorboard"] = 2898482353] = "CarpetFloorboard";
  MaterialHash2[MaterialHash2["Cloth"] = 122789469] = "Cloth";
  MaterialHash2[MaterialHash2["PlasterSolid"] = 3720844863] = "PlasterSolid";
  MaterialHash2[MaterialHash2["PlasterBrittle"] = 4043078398] = "PlasterBrittle";
  MaterialHash2[MaterialHash2["CardboardSheet"] = 236511221] = "CardboardSheet";
  MaterialHash2[MaterialHash2["CardboardBox"] = 2885912856] = "CardboardBox";
  MaterialHash2[MaterialHash2["Paper"] = 474149820] = "Paper";
  MaterialHash2[MaterialHash2["Foam"] = 808719444] = "Foam";
  MaterialHash2[MaterialHash2["FeatherPillow"] = 1341866303] = "FeatherPillow";
  MaterialHash2[MaterialHash2["Polystyrene"] = 2538039965] = "Polystyrene";
  MaterialHash2[MaterialHash2["Leather"] = 3724496396] = "Leather";
  MaterialHash2[MaterialHash2["Tvscreen"] = 1429989756] = "Tvscreen";
  MaterialHash2[MaterialHash2["SlattedBlinds"] = 673696729] = "SlattedBlinds";
  MaterialHash2[MaterialHash2["GlassShootThrough"] = 937503243] = "GlassShootThrough";
  MaterialHash2[MaterialHash2["GlassBulletproof"] = 244521486] = "GlassBulletproof";
  MaterialHash2[MaterialHash2["GlassOpaque"] = 1500272081] = "GlassOpaque";
  MaterialHash2[MaterialHash2["Perspex"] = 2675173228] = "Perspex";
  MaterialHash2[MaterialHash2["CarMetal"] = 4201905313] = "CarMetal";
  MaterialHash2[MaterialHash2["CarPlastic"] = 2137197282] = "CarPlastic";
  MaterialHash2[MaterialHash2["CarSofttop"] = 3315319434] = "CarSofttop";
  MaterialHash2[MaterialHash2["CarSofttopClear"] = 2130571536] = "CarSofttopClear";
  MaterialHash2[MaterialHash2["CarGlassWeak"] = 1247281098] = "CarGlassWeak";
  MaterialHash2[MaterialHash2["CarGlassMedium"] = 602884284] = "CarGlassMedium";
  MaterialHash2[MaterialHash2["CarGlassStrong"] = 1070994698] = "CarGlassStrong";
  MaterialHash2[MaterialHash2["CarGlassBulletproof"] = 2573051366] = "CarGlassBulletproof";
  MaterialHash2[MaterialHash2["CarGlassOpaque"] = 513061559] = "CarGlassOpaque";
  MaterialHash2[MaterialHash2["Water"] = 435688960] = "Water";
  MaterialHash2[MaterialHash2["Blood"] = 5236042] = "Blood";
  MaterialHash2[MaterialHash2["Oil"] = 3660485991] = "Oil";
  MaterialHash2[MaterialHash2["Petrol"] = 2660782956] = "Petrol";
  MaterialHash2[MaterialHash2["FreshMeat"] = 868733839] = "FreshMeat";
  MaterialHash2[MaterialHash2["DriedMeat"] = 2849806867] = "DriedMeat";
  MaterialHash2[MaterialHash2["EmissiveGlass"] = 1501078253] = "EmissiveGlass";
  MaterialHash2[MaterialHash2["EmissivePlastic"] = 1059629996] = "EmissivePlastic";
  MaterialHash2[MaterialHash2["VfxMetalElectrified"] = 3985833031] = "VfxMetalElectrified";
  MaterialHash2[MaterialHash2["VfxMetalWaterTower"] = 611561919] = "VfxMetalWaterTower";
  MaterialHash2[MaterialHash2["VfxMetalSteam"] = 3603690002] = "VfxMetalSteam";
  MaterialHash2[MaterialHash2["VfxMetalFlame"] = 332778253] = "VfxMetalFlame";
  MaterialHash2[MaterialHash2["PhysNoFriction"] = 1666473731] = "PhysNoFriction";
  MaterialHash2[MaterialHash2["PhysGolfBall"] = 2601153738] = "PhysGolfBall";
  MaterialHash2[MaterialHash2["PhysTennisBall"] = 4038262533] = "PhysTennisBall";
  MaterialHash2[MaterialHash2["PhysCaster"] = 4059664613] = "PhysCaster";
  MaterialHash2[MaterialHash2["PhysCasterRusty"] = 2016463089] = "PhysCasterRusty";
  MaterialHash2[MaterialHash2["PhysCarVoid"] = 1345867677] = "PhysCarVoid";
  MaterialHash2[MaterialHash2["PhysPedCapsule"] = 4003336261] = "PhysPedCapsule";
  MaterialHash2[MaterialHash2["PhysElectricFence"] = 3124923563] = "PhysElectricFence";
  MaterialHash2[MaterialHash2["PhysElectricMetal"] = 2281206151] = "PhysElectricMetal";
  MaterialHash2[MaterialHash2["PhysBarbedWire"] = 2751643840] = "PhysBarbedWire";
  MaterialHash2[MaterialHash2["PhysPooltableSurface"] = 605776921] = "PhysPooltableSurface";
  MaterialHash2[MaterialHash2["PhysPooltableCushion"] = 972939963] = "PhysPooltableCushion";
  MaterialHash2[MaterialHash2["PhysPooltableBall"] = 3546625734] = "PhysPooltableBall";
  MaterialHash2[MaterialHash2["Buttocks"] = 483400232] = "Buttocks";
  MaterialHash2[MaterialHash2["ThighLeft"] = 3834431425] = "ThighLeft";
  MaterialHash2[MaterialHash2["ShinLeft"] = 652772852] = "ShinLeft";
  MaterialHash2[MaterialHash2["FootLeft"] = 1926285543] = "FootLeft";
  MaterialHash2[MaterialHash2["ThighRight"] = 4057986041] = "ThighRight";
  MaterialHash2[MaterialHash2["ShinRight"] = 3848931141] = "ShinRight";
  MaterialHash2[MaterialHash2["FootRight"] = 2925830612] = "FootRight";
  MaterialHash2[MaterialHash2["Spine0"] = 2372680412] = "Spine0";
  MaterialHash2[MaterialHash2["Spine1"] = 3154854427] = "Spine1";
  MaterialHash2[MaterialHash2["Spine2"] = 1457572381] = "Spine2";
  MaterialHash2[MaterialHash2["Spine3"] = 32752644] = "Spine3";
  MaterialHash2[MaterialHash2["ClavicleLeft"] = 2825350831] = "ClavicleLeft";
  MaterialHash2[MaterialHash2["UpperArmLeft"] = 3784624938] = "UpperArmLeft";
  MaterialHash2[MaterialHash2["LowerArmLeft"] = 1045062756] = "LowerArmLeft";
  MaterialHash2[MaterialHash2["HandLeft"] = 113101985] = "HandLeft";
  MaterialHash2[MaterialHash2["ClavicleRight"] = 2737678298] = "ClavicleRight";
  MaterialHash2[MaterialHash2["UpperArmRight"] = 1501153539] = "UpperArmRight";
  MaterialHash2[MaterialHash2["LowerArmRight"] = 1777921590] = "LowerArmRight";
  MaterialHash2[MaterialHash2["HandRight"] = 2000961972] = "HandRight";
  MaterialHash2[MaterialHash2["Neck"] = 1718294164] = "Neck";
  MaterialHash2[MaterialHash2["Head"] = 3559574543] = "Head";
  MaterialHash2[MaterialHash2["AnimalDefault"] = 286224918] = "AnimalDefault";
  MaterialHash2[MaterialHash2["CarEngine"] = 2378027672] = "CarEngine";
  MaterialHash2[MaterialHash2["Puddle"] = 999829011] = "Puddle";
  MaterialHash2[MaterialHash2["ConcretePavement"] = 2015599386] = "ConcretePavement";
  MaterialHash2[MaterialHash2["BrickPavement"] = 3147605720] = "BrickPavement";
  MaterialHash2[MaterialHash2["PhysDynamicCoverBound"] = 2247498441] = "PhysDynamicCoverBound";
  MaterialHash2[MaterialHash2["VfxWoodBeerBarrel"] = 998201806] = "VfxWoodBeerBarrel";
  MaterialHash2[MaterialHash2["WoodHighFriction"] = 2154880249] = "WoodHighFriction";
  MaterialHash2[MaterialHash2["RockNoinst"] = 127813971] = "RockNoinst";
  MaterialHash2[MaterialHash2["BushesNoinst"] = 1441114862] = "BushesNoinst";
  MaterialHash2[MaterialHash2["MetalSolidRoadSurface"] = 3565854962] = "MetalSolidRoadSurface";
  MaterialHash2[MaterialHash2["StuntRampSurface"] = 2206792300] = "StuntRampSurface";
  MaterialHash2[MaterialHash2["Temp01"] = 746881105] = "Temp01";
  MaterialHash2[MaterialHash2["Temp02"] = 2316997185] = "Temp02";
  MaterialHash2[MaterialHash2["Temp03"] = 1911121241] = "Temp03";
  MaterialHash2[MaterialHash2["Temp04"] = 1923995104] = "Temp04";
  MaterialHash2[MaterialHash2["Temp05"] = 2901304848] = "Temp05";
  MaterialHash2[MaterialHash2["Temp06"] = 1061250033] = "Temp06";
  MaterialHash2[MaterialHash2["Temp07"] = 2529443614] = "Temp07";
  MaterialHash2[MaterialHash2["Temp08"] = 1343679702] = "Temp08";
  MaterialHash2[MaterialHash2["Temp09"] = 1026054937] = "Temp09";
  MaterialHash2[MaterialHash2["Temp10"] = 63305994] = "Temp10";
  MaterialHash2[MaterialHash2["Temp11"] = 47470226] = "Temp11";
  MaterialHash2[MaterialHash2["Temp12"] = 702596674] = "Temp12";
  MaterialHash2[MaterialHash2["Temp13"] = 2657481383] = "Temp13";
  MaterialHash2[MaterialHash2["Temp14"] = 3649011722] = "Temp14";
  MaterialHash2[MaterialHash2["Temp15"] = 2710969365] = "Temp15";
  MaterialHash2[MaterialHash2["Temp16"] = 2782232023] = "Temp16";
  MaterialHash2[MaterialHash2["Temp17"] = 1011960114] = "Temp17";
  MaterialHash2[MaterialHash2["Temp18"] = 1354993138] = "Temp18";
  MaterialHash2[MaterialHash2["Temp19"] = 3493162850] = "Temp19";
  MaterialHash2[MaterialHash2["Temp20"] = 2242086891] = "Temp20";
  MaterialHash2[MaterialHash2["Temp21"] = 3257211236] = "Temp21";
  MaterialHash2[MaterialHash2["Temp22"] = 3674578943] = "Temp22";
  MaterialHash2[MaterialHash2["Temp23"] = 465002639] = "Temp23";
  MaterialHash2[MaterialHash2["Temp24"] = 1963820161] = "Temp24";
  MaterialHash2[MaterialHash2["Temp25"] = 1952288305] = "Temp25";
  MaterialHash2[MaterialHash2["Temp26"] = 3178714198] = "Temp26";
  MaterialHash2[MaterialHash2["Temp27"] = 889255498] = "Temp27";
  MaterialHash2[MaterialHash2["Temp28"] = 3115293198] = "Temp28";
  MaterialHash2[MaterialHash2["Temp29"] = 1078418101] = "Temp29";
  MaterialHash2[MaterialHash2["Temp30"] = 13626292] = "Temp30";
  return MaterialHash2;
})(MaterialHash || {});
export {
  MaterialHash
};
