// @nativewrappers/fivem/hashes/PedHash.js
var PedHash = /* @__PURE__ */ ((PedHash2) => {
  PedHash2[PedHash2["Michael"] = 225514697] = "Michael";
  PedHash2[PedHash2["Franklin"] = 2602752943] = "Franklin";
  PedHash2[PedHash2["Trevor"] = 2608926626] = "Trevor";
  PedHash2[PedHash2["Abigail"] = 1074457665] = "Abigail";
  PedHash2[PedHash2["Agent"] = 610988552] = "Agent";
  PedHash2[PedHash2["Agent14"] = 4227433577] = "Agent14";
  PedHash2[PedHash2["AmandaTownley"] = 1830688247] = "AmandaTownley";
  PedHash2[PedHash2["Andreas"] = 1206185632] = "Andreas";
  PedHash2[PedHash2["Ashley"] = 2129936603] = "Ashley";
  PedHash2[PedHash2["AviSchwartzman"] = 939183526] = "AviSchwartzman";
  PedHash2[PedHash2["Ballasog"] = 2802535058] = "Ballasog";
  PedHash2[PedHash2["Bankman"] = 2426248831] = "Bankman";
  PedHash2[PedHash2["Barry"] = 797459875] = "Barry";
  PedHash2[PedHash2["Bestmen"] = 1464257942] = "Bestmen";
  PedHash2[PedHash2["Beverly"] = 3181518428] = "Beverly";
  PedHash2[PedHash2["Brad"] = 3183167778] = "Brad";
  PedHash2[PedHash2["Bride"] = 1633872967] = "Bride";
  PedHash2[PedHash2["Car3Guy1"] = 2230970679] = "Car3Guy1";
  PedHash2[PedHash2["Car3Guy2"] = 1975732938] = "Car3Guy2";
  PedHash2[PedHash2["Casey"] = 3774489940] = "Casey";
  PedHash2[PedHash2["Chef"] = 1240128502] = "Chef";
  PedHash2[PedHash2["Chef2"] = 2240322243] = "Chef2";
  PedHash2[PedHash2["Clay"] = 1825562762] = "Clay";
  PedHash2[PedHash2["Claypain"] = 2634057640] = "Claypain";
  PedHash2[PedHash2["Cletus"] = 3865252245] = "Cletus";
  PedHash2[PedHash2["CrisFormage"] = 678319271] = "CrisFormage";
  PedHash2[PedHash2["Dale"] = 1182012905] = "Dale";
  PedHash2[PedHash2["DaveNorton"] = 365775923] = "DaveNorton";
  PedHash2[PedHash2["Denise"] = 2181772221] = "Denise";
  PedHash2[PedHash2["Devin"] = 1952555184] = "Devin";
  PedHash2[PedHash2["DoaMan"] = 1646160893] = "DoaMan";
  PedHash2[PedHash2["Dom"] = 2620240008] = "Dom";
  PedHash2[PedHash2["Dreyfuss"] = 3666413874] = "Dreyfuss";
  PedHash2[PedHash2["DrFriedlander"] = 3422293493] = "DrFriedlander";
  PedHash2[PedHash2["EdToh"] = 712602007] = "EdToh";
  PedHash2[PedHash2["Fabien"] = 3499148112] = "Fabien";
  PedHash2[PedHash2["FbiSuit01"] = 988062523] = "FbiSuit01";
  PedHash2[PedHash2["Floyd"] = 2981205682] = "Floyd";
  PedHash2[PedHash2["G"] = 2216405299] = "G";
  PedHash2[PedHash2["Groom"] = 4274948997] = "Groom";
  PedHash2[PedHash2["Hao"] = 1704428387] = "Hao";
  PedHash2[PedHash2["Hunter"] = 3457361118] = "Hunter";
  PedHash2[PedHash2["Janet"] = 225287241] = "Janet";
  PedHash2[PedHash2["JayNorris"] = 2050158196] = "JayNorris";
  PedHash2[PedHash2["Jewelass"] = 257763003] = "Jewelass";
  PedHash2[PedHash2["JimmyBoston"] = 3986688045] = "JimmyBoston";
  PedHash2[PedHash2["JimmyDisanto"] = 1459905209] = "JimmyDisanto";
  PedHash2[PedHash2["JoeMinuteman"] = 3189787803] = "JoeMinuteman";
  PedHash2[PedHash2["JohnnyKlebitz"] = 2278195374] = "JohnnyKlebitz";
  PedHash2[PedHash2["Josef"] = 3776618420] = "Josef";
  PedHash2[PedHash2["Josh"] = 2040438510] = "Josh";
  PedHash2[PedHash2["KarenDaniels"] = 3948009817] = "KarenDaniels";
  PedHash2[PedHash2["KerryMcintosh"] = 1530648845] = "KerryMcintosh";
  PedHash2[PedHash2["LamarDavis"] = 1706635382] = "LamarDavis";
  PedHash2[PedHash2["Lazlow"] = 3756278757] = "Lazlow";
  PedHash2[PedHash2["LesterCrest"] = 1302784073] = "LesterCrest";
  PedHash2[PedHash2["Lifeinvad01"] = 1401530684] = "Lifeinvad01";
  PedHash2[PedHash2["Lifeinvad02"] = 666718676] = "Lifeinvad02";
  PedHash2[PedHash2["Magenta"] = 4242313482] = "Magenta";
  PedHash2[PedHash2["Malc"] = 4055673113] = "Malc";
  PedHash2[PedHash2["Manuel"] = 4248931856] = "Manuel";
  PedHash2[PedHash2["Marnie"] = 411185872] = "Marnie";
  PedHash2[PedHash2["MaryAnn"] = 2741999622] = "MaryAnn";
  PedHash2[PedHash2["Maude"] = 1005070462] = "Maude";
  PedHash2[PedHash2["Michelle"] = 3214308084] = "Michelle";
  PedHash2[PedHash2["Milton"] = 3408943538] = "Milton";
  PedHash2[PedHash2["Molly"] = 2936266209] = "Molly";
  PedHash2[PedHash2["MrK"] = 3990661997] = "MrK";
  PedHash2[PedHash2["MrsPhillips"] = 946007720] = "MrsPhillips";
  PedHash2[PedHash2["MrsThornhill"] = 503621995] = "MrsThornhill";
  PedHash2[PedHash2["Natalia"] = 3726105915] = "Natalia";
  PedHash2[PedHash2["NervousRon"] = 3170921201] = "NervousRon";
  PedHash2[PedHash2["Nigel"] = 3367442045] = "Nigel";
  PedHash2[PedHash2["OldMan1a"] = 1906124788] = "OldMan1a";
  PedHash2[PedHash2["OldMan2"] = 4011150407] = "OldMan2";
  PedHash2[PedHash2["Omega"] = 1625728984] = "Omega";
  PedHash2[PedHash2["ONeil"] = 768005095] = "ONeil";
  PedHash2[PedHash2["Orleans"] = 1641334641] = "Orleans";
  PedHash2[PedHash2["Ortega"] = 648372919] = "Ortega";
  PedHash2[PedHash2["Paper"] = 2577072326] = "Paper";
  PedHash2[PedHash2["Patricia"] = 3312325004] = "Patricia";
  PedHash2[PedHash2["Popov"] = 645279998] = "Popov";
  PedHash2[PedHash2["Paige"] = 357551935] = "Paige";
  PedHash2[PedHash2["Priest"] = 1681385341] = "Priest";
  PedHash2[PedHash2["PrologueDriver"] = 2237544099] = "PrologueDriver";
  PedHash2[PedHash2["PrologueSec01"] = 1888624839] = "PrologueSec01";
  PedHash2[PedHash2["PrologueSec02"] = 666086773] = "PrologueSec02";
  PedHash2[PedHash2["RampGang"] = 3845001836] = "RampGang";
  PedHash2[PedHash2["RampHic"] = 1165307954] = "RampHic";
  PedHash2[PedHash2["RampHipster"] = 3740245870] = "RampHipster";
  PedHash2[PedHash2["RampMex"] = 3870061732] = "RampMex";
  PedHash2[PedHash2["Rashkovsky"] = 940326374] = "Rashkovsky";
  PedHash2[PedHash2["RoccoPelosi"] = 3585757951] = "RoccoPelosi";
  PedHash2[PedHash2["RussianDrunk"] = 1024089777] = "RussianDrunk";
  PedHash2[PedHash2["ScreenWriter"] = 4293277303] = "ScreenWriter";
  PedHash2[PedHash2["SiemonYetarian"] = 1283141381] = "SiemonYetarian";
  PedHash2[PedHash2["Solomon"] = 2260598310] = "Solomon";
  PedHash2[PedHash2["SteveHains"] = 941695432] = "SteveHains";
  PedHash2[PedHash2["Stretch"] = 915948376] = "Stretch";
  PedHash2[PedHash2["Talina"] = 3885222120] = "Talina";
  PedHash2[PedHash2["Tanisha"] = 226559113] = "Tanisha";
  PedHash2[PedHash2["TaoCheng"] = 3697041061] = "TaoCheng";
  PedHash2[PedHash2["TaosTranslator"] = 2089096292] = "TaosTranslator";
  PedHash2[PedHash2["TennisCoach"] = 2721800023] = "TennisCoach";
  PedHash2[PedHash2["Terry"] = 1728056212] = "Terry";
  PedHash2[PedHash2["TomEpsilon"] = 3447159466] = "TomEpsilon";
  PedHash2[PedHash2["Tonya"] = 3402126148] = "Tonya";
  PedHash2[PedHash2["TracyDisanto"] = 3728026165] = "TracyDisanto";
  PedHash2[PedHash2["TrafficWarden"] = 1461287021] = "TrafficWarden";
  PedHash2[PedHash2["TylerDixon"] = 1382414087] = "TylerDixon";
  PedHash2[PedHash2["VagosSpeak"] = 4194109068] = "VagosSpeak";
  PedHash2[PedHash2["Wade"] = 2459507570] = "Wade";
  PedHash2[PedHash2["WeiCheng"] = 2867128955] = "WeiCheng";
  PedHash2[PedHash2["Zimbor"] = 188012277] = "Zimbor";
  PedHash2[PedHash2["AbigailCutscene"] = 2306246977] = "AbigailCutscene";
  PedHash2[PedHash2["AgentCutscene"] = 3614493108] = "AgentCutscene";
  PedHash2[PedHash2["Agent14Cutscene"] = 1841036427] = "Agent14Cutscene";
  PedHash2[PedHash2["AmandaTownleyCutscene"] = 2515474659] = "AmandaTownleyCutscene";
  PedHash2[PedHash2["AndreasCutscene"] = 3881194279] = "AndreasCutscene";
  PedHash2[PedHash2["AnitaCutscene"] = 117698822] = "AnitaCutscene";
  PedHash2[PedHash2["AntonCutscene"] = 2781317046] = "AntonCutscene";
  PedHash2[PedHash2["AshleyCutscene"] = 650367097] = "AshleyCutscene";
  PedHash2[PedHash2["AviSchwartzmanCutscene"] = 2560490906] = "AviSchwartzmanCutscene";
  PedHash2[PedHash2["BallasogCutscene"] = 2884567044] = "BallasogCutscene";
  PedHash2[PedHash2["BankmanCutscene"] = 2539657518] = "BankmanCutscene";
  PedHash2[PedHash2["BarryCutscene"] = 1767447799] = "BarryCutscene";
  PedHash2[PedHash2["BeverlyCutscene"] = 3027157846] = "BeverlyCutscene";
  PedHash2[PedHash2["BradCutscene"] = 4024807398] = "BradCutscene";
  PedHash2[PedHash2["BradCadaverCutscene"] = 1915268960] = "BradCadaverCutscene";
  PedHash2[PedHash2["BrideCutscene"] = 2193587873] = "BrideCutscene";
  PedHash2[PedHash2["BurgerDrugCutscene"] = 2363277399] = "BurgerDrugCutscene";
  PedHash2[PedHash2["Car3Guy1Cutscene"] = 71501447] = "Car3Guy1Cutscene";
  PedHash2[PedHash2["Car3Guy2Cutscene"] = 327394568] = "Car3Guy2Cutscene";
  PedHash2[PedHash2["CarBuyerCutscene"] = 2362341647] = "CarBuyerCutscene";
  PedHash2[PedHash2["CaseyCutscene"] = 3935738944] = "CaseyCutscene";
  PedHash2[PedHash2["ChefCutscene"] = 2739391114] = "ChefCutscene";
  PedHash2[PedHash2["Chef2Cutscene"] = 2925257274] = "Chef2Cutscene";
  PedHash2[PedHash2["ChinGoonCutscene"] = 2831296918] = "ChinGoonCutscene";
  PedHash2[PedHash2["ClayCutscene"] = 3687553076] = "ClayCutscene";
  PedHash2[PedHash2["CletusCutscene"] = 3404326357] = "CletusCutscene";
  PedHash2[PedHash2["CopCutscene"] = 2595446627] = "CopCutscene";
  PedHash2[PedHash2["CrisFormageCutscene"] = 3253960934] = "CrisFormageCutscene";
  PedHash2[PedHash2["CustomerCutscene"] = 2756669323] = "CustomerCutscene";
  PedHash2[PedHash2["DaleCutscene"] = 216536661] = "DaleCutscene";
  PedHash2[PedHash2["DaveNortonCutscene"] = 2240226444] = "DaveNortonCutscene";
  PedHash2[PedHash2["DebraCutscene"] = 3973074921] = "DebraCutscene";
  PedHash2[PedHash2["DeniseCutscene"] = 1870669624] = "DeniseCutscene";
  PedHash2[PedHash2["DeniseFriendCutscene"] = 3045926185] = "DeniseFriendCutscene";
  PedHash2[PedHash2["DevinCutscene"] = 788622594] = "DevinCutscene";
  PedHash2[PedHash2["DomCutscene"] = 1198698306] = "DomCutscene";
  PedHash2[PedHash2["DreyfussCutscene"] = 1012965715] = "DreyfussCutscene";
  PedHash2[PedHash2["DrFriedlanderCutscene"] = 2745392175] = "DrFriedlanderCutscene";
  PedHash2[PedHash2["FabienCutscene"] = 1191403201] = "FabienCutscene";
  PedHash2[PedHash2["FbiSuit01Cutscene"] = 1482427218] = "FbiSuit01Cutscene";
  PedHash2[PedHash2["FloydCutscene"] = 103106535] = "FloydCutscene";
  PedHash2[PedHash2["FosRepCutscene"] = 466359675] = "FosRepCutscene";
  PedHash2[PedHash2["GCutscene"] = 2727244247] = "GCutscene";
  PedHash2[PedHash2["GroomCutscene"] = 2058033618] = "GroomCutscene";
  PedHash2[PedHash2["GroveStrDlrCutscene"] = 3898166818] = "GroveStrDlrCutscene";
  PedHash2[PedHash2["GuadalopeCutscene"] = 261428209] = "GuadalopeCutscene";
  PedHash2[PedHash2["GurkCutscene"] = 3272931111] = "GurkCutscene";
  PedHash2[PedHash2["HaoCutscene"] = 3969814300] = "HaoCutscene";
  PedHash2[PedHash2["HughCutscene"] = 1863555924] = "HughCutscene";
  PedHash2[PedHash2["HunterCutscene"] = 1531218220] = "HunterCutscene";
  PedHash2[PedHash2["ImranCutscene"] = 3812756443] = "ImranCutscene";
  PedHash2[PedHash2["JackHowitzerCutscene"] = 1153203121] = "JackHowitzerCutscene";
  PedHash2[PedHash2["JanetCutscene"] = 808778210] = "JanetCutscene";
  PedHash2[PedHash2["JanitorCutscene"] = 3254803008] = "JanitorCutscene";
  PedHash2[PedHash2["JewelassCutscene"] = 1145088004] = "JewelassCutscene";
  PedHash2[PedHash2["JimmyBostonCutscene"] = 60192701] = "JimmyBostonCutscene";
  PedHash2[PedHash2["JimmyDisantoCutscene"] = 3100414644] = "JimmyDisantoCutscene";
  PedHash2[PedHash2["JoeMinutemanCutscene"] = 4036845097] = "JoeMinutemanCutscene";
  PedHash2[PedHash2["JohnnyKlebitzCutscene"] = 4203395201] = "JohnnyKlebitzCutscene";
  PedHash2[PedHash2["JosefCutscene"] = 1167549130] = "JosefCutscene";
  PedHash2[PedHash2["JoshCutscene"] = 1158606749] = "JoshCutscene";
  PedHash2[PedHash2["KarenDanielsCutscene"] = 1269774364] = "KarenDanielsCutscene";
  PedHash2[PedHash2["LamarDavisCutscene"] = 1162230285] = "LamarDavisCutscene";
  PedHash2[PedHash2["LazlowCutscene"] = 949295643] = "LazlowCutscene";
  PedHash2[PedHash2["LesterCrestCutscene"] = 3046438339] = "LesterCrestCutscene";
  PedHash2[PedHash2["Lifeinvad01Cutscene"] = 1918178165] = "Lifeinvad01Cutscene";
  PedHash2[PedHash2["MagentaCutscene"] = 1477887514] = "MagentaCutscene";
  PedHash2[PedHash2["ManuelCutscene"] = 4222842058] = "ManuelCutscene";
  PedHash2[PedHash2["MarnieCutscene"] = 1464721716] = "MarnieCutscene";
  PedHash2[PedHash2["MartinMadrazoCutscene"] = 1129928304] = "MartinMadrazoCutscene";
  PedHash2[PedHash2["MaryannCutscene"] = 161007533] = "MaryannCutscene";
  PedHash2[PedHash2["MaudeCutscene"] = 3166991819] = "MaudeCutscene";
  PedHash2[PedHash2["MerryWeatherCutscene"] = 1631478380] = "MerryWeatherCutscene";
  PedHash2[PedHash2["MichelleCutscene"] = 1890499016] = "MichelleCutscene";
  PedHash2[PedHash2["MiltonCutscene"] = 3077190415] = "MiltonCutscene";
  PedHash2[PedHash2["MollyCutscene"] = 1167167044] = "MollyCutscene";
  PedHash2[PedHash2["MoviePremFemaleCutscene"] = 1270514905] = "MoviePremFemaleCutscene";
  PedHash2[PedHash2["MoviePremMaleCutscene"] = 2372398717] = "MoviePremMaleCutscene";
  PedHash2[PedHash2["MrKCutscene"] = 3284966005] = "MrKCutscene";
  PedHash2[PedHash2["MrsPhillipsCutscene"] = 3422397391] = "MrsPhillipsCutscene";
  PedHash2[PedHash2["MrsThornhillCutscene"] = 1334976110] = "MrsThornhillCutscene";
  PedHash2[PedHash2["NataliaCutscene"] = 1325314544] = "NataliaCutscene";
  PedHash2[PedHash2["NervousRonCutscene"] = 2023152276] = "NervousRonCutscene";
  PedHash2[PedHash2["NigelCutscene"] = 3779566603] = "NigelCutscene";
  PedHash2[PedHash2["OldMan1aCutscene"] = 518814684] = "OldMan1aCutscene";
  PedHash2[PedHash2["OldMan2Cutscene"] = 2566514544] = "OldMan2Cutscene";
  PedHash2[PedHash2["OmegaCutscene"] = 2339419141] = "OmegaCutscene";
  PedHash2[PedHash2["OrleansCutscene"] = 2905870170] = "OrleansCutscene";
  PedHash2[PedHash2["OrtegaCutscene"] = 3235579087] = "OrtegaCutscene";
  PedHash2[PedHash2["OscarCutscene"] = 4095687067] = "OscarCutscene";
  PedHash2[PedHash2["PaigeCutscene"] = 1528799427] = "PaigeCutscene";
  PedHash2[PedHash2["PaperCutscene"] = 1798879480] = "PaperCutscene";
  PedHash2[PedHash2["PopovCutscene"] = 1635617250] = "PopovCutscene";
  PedHash2[PedHash2["PatriciaCutscene"] = 3750433537] = "PatriciaCutscene";
  PedHash2[PedHash2["PornDudesCutscene"] = 793443893] = "PornDudesCutscene";
  PedHash2[PedHash2["PriestCutscene"] = 1299047806] = "PriestCutscene";
  PedHash2[PedHash2["PrologueDriverCutscene"] = 4027271643] = "PrologueDriverCutscene";
  PedHash2[PedHash2["PrologueSec01Cutscene"] = 2141384740] = "PrologueSec01Cutscene";
  PedHash2[PedHash2["PrologueSec02Cutscene"] = 512955554] = "PrologueSec02Cutscene";
  PedHash2[PedHash2["RampGangCutscene"] = 3263172030] = "RampGangCutscene";
  PedHash2[PedHash2["RampHicCutscene"] = 2240582840] = "RampHicCutscene";
  PedHash2[PedHash2["RampHipsterCutscene"] = 569740212] = "RampHipsterCutscene";
  PedHash2[PedHash2["RampMarineCutscene"] = 1634506681] = "RampMarineCutscene";
  PedHash2[PedHash2["RampMexCutscene"] = 4132362192] = "RampMexCutscene";
  PedHash2[PedHash2["RashkovskyCutscene"] = 411081129] = "RashkovskyCutscene";
  PedHash2[PedHash2["ReporterCutscene"] = 776079908] = "ReporterCutscene";
  PedHash2[PedHash2["RoccoPelosiCutscene"] = 2858686092] = "RoccoPelosiCutscene";
  PedHash2[PedHash2["RussianDrunkCutscene"] = 1179785778] = "RussianDrunkCutscene";
  PedHash2[PedHash2["ScreenWriterCutscene"] = 2346790124] = "ScreenWriterCutscene";
  PedHash2[PedHash2["SiemonYetarianCutscene"] = 3230888450] = "SiemonYetarianCutscene";
  PedHash2[PedHash2["SolomonCutscene"] = 4140949582] = "SolomonCutscene";
  PedHash2[PedHash2["SteveHainsCutscene"] = 2766184958] = "SteveHainsCutscene";
  PedHash2[PedHash2["StretchCutscene"] = 2302502917] = "StretchCutscene";
  PedHash2[PedHash2["Stripper01Cutscene"] = 2934601397] = "Stripper01Cutscene";
  PedHash2[PedHash2["Stripper02Cutscene"] = 2168724337] = "Stripper02Cutscene";
  PedHash2[PedHash2["TanishaCutscene"] = 1123963760] = "TanishaCutscene";
  PedHash2[PedHash2["TaoChengCutscene"] = 2288257085] = "TaoChengCutscene";
  PedHash2[PedHash2["TaosTranslatorCutscene"] = 1397974313] = "TaosTranslatorCutscene";
  PedHash2[PedHash2["TennisCoachCutscene"] = 1545995274] = "TennisCoachCutscene";
  PedHash2[PedHash2["TerryCutscene"] = 978452933] = "TerryCutscene";
  PedHash2[PedHash2["TomCutscene"] = 1776856003] = "TomCutscene";
  PedHash2[PedHash2["TomEpsilonCutscene"] = 2349847778] = "TomEpsilonCutscene";
  PedHash2[PedHash2["TonyaCutscene"] = 1665391897] = "TonyaCutscene";
  PedHash2[PedHash2["TracyDisantoCutscene"] = 101298480] = "TracyDisantoCutscene";
  PedHash2[PedHash2["TrafficWardenCutscene"] = 3727243251] = "TrafficWardenCutscene";
  PedHash2[PedHash2["UndercoverCopCutscene"] = 4017642090] = "UndercoverCopCutscene";
  PedHash2[PedHash2["VagosSpeakCutscene"] = 1224690857] = "VagosSpeakCutscene";
  PedHash2[PedHash2["WadeCutscene"] = 3529955798] = "WadeCutscene";
  PedHash2[PedHash2["WeiChengCutscene"] = 819699067] = "WeiChengCutscene";
  PedHash2[PedHash2["ZimborCutscene"] = 3937184496] = "ZimborCutscene";
  PedHash2[PedHash2["Boar"] = 3462393972] = "Boar";
  PedHash2[PedHash2["Cat"] = 1462895032] = "Cat";
  PedHash2[PedHash2["ChickenHawk"] = 2864127842] = "ChickenHawk";
  PedHash2[PedHash2["Chimp"] = 2825402133] = "Chimp";
  PedHash2[PedHash2["Chop"] = 351016938] = "Chop";
  PedHash2[PedHash2["Cormorant"] = 1457690978] = "Cormorant";
  PedHash2[PedHash2["Cow"] = 4244282910] = "Cow";
  PedHash2[PedHash2["Coyote"] = 1682622302] = "Coyote";
  PedHash2[PedHash2["Crow"] = 402729631] = "Crow";
  PedHash2[PedHash2["Deer"] = 3630914197] = "Deer";
  PedHash2[PedHash2["Dolphin"] = 2344268885] = "Dolphin";
  PedHash2[PedHash2["Fish"] = 802685111] = "Fish";
  PedHash2[PedHash2["Hen"] = 1794449327] = "Hen";
  PedHash2[PedHash2["HammerShark"] = 1015224100] = "HammerShark";
  PedHash2[PedHash2["Humpback"] = 1193010354] = "Humpback";
  PedHash2[PedHash2["Husky"] = 1318032802] = "Husky";
  PedHash2[PedHash2["KillerWhale"] = 2374682809] = "KillerWhale";
  PedHash2[PedHash2["MountainLion"] = 307287994] = "MountainLion";
  PedHash2[PedHash2["Pig"] = 2971380566] = "Pig";
  PedHash2[PedHash2["Pigeon"] = 111281960] = "Pigeon";
  PedHash2[PedHash2["Poodle"] = 1125994524] = "Poodle";
  PedHash2[PedHash2["Pug"] = 1832265812] = "Pug";
  PedHash2[PedHash2["Rabbit"] = 3753204865] = "Rabbit";
  PedHash2[PedHash2["Rat"] = 3283429734] = "Rat";
  PedHash2[PedHash2["Retriever"] = 882848737] = "Retriever";
  PedHash2[PedHash2["Rhesus"] = 3268439891] = "Rhesus";
  PedHash2[PedHash2["Rottweiler"] = 2506301981] = "Rottweiler";
  PedHash2[PedHash2["Seagull"] = 3549666813] = "Seagull";
  PedHash2[PedHash2["Shepherd"] = 1126154828] = "Shepherd";
  PedHash2[PedHash2["Stingray"] = 2705875277] = "Stingray";
  PedHash2[PedHash2["TigerShark"] = 113504370] = "TigerShark";
  PedHash2[PedHash2["Westy"] = 2910340283] = "Westy";
  PedHash2[PedHash2["Abner"] = 4037813798] = "Abner";
  PedHash2[PedHash2["AlDiNapoli"] = 4042020578] = "AlDiNapoli";
  PedHash2[PedHash2["Antonb"] = 3479321132] = "Antonb";
  PedHash2[PedHash2["Armoured01"] = 3455013896] = "Armoured01";
  PedHash2[PedHash2["Babyd"] = 3658575486] = "Babyd";
  PedHash2[PedHash2["Bankman01"] = 3272005365] = "Bankman01";
  PedHash2[PedHash2["Baygor"] = 1380197501] = "Baygor";
  PedHash2[PedHash2["Benny"] = 3300333010] = "Benny";
  PedHash2[PedHash2["BikeHire01"] = 1984382277] = "BikeHire01";
  PedHash2[PedHash2["BikerChic"] = 4198014287] = "BikerChic";
  PedHash2[PedHash2["BoatStaff01M"] = 3361671816] = "BoatStaff01M";
  PedHash2[PedHash2["BoatStaff01F"] = 848542878] = "BoatStaff01F";
  PedHash2[PedHash2["BurgerDrug"] = 2340239206] = "BurgerDrug";
  PedHash2[PedHash2["Chip"] = 610290475] = "Chip";
  PedHash2[PedHash2["Claude01"] = 3237179831] = "Claude01";
  PedHash2[PedHash2["ClubHouseBar01"] = 1914945105] = "ClubHouseBar01";
  PedHash2[PedHash2["CocaineFemale01"] = 1897303236] = "CocaineFemale01";
  PedHash2[PedHash2["CocaineMale01"] = 3455927962] = "CocaineMale01";
  PedHash2[PedHash2["ComJane"] = 3064628686] = "ComJane";
  PedHash2[PedHash2["Corpse01"] = 773063444] = "Corpse01";
  PedHash2[PedHash2["Corpse02"] = 228356856] = "Corpse02";
  PedHash2[PedHash2["CounterfeitFemale01"] = 1074385436] = "CounterfeitFemale01";
  PedHash2[PedHash2["CounterfeitMale01"] = 2625926338] = "CounterfeitMale01";
  PedHash2[PedHash2["Cyclist01"] = 755956971] = "Cyclist01";
  PedHash2[PedHash2["DeadHooker"] = 1943971979] = "DeadHooker";
  PedHash2[PedHash2["Drowned"] = 3623056905] = "Drowned";
  PedHash2[PedHash2["ExArmy01"] = 1161072059] = "ExArmy01";
  PedHash2[PedHash2["ExecutivePAMale01"] = 983887149] = "ExecutivePAMale01";
  PedHash2[PedHash2["ExecutivePAFemale01"] = 2913175640] = "ExecutivePAFemale01";
  PedHash2[PedHash2["Famdd01"] = 866411749] = "Famdd01";
  PedHash2[PedHash2["FibArchitect"] = 874722259] = "FibArchitect";
  PedHash2[PedHash2["FibMugger01"] = 2243544680] = "FibMugger01";
  PedHash2[PedHash2["FibSec01"] = 1558115333] = "FibSec01";
  PedHash2[PedHash2["FilmDirector"] = 728636342] = "FilmDirector";
  PedHash2[PedHash2["FilmNoir"] = 732742363] = "FilmNoir";
  PedHash2[PedHash2["Finguru01"] = 1189322339] = "Finguru01";
  PedHash2[PedHash2["ForgeryFemale01"] = 3691903615] = "ForgeryFemale01";
  PedHash2[PedHash2["ForgeryMale01"] = 325317957] = "ForgeryMale01";
  PedHash2[PedHash2["FreemodeFemale01"] = 2627665880] = "FreemodeFemale01";
  PedHash2[PedHash2["FreemodeMale01"] = 1885233650] = "FreemodeMale01";
  PedHash2[PedHash2["Glenstank01"] = 1169888870] = "Glenstank01";
  PedHash2[PedHash2["Griff01"] = 3293887675] = "Griff01";
  PedHash2[PedHash2["Guido01"] = 3333724719] = "Guido01";
  PedHash2[PedHash2["GunVend01"] = 3005388626] = "GunVend01";
  PedHash2[PedHash2["Hacker"] = 2579169528] = "Hacker";
  PedHash2[PedHash2["HeliStaff01"] = 431423238] = "HeliStaff01";
  PedHash2[PedHash2["Hippie01"] = 4030826507] = "Hippie01";
  PedHash2[PedHash2["Hotposh01"] = 2526768638] = "Hotposh01";
  PedHash2[PedHash2["Imporage"] = 880829941] = "Imporage";
  PedHash2[PedHash2["Jesus01"] = 3459037009] = "Jesus01";
  PedHash2[PedHash2["Jewelass01"] = 4040474158] = "Jewelass01";
  PedHash2[PedHash2["JewelSec01"] = 2899099062] = "JewelSec01";
  PedHash2[PedHash2["JewelThief"] = 3872144604] = "JewelThief";
  PedHash2[PedHash2["Justin"] = 2109968527] = "Justin";
  PedHash2[PedHash2["Mani"] = 3367706194] = "Mani";
  PedHash2[PedHash2["Markfost"] = 479578891] = "Markfost";
  PedHash2[PedHash2["Marston01"] = 943915367] = "Marston01";
  PedHash2[PedHash2["MethFemale01"] = 3778572496] = "MethFemale01";
  PedHash2[PedHash2["MethMale01"] = 1293671805] = "MethMale01";
  PedHash2[PedHash2["MilitaryBum"] = 1191548746] = "MilitaryBum";
  PedHash2[PedHash2["Miranda"] = 1095737979] = "Miranda";
  PedHash2[PedHash2["Mistress"] = 1573528872] = "Mistress";
  PedHash2[PedHash2["Misty01"] = 3509125021] = "Misty01";
  PedHash2[PedHash2["MovieStar"] = 894928436] = "MovieStar";
  PedHash2[PedHash2["MPros01"] = 1822283721] = "MPros01";
  PedHash2[PedHash2["Niko01"] = 4007317449] = "Niko01";
  PedHash2[PedHash2["Paparazzi"] = 1346941736] = "Paparazzi";
  PedHash2[PedHash2["Party01"] = 921110016] = "Party01";
  PedHash2[PedHash2["PartyTarget"] = 2180468199] = "PartyTarget";
  PedHash2[PedHash2["PestContDriver"] = 994527967] = "PestContDriver";
  PedHash2[PedHash2["PestContGunman"] = 193469166] = "PestContGunman";
  PedHash2[PedHash2["Pogo01"] = 3696858125] = "Pogo01";
  PedHash2[PedHash2["Poppymich"] = 602513566] = "Poppymich";
  PedHash2[PedHash2["Princess"] = 3538133636] = "Princess";
  PedHash2[PedHash2["Prisoner01"] = 2073775040] = "Prisoner01";
  PedHash2[PedHash2["PrologueHostage01"] = 3306347811] = "PrologueHostage01";
  PedHash2[PedHash2["PrologueMournFemale01"] = 2718472679] = "PrologueMournFemale01";
  PedHash2[PedHash2["PrologueMournMale01"] = 3465937675] = "PrologueMournMale01";
  PedHash2[PedHash2["RivalPaparazzi"] = 1624626906] = "RivalPaparazzi";
  PedHash2[PedHash2["ShopKeep01"] = 416176080] = "ShopKeep01";
  PedHash2[PedHash2["SpyActor"] = 2886641112] = "SpyActor";
  PedHash2[PedHash2["SpyActress"] = 1535236204] = "SpyActress";
  PedHash2[PedHash2["StripperLite"] = 695248020] = "StripperLite";
  PedHash2[PedHash2["Taphillbilly"] = 2585681490] = "Taphillbilly";
  PedHash2[PedHash2["Tramp01"] = 1787764635] = "Tramp01";
  PedHash2[PedHash2["VagosFun01"] = 3299219389] = "VagosFun01";
  PedHash2[PedHash2["WillyFist"] = 2423691919] = "WillyFist";
  PedHash2[PedHash2["WeedFemale01"] = 1596374223] = "WeedFemale01";
  PedHash2[PedHash2["WeedMale01"] = 2648833641] = "WeedMale01";
  PedHash2[PedHash2["Zombie01"] = 2890614022] = "Zombie01";
  PedHash2[PedHash2["Acult01AMM"] = 1413662315] = "Acult01AMM";
  PedHash2[PedHash2["Acult01AMO"] = 1430544400] = "Acult01AMO";
  PedHash2[PedHash2["Acult01AMY"] = 3043264555] = "Acult01AMY";
  PedHash2[PedHash2["Acult02AMO"] = 1268862154] = "Acult02AMO";
  PedHash2[PedHash2["Acult02AMY"] = 2162532142] = "Acult02AMY";
  PedHash2[PedHash2["AfriAmer01AMM"] = 3513928062] = "AfriAmer01AMM";
  PedHash2[PedHash2["Airhostess01SFY"] = 1567728751] = "Airhostess01SFY";
  PedHash2[PedHash2["AirworkerSMY"] = 1644266841] = "AirworkerSMY";
  PedHash2[PedHash2["Ammucity01SMY"] = 2651349821] = "Ammucity01SMY";
  PedHash2[PedHash2["AmmuCountrySMM"] = 233415434] = "AmmuCountrySMM";
  PedHash2[PedHash2["ArmBoss01GMM"] = 4058522530] = "ArmBoss01GMM";
  PedHash2[PedHash2["ArmGoon01GMM"] = 4255728232] = "ArmGoon01GMM";
  PedHash2[PedHash2["ArmGoon02GMY"] = 3310258058] = "ArmGoon02GMY";
  PedHash2[PedHash2["ArmLieut01GMM"] = 3882958867] = "ArmLieut01GMM";
  PedHash2[PedHash2["Armoured01SMM"] = 2512875213] = "Armoured01SMM";
  PedHash2[PedHash2["Armoured02SMM"] = 1669696074] = "Armoured02SMM";
  PedHash2[PedHash2["Armymech01SMY"] = 1657546978] = "Armymech01SMY";
  PedHash2[PedHash2["Autopsy01SMY"] = 2988916046] = "Autopsy01SMY";
  PedHash2[PedHash2["Autoshop01SMM"] = 68070371] = "Autoshop01SMM";
  PedHash2[PedHash2["Autoshop02SMM"] = 4033578141] = "Autoshop02SMM";
  PedHash2[PedHash2["Azteca01GMY"] = 1752208920] = "Azteca01GMY";
  PedHash2[PedHash2["BallaEast01GMY"] = 4096714883] = "BallaEast01GMY";
  PedHash2[PedHash2["BallaOrig01GMY"] = 588969535] = "BallaOrig01GMY";
  PedHash2[PedHash2["Ballas01GFY"] = 361513884] = "Ballas01GFY";
  PedHash2[PedHash2["BallaSout01GMY"] = 599294057] = "BallaSout01GMY";
  PedHash2[PedHash2["Barman01SMY"] = 3852538118] = "Barman01SMY";
  PedHash2[PedHash2["Bartender01SFY"] = 2014052797] = "Bartender01SFY";
  PedHash2[PedHash2["Baywatch01SFY"] = 1250841910] = "Baywatch01SFY";
  PedHash2[PedHash2["Baywatch01SMY"] = 189425762] = "Baywatch01SMY";
  PedHash2[PedHash2["Beach01AFM"] = 808859815] = "Beach01AFM";
  PedHash2[PedHash2["Beach01AFY"] = 3349113128] = "Beach01AFY";
  PedHash2[PedHash2["Beach01AMM"] = 1077785853] = "Beach01AMM";
  PedHash2[PedHash2["Beach01AMO"] = 2217202584] = "Beach01AMO";
  PedHash2[PedHash2["Beach01AMY"] = 3523131524] = "Beach01AMY";
  PedHash2[PedHash2["Beach02AMM"] = 2021631368] = "Beach02AMM";
  PedHash2[PedHash2["Beach02AMY"] = 600300561] = "Beach02AMY";
  PedHash2[PedHash2["Beach03AMY"] = 3886638041] = "Beach03AMY";
  PedHash2[PedHash2["Beachvesp01AMY"] = 2114544056] = "Beachvesp01AMY";
  PedHash2[PedHash2["Beachvesp02AMY"] = 3394697810] = "Beachvesp02AMY";
  PedHash2[PedHash2["Bevhills01AFM"] = 3188223741] = "Bevhills01AFM";
  PedHash2[PedHash2["Bevhills01AFY"] = 1146800212] = "Bevhills01AFY";
  PedHash2[PedHash2["Bevhills01AMM"] = 1423699487] = "Bevhills01AMM";
  PedHash2[PedHash2["Bevhills01AMY"] = 1982350912] = "Bevhills01AMY";
  PedHash2[PedHash2["Bevhills02AFM"] = 2688103263] = "Bevhills02AFM";
  PedHash2[PedHash2["Bevhills02AFY"] = 1546450936] = "Bevhills02AFY";
  PedHash2[PedHash2["Bevhills02AMM"] = 1068876755] = "Bevhills02AMM";
  PedHash2[PedHash2["Bevhills02AMY"] = 1720428295] = "Bevhills02AMY";
  PedHash2[PedHash2["Bevhills03AFY"] = 549978415] = "Bevhills03AFY";
  PedHash2[PedHash2["Bevhills04AFY"] = 920595805] = "Bevhills04AFY";
  PedHash2[PedHash2["Blackops01SMY"] = 3019107892] = "Blackops01SMY";
  PedHash2[PedHash2["Blackops02SMY"] = 2047212121] = "Blackops02SMY";
  PedHash2[PedHash2["Blackops03SMY"] = 1349953339] = "Blackops03SMY";
  PedHash2[PedHash2["Bodybuild01AFM"] = 1004114196] = "Bodybuild01AFM";
  PedHash2[PedHash2["Bouncer01SMM"] = 2681481517] = "Bouncer01SMM";
  PedHash2[PedHash2["Breakdance01AMY"] = 933205398] = "Breakdance01AMY";
  PedHash2[PedHash2["Busboy01SMY"] = 3640249671] = "Busboy01SMY";
  PedHash2[PedHash2["Busicas01AMY"] = 2597531625] = "Busicas01AMY";
  PedHash2[PedHash2["Business01AFY"] = 664399832] = "Business01AFY";
  PedHash2[PedHash2["Business01AMM"] = 2120901815] = "Business01AMM";
  PedHash2[PedHash2["Business01AMY"] = 3382649284] = "Business01AMY";
  PedHash2[PedHash2["Business02AFM"] = 532905404] = "Business02AFM";
  PedHash2[PedHash2["Business02AFY"] = 826475330] = "Business02AFY";
  PedHash2[PedHash2["Business02AMY"] = 3014915558] = "Business02AMY";
  PedHash2[PedHash2["Business03AFY"] = 2928082356] = "Business03AFY";
  PedHash2[PedHash2["Business03AMY"] = 2705543429] = "Business03AMY";
  PedHash2[PedHash2["Business04AFY"] = 3083210802] = "Business04AFY";
  PedHash2[PedHash2["Busker01SMO"] = 2912874939] = "Busker01SMO";
  PedHash2[PedHash2["CCrew01SMM"] = 3387290987] = "CCrew01SMM";
  PedHash2[PedHash2["Chef01SMY"] = 261586155] = "Chef01SMY";
  PedHash2[PedHash2["ChemSec01SMM"] = 788443093] = "ChemSec01SMM";
  PedHash2[PedHash2["ChemWork01GMM"] = 4128603535] = "ChemWork01GMM";
  PedHash2[PedHash2["ChiBoss01GMM"] = 3118269184] = "ChiBoss01GMM";
  PedHash2[PedHash2["ChiCold01GMM"] = 275618457] = "ChiCold01GMM";
  PedHash2[PedHash2["ChiGoon01GMM"] = 2119136831] = "ChiGoon01GMM";
  PedHash2[PedHash2["ChiGoon02GMM"] = 4285659174] = "ChiGoon02GMM";
  PedHash2[PedHash2["CiaSec01SMM"] = 1650288984] = "CiaSec01SMM";
  PedHash2[PedHash2["Clown01SMY"] = 71929310] = "Clown01SMY";
  PedHash2[PedHash2["Cntrybar01SMM"] = 436345731] = "Cntrybar01SMM";
  PedHash2[PedHash2["Construct01SMY"] = 3621428889] = "Construct01SMY";
  PedHash2[PedHash2["Construct02SMY"] = 3321821918] = "Construct02SMY";
  PedHash2[PedHash2["Cop01SFY"] = 368603149] = "Cop01SFY";
  PedHash2[PedHash2["Cop01SMY"] = 1581098148] = "Cop01SMY";
  PedHash2[PedHash2["Cyclist01AMY"] = 4257633223] = "Cyclist01AMY";
  PedHash2[PedHash2["Dealer01SMY"] = 3835149295] = "Dealer01SMY";
  PedHash2[PedHash2["Devinsec01SMY"] = 2606068340] = "Devinsec01SMY";
  PedHash2[PedHash2["Dhill01AMY"] = 4282288299] = "Dhill01AMY";
  PedHash2[PedHash2["Dockwork01SMM"] = 349680864] = "Dockwork01SMM";
  PedHash2[PedHash2["Dockwork01SMY"] = 2255894993] = "Dockwork01SMY";
  PedHash2[PedHash2["Doctor01SMM"] = 3564307372] = "Doctor01SMM";
  PedHash2[PedHash2["Doorman01SMY"] = 579932932] = "Doorman01SMY";
  PedHash2[PedHash2["Downtown01AFM"] = 1699403886] = "Downtown01AFM";
  PedHash2[PedHash2["Downtown01AMY"] = 766375082] = "Downtown01AMY";
  PedHash2[PedHash2["DwService01SMY"] = 1976765073] = "DwService01SMY";
  PedHash2[PedHash2["DwService02SMY"] = 4119890438] = "DwService02SMY";
  PedHash2[PedHash2["Eastsa01AFM"] = 2638072698] = "Eastsa01AFM";
  PedHash2[PedHash2["Eastsa01AFY"] = 4121954205] = "Eastsa01AFY";
  PedHash2[PedHash2["Eastsa01AMM"] = 4188468543] = "Eastsa01AMM";
  PedHash2[PedHash2["Eastsa01AMY"] = 2756120947] = "Eastsa01AMY";
  PedHash2[PedHash2["Eastsa02AFM"] = 1674107025] = "Eastsa02AFM";
  PedHash2[PedHash2["Eastsa02AFY"] = 70821038] = "Eastsa02AFY";
  PedHash2[PedHash2["Eastsa02AMM"] = 131961260] = "Eastsa02AMM";
  PedHash2[PedHash2["Eastsa02AMY"] = 377976310] = "Eastsa02AMY";
  PedHash2[PedHash2["Eastsa03AFY"] = 1371553700] = "Eastsa03AFY";
  PedHash2[PedHash2["Epsilon01AFY"] = 1755064960] = "Epsilon01AFY";
  PedHash2[PedHash2["Epsilon01AMY"] = 2010389054] = "Epsilon01AMY";
  PedHash2[PedHash2["Epsilon02AMY"] = 2860711835] = "Epsilon02AMY";
  PedHash2[PedHash2["Factory01SFY"] = 1777626099] = "Factory01SFY";
  PedHash2[PedHash2["Factory01SMY"] = 1097048408] = "Factory01SMY";
  PedHash2[PedHash2["Famca01GMY"] = 3896218551] = "Famca01GMY";
  PedHash2[PedHash2["Famdnf01GMY"] = 3681718840] = "Famdnf01GMY";
  PedHash2[PedHash2["Famfor01GMY"] = 2217749257] = "Famfor01GMY";
  PedHash2[PedHash2["Families01GFY"] = 1309468115] = "Families01GFY";
  PedHash2[PedHash2["Farmer01AMM"] = 2488675799] = "Farmer01AMM";
  PedHash2[PedHash2["FatBla01AFM"] = 4206136267] = "FatBla01AFM";
  PedHash2[PedHash2["FatCult01AFM"] = 3050275044] = "FatCult01AFM";
  PedHash2[PedHash2["Fatlatin01AMM"] = 1641152947] = "Fatlatin01AMM";
  PedHash2[PedHash2["FatWhite01AFM"] = 951767867] = "FatWhite01AFM";
  PedHash2[PedHash2["FemBarberSFM"] = 373000027] = "FemBarberSFM";
  PedHash2[PedHash2["FibOffice01SMM"] = 3988550982] = "FibOffice01SMM";
  PedHash2[PedHash2["FibOffice02SMM"] = 653289389] = "FibOffice02SMM";
  PedHash2[PedHash2["FibSec01SMM"] = 2072724299] = "FibSec01SMM";
  PedHash2[PedHash2["Fireman01SMY"] = 3065114024] = "Fireman01SMY";
  PedHash2[PedHash2["Fitness01AFY"] = 1165780219] = "Fitness01AFY";
  PedHash2[PedHash2["Fitness02AFY"] = 331645324] = "Fitness02AFY";
  PedHash2[PedHash2["Gaffer01SMM"] = 2841034142] = "Gaffer01SMM";
  PedHash2[PedHash2["GarbageSMY"] = 4000686095] = "GarbageSMY";
  PedHash2[PedHash2["Gardener01SMM"] = 1240094341] = "Gardener01SMM";
  PedHash2[PedHash2["Gay01AMY"] = 3519864886] = "Gay01AMY";
  PedHash2[PedHash2["Gay02AMY"] = 2775713665] = "Gay02AMY";
  PedHash2[PedHash2["Genfat01AMM"] = 115168927] = "Genfat01AMM";
  PedHash2[PedHash2["Genfat02AMM"] = 330231874] = "Genfat02AMM";
  PedHash2[PedHash2["Genhot01AFY"] = 793439294] = "Genhot01AFY";
  PedHash2[PedHash2["Genstreet01AFO"] = 1640504453] = "Genstreet01AFO";
  PedHash2[PedHash2["Genstreet01AMO"] = 2908022696] = "Genstreet01AMO";
  PedHash2[PedHash2["Genstreet01AMY"] = 2557996913] = "Genstreet01AMY";
  PedHash2[PedHash2["Genstreet02AMY"] = 891398354] = "Genstreet02AMY";
  PedHash2[PedHash2["GentransportSMM"] = 411102470] = "GentransportSMM";
  PedHash2[PedHash2["Golfer01AFY"] = 2111372120] = "Golfer01AFY";
  PedHash2[PedHash2["Golfer01AMM"] = 2850754114] = "Golfer01AMM";
  PedHash2[PedHash2["Golfer01AMY"] = 3609190705] = "Golfer01AMY";
  PedHash2[PedHash2["Grip01SMY"] = 815693290] = "Grip01SMY";
  PedHash2[PedHash2["Hairdress01SMM"] = 1099825042] = "Hairdress01SMM";
  PedHash2[PedHash2["Hasjew01AMM"] = 1809430156] = "Hasjew01AMM";
  PedHash2[PedHash2["Hasjew01AMY"] = 3782053633] = "Hasjew01AMY";
  PedHash2[PedHash2["Highsec01SMM"] = 4049719826] = "Highsec01SMM";
  PedHash2[PedHash2["Highsec02SMM"] = 691061163] = "Highsec02SMM";
  PedHash2[PedHash2["Hiker01AFY"] = 813893651] = "Hiker01AFY";
  PedHash2[PedHash2["Hiker01AMY"] = 1358380044] = "Hiker01AMY";
  PedHash2[PedHash2["Hillbilly01AMM"] = 1822107721] = "Hillbilly01AMM";
  PedHash2[PedHash2["Hillbilly02AMM"] = 2064532783] = "Hillbilly02AMM";
  PedHash2[PedHash2["Hippie01AFY"] = 343259175] = "Hippie01AFY";
  PedHash2[PedHash2["Hippy01AMY"] = 2097407511] = "Hippy01AMY";
  PedHash2[PedHash2["Hipster01AFY"] = 2185745201] = "Hipster01AFY";
  PedHash2[PedHash2["Hipster01AMY"] = 587703123] = "Hipster01AMY";
  PedHash2[PedHash2["Hipster02AFY"] = 2549481101] = "Hipster02AFY";
  PedHash2[PedHash2["Hipster02AMY"] = 349505262] = "Hipster02AMY";
  PedHash2[PedHash2["Hipster03AFY"] = 2780469782] = "Hipster03AFY";
  PedHash2[PedHash2["Hipster03AMY"] = 1312913862] = "Hipster03AMY";
  PedHash2[PedHash2["Hipster04AFY"] = 429425116] = "Hipster04AFY";
  PedHash2[PedHash2["Hooker01SFY"] = 42647445] = "Hooker01SFY";
  PedHash2[PedHash2["Hooker02SFY"] = 348382215] = "Hooker02SFY";
  PedHash2[PedHash2["Hooker03SFY"] = 51789996] = "Hooker03SFY";
  PedHash2[PedHash2["Hwaycop01SMY"] = 1939545845] = "Hwaycop01SMY";
  PedHash2[PedHash2["Indian01AFO"] = 3134700416] = "Indian01AFO";
  PedHash2[PedHash2["Indian01AFY"] = 153984193] = "Indian01AFY";
  PedHash2[PedHash2["Indian01AMM"] = 3721046572] = "Indian01AMM";
  PedHash2[PedHash2["Indian01AMY"] = 706935758] = "Indian01AMY";
  PedHash2[PedHash2["JanitorSMM"] = 2842417644] = "JanitorSMM";
  PedHash2[PedHash2["Jetski01AMY"] = 767028979] = "Jetski01AMY";
  PedHash2[PedHash2["Juggalo01AFY"] = 3675473203] = "Juggalo01AFY";
  PedHash2[PedHash2["Juggalo01AMY"] = 2445950508] = "Juggalo01AMY";
  PedHash2[PedHash2["KorBoss01GMM"] = 891945583] = "KorBoss01GMM";
  PedHash2[PedHash2["Korean01GMY"] = 611648169] = "Korean01GMY";
  PedHash2[PedHash2["Korean02GMY"] = 2414729609] = "Korean02GMY";
  PedHash2[PedHash2["KorLieut01GMY"] = 2093736314] = "KorLieut01GMY";
  PedHash2[PedHash2["Ktown01AFM"] = 1388848350] = "Ktown01AFM";
  PedHash2[PedHash2["Ktown01AFO"] = 1204772502] = "Ktown01AFO";
  PedHash2[PedHash2["Ktown01AMM"] = 3512565361] = "Ktown01AMM";
  PedHash2[PedHash2["Ktown01AMO"] = 355916122] = "Ktown01AMO";
  PedHash2[PedHash2["Ktown01AMY"] = 452351020] = "Ktown01AMY";
  PedHash2[PedHash2["Ktown02AFM"] = 1090617681] = "Ktown02AFM";
  PedHash2[PedHash2["Ktown02AMY"] = 696250687] = "Ktown02AMY";
  PedHash2[PedHash2["Lathandy01SMM"] = 2659242702] = "Lathandy01SMM";
  PedHash2[PedHash2["Latino01AMY"] = 321657486] = "Latino01AMY";
  PedHash2[PedHash2["Lifeinvad01SMM"] = 3724572669] = "Lifeinvad01SMM";
  PedHash2[PedHash2["LinecookSMM"] = 3684436375] = "LinecookSMM";
  PedHash2[PedHash2["Lost01GFY"] = 4250220510] = "Lost01GFY";
  PedHash2[PedHash2["Lost01GMY"] = 1330042375] = "Lost01GMY";
  PedHash2[PedHash2["Lost02GMY"] = 1032073858] = "Lost02GMY";
  PedHash2[PedHash2["Lost03GMY"] = 850468060] = "Lost03GMY";
  PedHash2[PedHash2["Lsmetro01SMM"] = 1985653476] = "Lsmetro01SMM";
  PedHash2[PedHash2["Maid01SFM"] = 3767780806] = "Maid01SFM";
  PedHash2[PedHash2["Malibu01AMM"] = 803106487] = "Malibu01AMM";
  PedHash2[PedHash2["Mariachi01SMM"] = 2124742566] = "Mariachi01SMM";
  PedHash2[PedHash2["Marine01SMM"] = 4074414829] = "Marine01SMM";
  PedHash2[PedHash2["Marine01SMY"] = 1702441027] = "Marine01SMY";
  PedHash2[PedHash2["Marine02SMM"] = 4028996995] = "Marine02SMM";
  PedHash2[PedHash2["Marine02SMY"] = 1490458366] = "Marine02SMY";
  PedHash2[PedHash2["Marine03SMY"] = 1925237458] = "Marine03SMY";
  PedHash2[PedHash2["Methhead01AMY"] = 1768677545] = "Methhead01AMY";
  PedHash2[PedHash2["MexBoss01GMM"] = 1466037421] = "MexBoss01GMM";
  PedHash2[PedHash2["MexBoss02GMM"] = 1226102803] = "MexBoss02GMM";
  PedHash2[PedHash2["MexCntry01AMM"] = 3716251309] = "MexCntry01AMM";
  PedHash2[PedHash2["MexGang01GMY"] = 3185399110] = "MexGang01GMY";
  PedHash2[PedHash2["MexGoon01GMY"] = 653210662] = "MexGoon01GMY";
  PedHash2[PedHash2["MexGoon02GMY"] = 832784782] = "MexGoon02GMY";
  PedHash2[PedHash2["MexGoon03GMY"] = 2521633500] = "MexGoon03GMY";
  PedHash2[PedHash2["MexLabor01AMM"] = 2992445106] = "MexLabor01AMM";
  PedHash2[PedHash2["MexThug01AMY"] = 810804565] = "MexThug01AMY";
  PedHash2[PedHash2["Migrant01SFY"] = 3579522037] = "Migrant01SFY";
  PedHash2[PedHash2["Migrant01SMM"] = 3977045190] = "Migrant01SMM";
  PedHash2[PedHash2["MimeSMY"] = 1021093698] = "MimeSMY";
  PedHash2[PedHash2["Motox01AMY"] = 1694362237] = "Motox01AMY";
  PedHash2[PedHash2["Motox02AMY"] = 2007797722] = "Motox02AMY";
  PedHash2[PedHash2["MovAlien01"] = 1684083350] = "MovAlien01";
  PedHash2[PedHash2["MovPrem01SFY"] = 587253782] = "MovPrem01SFY";
  PedHash2[PedHash2["Movprem01SMM"] = 3630066984] = "Movprem01SMM";
  PedHash2[PedHash2["Movspace01SMM"] = 3887273010] = "Movspace01SMM";
  PedHash2[PedHash2["Musclbeac01AMY"] = 1264920838] = "Musclbeac01AMY";
  PedHash2[PedHash2["Musclbeac02AMY"] = 3374523516] = "Musclbeac02AMY";
  PedHash2[PedHash2["OgBoss01AMM"] = 1746653202] = "OgBoss01AMM";
  PedHash2[PedHash2["Paparazzi01AMM"] = 3972697109] = "Paparazzi01AMM";
  PedHash2[PedHash2["Paramedic01SMM"] = 3008586398] = "Paramedic01SMM";
  PedHash2[PedHash2["PestCont01SMY"] = 1209091352] = "PestCont01SMY";
  PedHash2[PedHash2["Pilot01SMM"] = 3881519900] = "Pilot01SMM";
  PedHash2[PedHash2["Pilot01SMY"] = 2872052743] = "Pilot01SMY";
  PedHash2[PedHash2["Pilot02SMM"] = 4131252449] = "Pilot02SMM";
  PedHash2[PedHash2["PoloGoon01GMY"] = 1329576454] = "PoloGoon01GMY";
  PedHash2[PedHash2["PoloGoon02GMY"] = 2733138262] = "PoloGoon02GMY";
  PedHash2[PedHash2["Polynesian01AMM"] = 2849617566] = "Polynesian01AMM";
  PedHash2[PedHash2["Polynesian01AMY"] = 2206530719] = "Polynesian01AMY";
  PedHash2[PedHash2["Postal01SMM"] = 1650036788] = "Postal01SMM";
  PedHash2[PedHash2["Postal02SMM"] = 1936142927] = "Postal02SMM";
  PedHash2[PedHash2["Prisguard01SMM"] = 1456041926] = "Prisguard01SMM";
  PedHash2[PedHash2["PrisMuscl01SMY"] = 1596003233] = "PrisMuscl01SMY";
  PedHash2[PedHash2["Prisoner01SMY"] = 2981862233] = "Prisoner01SMY";
  PedHash2[PedHash2["PrologueHostage01AFM"] = 379310561] = "PrologueHostage01AFM";
  PedHash2[PedHash2["PrologueHostage01AMM"] = 2534589327] = "PrologueHostage01AMM";
  PedHash2[PedHash2["Ranger01SFY"] = 2680682039] = "Ranger01SFY";
  PedHash2[PedHash2["Ranger01SMY"] = 4017173934] = "Ranger01SMY";
  PedHash2[PedHash2["Roadcyc01AMY"] = 4116817094] = "Roadcyc01AMY";
  PedHash2[PedHash2["Robber01SMY"] = 3227390873] = "Robber01SMY";
  PedHash2[PedHash2["RsRanger01AMO"] = 1011059922] = "RsRanger01AMO";
  PedHash2[PedHash2["Runner01AFY"] = 3343476521] = "Runner01AFY";
  PedHash2[PedHash2["Runner01AMY"] = 623927022] = "Runner01AMY";
  PedHash2[PedHash2["Runner02AMY"] = 2218630415] = "Runner02AMY";
  PedHash2[PedHash2["Rurmeth01AFY"] = 1064866854] = "Rurmeth01AFY";
  PedHash2[PedHash2["Rurmeth01AMM"] = 1001210244] = "Rurmeth01AMM";
  PedHash2[PedHash2["Salton01AFM"] = 3725461865] = "Salton01AFM";
  PedHash2[PedHash2["Salton01AFO"] = 3439295882] = "Salton01AFO";
  PedHash2[PedHash2["Salton01AMM"] = 1328415626] = "Salton01AMM";
  PedHash2[PedHash2["Salton01AMO"] = 539004493] = "Salton01AMO";
  PedHash2[PedHash2["Salton01AMY"] = 3613420592] = "Salton01AMY";
  PedHash2[PedHash2["Salton02AMM"] = 1626646295] = "Salton02AMM";
  PedHash2[PedHash2["Salton03AMM"] = 2995538501] = "Salton03AMM";
  PedHash2[PedHash2["Salton04AMM"] = 2521108919] = "Salton04AMM";
  PedHash2[PedHash2["SalvaBoss01GMY"] = 2422005962] = "SalvaBoss01GMY";
  PedHash2[PedHash2["SalvaGoon01GMY"] = 663522487] = "SalvaGoon01GMY";
  PedHash2[PedHash2["SalvaGoon02GMY"] = 846439045] = "SalvaGoon02GMY";
  PedHash2[PedHash2["SalvaGoon03GMY"] = 62440720] = "SalvaGoon03GMY";
  PedHash2[PedHash2["SbikeAMO"] = 1794381917] = "SbikeAMO";
  PedHash2[PedHash2["Scdressy01AFY"] = 3680420864] = "Scdressy01AFY";
  PedHash2[PedHash2["Scientist01SMM"] = 1092080539] = "Scientist01SMM";
  PedHash2[PedHash2["Scrubs01SFY"] = 2874755766] = "Scrubs01SFY";
  PedHash2[PedHash2["Security01SMM"] = 3613962792] = "Security01SMM";
  PedHash2[PedHash2["Sheriff01SFY"] = 1096929346] = "Sheriff01SFY";
  PedHash2[PedHash2["Sheriff01SMY"] = 2974087609] = "Sheriff01SMY";
  PedHash2[PedHash2["ShopHighSFM"] = 2923947184] = "ShopHighSFM";
  PedHash2[PedHash2["ShopLowSFY"] = 2842568196] = "ShopLowSFY";
  PedHash2[PedHash2["ShopMaskSMY"] = 1846684678] = "ShopMaskSMY";
  PedHash2[PedHash2["ShopMidSFY"] = 1055701597] = "ShopMidSFY";
  PedHash2[PedHash2["Skater01AFY"] = 1767892582] = "Skater01AFY";
  PedHash2[PedHash2["Skater01AMM"] = 3654768780] = "Skater01AMM";
  PedHash2[PedHash2["Skater01AMY"] = 3250873975] = "Skater01AMY";
  PedHash2[PedHash2["Skater02AMY"] = 2952446692] = "Skater02AMY";
  PedHash2[PedHash2["Skidrow01AFM"] = 2962707003] = "Skidrow01AFM";
  PedHash2[PedHash2["Skidrow01AMM"] = 32417469] = "Skidrow01AMM";
  PedHash2[PedHash2["Snowcop01SMM"] = 451459928] = "Snowcop01SMM";
  PedHash2[PedHash2["Socenlat01AMM"] = 193817059] = "Socenlat01AMM";
  PedHash2[PedHash2["Soucent01AFM"] = 1951946145] = "Soucent01AFM";
  PedHash2[PedHash2["Soucent01AFO"] = 1039800368] = "Soucent01AFO";
  PedHash2[PedHash2["Soucent01AFY"] = 744758650] = "Soucent01AFY";
  PedHash2[PedHash2["Soucent01AMM"] = 1750583735] = "Soucent01AMM";
  PedHash2[PedHash2["Soucent01AMO"] = 718836251] = "Soucent01AMO";
  PedHash2[PedHash2["Soucent01AMY"] = 3877027275] = "Soucent01AMY";
  PedHash2[PedHash2["Soucent02AFM"] = 4079145784] = "Soucent02AFM";
  PedHash2[PedHash2["Soucent02AFO"] = 2775443222] = "Soucent02AFO";
  PedHash2[PedHash2["Soucent02AFY"] = 1519319503] = "Soucent02AFY";
  PedHash2[PedHash2["Soucent02AMM"] = 2674735073] = "Soucent02AMM";
  PedHash2[PedHash2["Soucent02AMO"] = 1082572151] = "Soucent02AMO";
  PedHash2[PedHash2["Soucent02AMY"] = 2896414922] = "Soucent02AMY";
  PedHash2[PedHash2["Soucent03AFY"] = 2276611093] = "Soucent03AFY";
  PedHash2[PedHash2["Soucent03AMM"] = 2346291386] = "Soucent03AMM";
  PedHash2[PedHash2["Soucent03AMO"] = 238213328] = "Soucent03AMO";
  PedHash2[PedHash2["Soucent03AMY"] = 3287349092] = "Soucent03AMY";
  PedHash2[PedHash2["Soucent04AMM"] = 3271294718] = "Soucent04AMM";
  PedHash2[PedHash2["Soucent04AMY"] = 2318861297] = "Soucent04AMY";
  PedHash2[PedHash2["Soucentmc01AFM"] = 3454621138] = "Soucentmc01AFM";
  PedHash2[PedHash2["Staggrm01AMO"] = 2442448387] = "Staggrm01AMO";
  PedHash2[PedHash2["Stbla01AMY"] = 3482496489] = "Stbla01AMY";
  PedHash2[PedHash2["Stbla02AMY"] = 2563194959] = "Stbla02AMY";
  PedHash2[PedHash2["Stlat01AMY"] = 2255803900] = "Stlat01AMY";
  PedHash2[PedHash2["Stlat02AMM"] = 3265820418] = "Stlat02AMM";
  PedHash2[PedHash2["Stripper01SFY"] = 1381498905] = "Stripper01SFY";
  PedHash2[PedHash2["Stripper02SFY"] = 1846523796] = "Stripper02SFY";
  PedHash2[PedHash2["StripperLiteSFY"] = 1544875514] = "StripperLiteSFY";
  PedHash2[PedHash2["Strperf01SMM"] = 2035992488] = "Strperf01SMM";
  PedHash2[PedHash2["Strpreach01SMM"] = 469792763] = "Strpreach01SMM";
  PedHash2[PedHash2["StrPunk01GMY"] = 4246489531] = "StrPunk01GMY";
  PedHash2[PedHash2["StrPunk02GMY"] = 228715206] = "StrPunk02GMY";
  PedHash2[PedHash2["Strvend01SMM"] = 3465614249] = "Strvend01SMM";
  PedHash2[PedHash2["Strvend01SMY"] = 2457805603] = "Strvend01SMY";
  PedHash2[PedHash2["Stwhi01AMY"] = 605602864] = "Stwhi01AMY";
  PedHash2[PedHash2["Stwhi02AMY"] = 919005580] = "Stwhi02AMY";
  PedHash2[PedHash2["Sunbathe01AMY"] = 3072929548] = "Sunbathe01AMY";
  PedHash2[PedHash2["Surfer01AMY"] = 3938633710] = "Surfer01AMY";
  PedHash2[PedHash2["Swat01SMY"] = 2374966032] = "Swat01SMY";
  PedHash2[PedHash2["Sweatshop01SFM"] = 824925120] = "Sweatshop01SFM";
  PedHash2[PedHash2["Sweatshop01SFY"] = 2231547570] = "Sweatshop01SFY";
  PedHash2[PedHash2["Tattoo01AMO"] = 2494442380] = "Tattoo01AMO";
  PedHash2[PedHash2["Tennis01AFY"] = 1426880966] = "Tennis01AFY";
  PedHash2[PedHash2["Tennis01AMM"] = 1416254276] = "Tennis01AMM";
  PedHash2[PedHash2["Topless01AFY"] = 2633130371] = "Topless01AFY";
  PedHash2[PedHash2["Tourist01AFM"] = 1347814329] = "Tourist01AFM";
  PedHash2[PedHash2["Tourist01AFY"] = 1446741360] = "Tourist01AFY";
  PedHash2[PedHash2["Tourist01AMM"] = 3365863812] = "Tourist01AMM";
  PedHash2[PedHash2["Tourist02AFY"] = 2435054400] = "Tourist02AFY";
  PedHash2[PedHash2["Tramp01AFM"] = 1224306523] = "Tramp01AFM";
  PedHash2[PedHash2["Tramp01AMM"] = 516505552] = "Tramp01AMM";
  PedHash2[PedHash2["Tramp01AMO"] = 390939205] = "Tramp01AMO";
  PedHash2[PedHash2["TrampBeac01AFM"] = 2359345766] = "TrampBeac01AFM";
  PedHash2[PedHash2["TrampBeac01AMM"] = 1404403376] = "TrampBeac01AMM";
  PedHash2[PedHash2["Tranvest01AMM"] = 3773208948] = "Tranvest01AMM";
  PedHash2[PedHash2["Tranvest02AMM"] = 4144940484] = "Tranvest02AMM";
  PedHash2[PedHash2["Trucker01SMM"] = 1498487404] = "Trucker01SMM";
  PedHash2[PedHash2["Ups01SMM"] = 2680389410] = "Ups01SMM";
  PedHash2[PedHash2["Ups02SMM"] = 3502104854] = "Ups02SMM";
  PedHash2[PedHash2["Uscg01SMY"] = 3389018345] = "Uscg01SMY";
  PedHash2[PedHash2["Vagos01GFY"] = 1520708641] = "Vagos01GFY";
  PedHash2[PedHash2["Valet01SMY"] = 999748158] = "Valet01SMY";
  PedHash2[PedHash2["Vindouche01AMY"] = 3247667175] = "Vindouche01AMY";
  PedHash2[PedHash2["Vinewood01AFY"] = 435429221] = "Vinewood01AFY";
  PedHash2[PedHash2["Vinewood01AMY"] = 1264851357] = "Vinewood01AMY";
  PedHash2[PedHash2["Vinewood02AFY"] = 3669401835] = "Vinewood02AFY";
  PedHash2[PedHash2["Vinewood02AMY"] = 1561705728] = "Vinewood02AMY";
  PedHash2[PedHash2["Vinewood03AFY"] = 933092024] = "Vinewood03AFY";
  PedHash2[PedHash2["Vinewood03AMY"] = 534725268] = "Vinewood03AMY";
  PedHash2[PedHash2["Vinewood04AFY"] = 4209271110] = "Vinewood04AFY";
  PedHash2[PedHash2["Vinewood04AMY"] = 835315305] = "Vinewood04AMY";
  PedHash2[PedHash2["Waiter01SMY"] = 2907468364] = "Waiter01SMY";
  PedHash2[PedHash2["WinClean01SMY"] = 1426951581] = "WinClean01SMY";
  PedHash2[PedHash2["Xmech01SMY"] = 1142162924] = "Xmech01SMY";
  PedHash2[PedHash2["Xmech02SMY"] = 3189832196] = "Xmech02SMY";
  PedHash2[PedHash2["Xmech02SMYMP"] = 1755203590] = "Xmech02SMYMP";
  PedHash2[PedHash2["Yoga01AFY"] = 3290105390] = "Yoga01AFY";
  PedHash2[PedHash2["Yoga01AMY"] = 2869588309] = "Yoga01AMY";
  return PedHash2;
})(PedHash || {});
export {
  PedHash
};
