// @nativewrappers/fivem/hashes/VehicleHash.js
var VehicleHash = /* @__PURE__ */ ((VehicleHash2) => {
  VehicleHash2[VehicleHash2["Adder"] = 3078201489] = "Adder";
  VehicleHash2[VehicleHash2["Airbus"] = 1283517198] = "Airbus";
  VehicleHash2[VehicleHash2["Airtug"] = 1560980623] = "Airtug";
  VehicleHash2[VehicleHash2["Akuma"] = 1672195559] = "Akuma";
  VehicleHash2[VehicleHash2["Alpha"] = 767087018] = "Alpha";
  VehicleHash2[VehicleHash2["Ambulance"] = 1171614426] = "Ambulance";
  VehicleHash2[VehicleHash2["Annihilator"] = 837858166] = "Annihilator";
  VehicleHash2[VehicleHash2["Apc"] = 562680400] = "Apc";
  VehicleHash2[VehicleHash2["Ardent"] = 159274291] = "Ardent";
  VehicleHash2[VehicleHash2["ArmyTanker"] = 3087536137] = "ArmyTanker";
  VehicleHash2[VehicleHash2["ArmyTrailer"] = 2818520053] = "ArmyTrailer";
  VehicleHash2[VehicleHash2["ArmyTrailer2"] = 2657817814] = "ArmyTrailer2";
  VehicleHash2[VehicleHash2["Asea"] = 2485144969] = "Asea";
  VehicleHash2[VehicleHash2["Asea2"] = 2487343317] = "Asea2";
  VehicleHash2[VehicleHash2["Asterope"] = 2391954683] = "Asterope";
  VehicleHash2[VehicleHash2["Avarus"] = 2179174271] = "Avarus";
  VehicleHash2[VehicleHash2["Bagger"] = 2154536131] = "Bagger";
  VehicleHash2[VehicleHash2["BaleTrailer"] = 3895125590] = "BaleTrailer";
  VehicleHash2[VehicleHash2["Baller"] = 3486135912] = "Baller";
  VehicleHash2[VehicleHash2["Baller2"] = 142944341] = "Baller2";
  VehicleHash2[VehicleHash2["Baller3"] = 1878062887] = "Baller3";
  VehicleHash2[VehicleHash2["Baller4"] = 634118882] = "Baller4";
  VehicleHash2[VehicleHash2["Baller5"] = 470404958] = "Baller5";
  VehicleHash2[VehicleHash2["Baller6"] = 666166960] = "Baller6";
  VehicleHash2[VehicleHash2["Banshee"] = 3253274834] = "Banshee";
  VehicleHash2[VehicleHash2["Banshee2"] = 633712403] = "Banshee2";
  VehicleHash2[VehicleHash2["Barracks"] = 3471458123] = "Barracks";
  VehicleHash2[VehicleHash2["Barracks2"] = 1074326203] = "Barracks2";
  VehicleHash2[VehicleHash2["Barracks3"] = 630371791] = "Barracks3";
  VehicleHash2[VehicleHash2["Bati"] = 4180675781] = "Bati";
  VehicleHash2[VehicleHash2["Bati2"] = 3403504941] = "Bati2";
  VehicleHash2[VehicleHash2["Benson"] = 2053223216] = "Benson";
  VehicleHash2[VehicleHash2["Besra"] = 1824333165] = "Besra";
  VehicleHash2[VehicleHash2["BestiaGTS"] = 1274868363] = "BestiaGTS";
  VehicleHash2[VehicleHash2["BF400"] = 86520421] = "BF400";
  VehicleHash2[VehicleHash2["BfInjection"] = 1126868326] = "BfInjection";
  VehicleHash2[VehicleHash2["Biff"] = 850991848] = "Biff";
  VehicleHash2[VehicleHash2["Bifta"] = 3945366167] = "Bifta";
  VehicleHash2[VehicleHash2["Bison"] = 4278019151] = "Bison";
  VehicleHash2[VehicleHash2["Bison2"] = 2072156101] = "Bison2";
  VehicleHash2[VehicleHash2["Bison3"] = 1739845664] = "Bison3";
  VehicleHash2[VehicleHash2["BJXL"] = 850565707] = "BJXL";
  VehicleHash2[VehicleHash2["Blade"] = 3089165662] = "Blade";
  VehicleHash2[VehicleHash2["Blazer"] = 2166734073] = "Blazer";
  VehicleHash2[VehicleHash2["Blazer2"] = 4246935337] = "Blazer2";
  VehicleHash2[VehicleHash2["Blazer3"] = 3025077634] = "Blazer3";
  VehicleHash2[VehicleHash2["Blazer4"] = 3854198872] = "Blazer4";
  VehicleHash2[VehicleHash2["Blazer5"] = 2704629607] = "Blazer5";
  VehicleHash2[VehicleHash2["Blimp"] = 4143991942] = "Blimp";
  VehicleHash2[VehicleHash2["Blimp2"] = 3681241380] = "Blimp2";
  VehicleHash2[VehicleHash2["Blista"] = 3950024287] = "Blista";
  VehicleHash2[VehicleHash2["Blista2"] = 1039032026] = "Blista2";
  VehicleHash2[VehicleHash2["Blista3"] = 3703315515] = "Blista3";
  VehicleHash2[VehicleHash2["Bmx"] = 1131912276] = "Bmx";
  VehicleHash2[VehicleHash2["BoatTrailer"] = 524108981] = "BoatTrailer";
  VehicleHash2[VehicleHash2["BobcatXL"] = 1069929536] = "BobcatXL";
  VehicleHash2[VehicleHash2["Bodhi2"] = 2859047862] = "Bodhi2";
  VehicleHash2[VehicleHash2["Boxville"] = 2307837162] = "Boxville";
  VehicleHash2[VehicleHash2["Boxville2"] = 4061868990] = "Boxville2";
  VehicleHash2[VehicleHash2["Boxville3"] = 121658888] = "Boxville3";
  VehicleHash2[VehicleHash2["Boxville4"] = 444171386] = "Boxville4";
  VehicleHash2[VehicleHash2["Boxville5"] = 682434785] = "Boxville5";
  VehicleHash2[VehicleHash2["Brawler"] = 2815302597] = "Brawler";
  VehicleHash2[VehicleHash2["Brickade"] = 3989239879] = "Brickade";
  VehicleHash2[VehicleHash2["Brioso"] = 1549126457] = "Brioso";
  VehicleHash2[VehicleHash2["BType"] = 117401876] = "BType";
  VehicleHash2[VehicleHash2["BType2"] = 3463132580] = "BType2";
  VehicleHash2[VehicleHash2["BType3"] = 3692679425] = "BType3";
  VehicleHash2[VehicleHash2["Buccaneer"] = 3612755468] = "Buccaneer";
  VehicleHash2[VehicleHash2["Buccaneer2"] = 3281516360] = "Buccaneer2";
  VehicleHash2[VehicleHash2["Buffalo"] = 3990165190] = "Buffalo";
  VehicleHash2[VehicleHash2["Buffalo2"] = 736902334] = "Buffalo2";
  VehicleHash2[VehicleHash2["Buffalo3"] = 237764926] = "Buffalo3";
  VehicleHash2[VehicleHash2["Bulldozer"] = 1886712733] = "Bulldozer";
  VehicleHash2[VehicleHash2["Bullet"] = 2598821281] = "Bullet";
  VehicleHash2[VehicleHash2["Burrito"] = 2948279460] = "Burrito";
  VehicleHash2[VehicleHash2["Burrito2"] = 3387490166] = "Burrito2";
  VehicleHash2[VehicleHash2["Burrito3"] = 2551651283] = "Burrito3";
  VehicleHash2[VehicleHash2["Burrito4"] = 893081117] = "Burrito4";
  VehicleHash2[VehicleHash2["Burrito5"] = 1132262048] = "Burrito5";
  VehicleHash2[VehicleHash2["Bus"] = 3581397346] = "Bus";
  VehicleHash2[VehicleHash2["Buzzard"] = 788747387] = "Buzzard";
  VehicleHash2[VehicleHash2["Buzzard2"] = 745926877] = "Buzzard2";
  VehicleHash2[VehicleHash2["CableCar"] = 3334677549] = "CableCar";
  VehicleHash2[VehicleHash2["Caddy"] = 1147287684] = "Caddy";
  VehicleHash2[VehicleHash2["Caddy2"] = 3757070668] = "Caddy2";
  VehicleHash2[VehicleHash2["Caddy3"] = 3525819835] = "Caddy3";
  VehicleHash2[VehicleHash2["Camper"] = 1876516712] = "Camper";
  VehicleHash2[VehicleHash2["Carbonizzare"] = 2072687711] = "Carbonizzare";
  VehicleHash2[VehicleHash2["CarbonRS"] = 11251904] = "CarbonRS";
  VehicleHash2[VehicleHash2["Cargobob"] = 4244420235] = "Cargobob";
  VehicleHash2[VehicleHash2["Cargobob2"] = 1621617168] = "Cargobob2";
  VehicleHash2[VehicleHash2["Cargobob3"] = 1394036463] = "Cargobob3";
  VehicleHash2[VehicleHash2["Cargobob4"] = 2025593404] = "Cargobob4";
  VehicleHash2[VehicleHash2["CargoPlane"] = 368211810] = "CargoPlane";
  VehicleHash2[VehicleHash2["Casco"] = 941800958] = "Casco";
  VehicleHash2[VehicleHash2["Cavalcade"] = 2006918058] = "Cavalcade";
  VehicleHash2[VehicleHash2["Cavalcade2"] = 3505073125] = "Cavalcade2";
  VehicleHash2[VehicleHash2["Cheetah"] = 2983812512] = "Cheetah";
  VehicleHash2[VehicleHash2["Cheetah2"] = 223240013] = "Cheetah2";
  VehicleHash2[VehicleHash2["Chimera"] = 6774487] = "Chimera";
  VehicleHash2[VehicleHash2["Chino"] = 349605904] = "Chino";
  VehicleHash2[VehicleHash2["Chino2"] = 2933279331] = "Chino2";
  VehicleHash2[VehicleHash2["Cliffhanger"] = 390201602] = "Cliffhanger";
  VehicleHash2[VehicleHash2["Coach"] = 2222034228] = "Coach";
  VehicleHash2[VehicleHash2["Cog55"] = 906642318] = "Cog55";
  VehicleHash2[VehicleHash2["Cog552"] = 704435172] = "Cog552";
  VehicleHash2[VehicleHash2["CogCabrio"] = 330661258] = "CogCabrio";
  VehicleHash2[VehicleHash2["Cognoscenti"] = 2264796e3] = "Cognoscenti";
  VehicleHash2[VehicleHash2["Cognoscenti2"] = 3690124666] = "Cognoscenti2";
  VehicleHash2[VehicleHash2["Comet2"] = 3249425686] = "Comet2";
  VehicleHash2[VehicleHash2["Comet3"] = 2272483501] = "Comet3";
  VehicleHash2[VehicleHash2["Contender"] = 683047626] = "Contender";
  VehicleHash2[VehicleHash2["Coquette"] = 108773431] = "Coquette";
  VehicleHash2[VehicleHash2["Coquette2"] = 1011753235] = "Coquette2";
  VehicleHash2[VehicleHash2["Coquette3"] = 784565758] = "Coquette3";
  VehicleHash2[VehicleHash2["Cruiser"] = 448402357] = "Cruiser";
  VehicleHash2[VehicleHash2["Crusader"] = 321739290] = "Crusader";
  VehicleHash2[VehicleHash2["Cuban800"] = 3650256867] = "Cuban800";
  VehicleHash2[VehicleHash2["Cutter"] = 3288047904] = "Cutter";
  VehicleHash2[VehicleHash2["Daemon"] = 2006142190] = "Daemon";
  VehicleHash2[VehicleHash2["Daemon2"] = 2890830793] = "Daemon2";
  VehicleHash2[VehicleHash2["Defiler"] = 822018448] = "Defiler";
  VehicleHash2[VehicleHash2["Diablous"] = 4055125828] = "Diablous";
  VehicleHash2[VehicleHash2["Diablous2"] = 1790834270] = "Diablous2";
  VehicleHash2[VehicleHash2["Dilettante"] = 3164157193] = "Dilettante";
  VehicleHash2[VehicleHash2["Dilettante2"] = 1682114128] = "Dilettante2";
  VehicleHash2[VehicleHash2["Dinghy"] = 1033245328] = "Dinghy";
  VehicleHash2[VehicleHash2["Dinghy2"] = 276773164] = "Dinghy2";
  VehicleHash2[VehicleHash2["Dinghy3"] = 509498602] = "Dinghy3";
  VehicleHash2[VehicleHash2["Dinghy4"] = 867467158] = "Dinghy4";
  VehicleHash2[VehicleHash2["DLoader"] = 1770332643] = "DLoader";
  VehicleHash2[VehicleHash2["DockTrailer"] = 2154757102] = "DockTrailer";
  VehicleHash2[VehicleHash2["Docktug"] = 3410276810] = "Docktug";
  VehicleHash2[VehicleHash2["Dodo"] = 3393804037] = "Dodo";
  VehicleHash2[VehicleHash2["Dominator"] = 80636076] = "Dominator";
  VehicleHash2[VehicleHash2["Dominator2"] = 3379262425] = "Dominator2";
  VehicleHash2[VehicleHash2["Double"] = 2623969160] = "Double";
  VehicleHash2[VehicleHash2["Dubsta"] = 1177543287] = "Dubsta";
  VehicleHash2[VehicleHash2["Dubsta2"] = 3900892662] = "Dubsta2";
  VehicleHash2[VehicleHash2["Dubsta3"] = 3057713523] = "Dubsta3";
  VehicleHash2[VehicleHash2["Dukes"] = 723973206] = "Dukes";
  VehicleHash2[VehicleHash2["Dukes2"] = 3968823444] = "Dukes2";
  VehicleHash2[VehicleHash2["Dump"] = 2164484578] = "Dump";
  VehicleHash2[VehicleHash2["Dune"] = 2633113103] = "Dune";
  VehicleHash2[VehicleHash2["Dune2"] = 534258863] = "Dune2";
  VehicleHash2[VehicleHash2["Dune3"] = 1897744184] = "Dune3";
  VehicleHash2[VehicleHash2["Dune4"] = 3467805257] = "Dune4";
  VehicleHash2[VehicleHash2["Dune5"] = 3982671785] = "Dune5";
  VehicleHash2[VehicleHash2["Duster"] = 970356638] = "Duster";
  VehicleHash2[VehicleHash2["Elegy"] = 196747873] = "Elegy";
  VehicleHash2[VehicleHash2["Elegy2"] = 3728579874] = "Elegy2";
  VehicleHash2[VehicleHash2["Emperor"] = 3609690755] = "Emperor";
  VehicleHash2[VehicleHash2["Emperor2"] = 2411965148] = "Emperor2";
  VehicleHash2[VehicleHash2["Emperor3"] = 3053254478] = "Emperor3";
  VehicleHash2[VehicleHash2["Enduro"] = 1753414259] = "Enduro";
  VehicleHash2[VehicleHash2["EntityXF"] = 3003014393] = "EntityXF";
  VehicleHash2[VehicleHash2["Esskey"] = 2035069708] = "Esskey";
  VehicleHash2[VehicleHash2["Exemplar"] = 4289813342] = "Exemplar";
  VehicleHash2[VehicleHash2["F620"] = 3703357e3] = "F620";
  VehicleHash2[VehicleHash2["Faction"] = 2175389151] = "Faction";
  VehicleHash2[VehicleHash2["Faction2"] = 2504420315] = "Faction2";
  VehicleHash2[VehicleHash2["Faction3"] = 2255212070] = "Faction3";
  VehicleHash2[VehicleHash2["Faggio"] = 2452219115] = "Faggio";
  VehicleHash2[VehicleHash2["Faggio2"] = 55628203] = "Faggio2";
  VehicleHash2[VehicleHash2["Faggio3"] = 3005788552] = "Faggio3";
  VehicleHash2[VehicleHash2["FBI"] = 1127131465] = "FBI";
  VehicleHash2[VehicleHash2["FBI2"] = 2647026068] = "FBI2";
  VehicleHash2[VehicleHash2["FCR"] = 627535535] = "FCR";
  VehicleHash2[VehicleHash2["FCR2"] = 3537231886] = "FCR2";
  VehicleHash2[VehicleHash2["Felon"] = 3903372712] = "Felon";
  VehicleHash2[VehicleHash2["Felon2"] = 4205676014] = "Felon2";
  VehicleHash2[VehicleHash2["Feltzer2"] = 2299640309] = "Feltzer2";
  VehicleHash2[VehicleHash2["Feltzer3"] = 2728226064] = "Feltzer3";
  VehicleHash2[VehicleHash2["FireTruk"] = 1938952078] = "FireTruk";
  VehicleHash2[VehicleHash2["Fixter"] = 3458454463] = "Fixter";
  VehicleHash2[VehicleHash2["Flatbed"] = 1353720154] = "Flatbed";
  VehicleHash2[VehicleHash2["Forklift"] = 1491375716] = "Forklift";
  VehicleHash2[VehicleHash2["FMJ"] = 1426219628] = "FMJ";
  VehicleHash2[VehicleHash2["FQ2"] = 3157435195] = "FQ2";
  VehicleHash2[VehicleHash2["Freight"] = 1030400667] = "Freight";
  VehicleHash2[VehicleHash2["FreightCar"] = 184361638] = "FreightCar";
  VehicleHash2[VehicleHash2["FreightCont1"] = 920453016] = "FreightCont1";
  VehicleHash2[VehicleHash2["FreightCont2"] = 240201337] = "FreightCont2";
  VehicleHash2[VehicleHash2["FreightGrain"] = 642617954] = "FreightGrain";
  VehicleHash2[VehicleHash2["FreightTrailer"] = 3517691494] = "FreightTrailer";
  VehicleHash2[VehicleHash2["Frogger"] = 744705981] = "Frogger";
  VehicleHash2[VehicleHash2["Frogger2"] = 1949211328] = "Frogger2";
  VehicleHash2[VehicleHash2["Fugitive"] = 1909141499] = "Fugitive";
  VehicleHash2[VehicleHash2["Furoregt"] = 3205927392] = "Furoregt";
  VehicleHash2[VehicleHash2["Fusilade"] = 499169875] = "Fusilade";
  VehicleHash2[VehicleHash2["Futo"] = 2016857647] = "Futo";
  VehicleHash2[VehicleHash2["Gargoyle"] = 741090084] = "Gargoyle";
  VehicleHash2[VehicleHash2["Gauntlet"] = 2494797253] = "Gauntlet";
  VehicleHash2[VehicleHash2["Gauntlet2"] = 349315417] = "Gauntlet2";
  VehicleHash2[VehicleHash2["GBurrito"] = 2549763894] = "GBurrito";
  VehicleHash2[VehicleHash2["GBurrito2"] = 296357396] = "GBurrito2";
  VehicleHash2[VehicleHash2["Glendale"] = 75131841] = "Glendale";
  VehicleHash2[VehicleHash2["GP1"] = 1234311532] = "GP1";
  VehicleHash2[VehicleHash2["GrainTrailer"] = 1019737494] = "GrainTrailer";
  VehicleHash2[VehicleHash2["Granger"] = 2519238556] = "Granger";
  VehicleHash2[VehicleHash2["Gresley"] = 2751205197] = "Gresley";
  VehicleHash2[VehicleHash2["Guardian"] = 2186977100] = "Guardian";
  VehicleHash2[VehicleHash2["Habanero"] = 884422927] = "Habanero";
  VehicleHash2[VehicleHash2["Hakuchou"] = 1265391242] = "Hakuchou";
  VehicleHash2[VehicleHash2["Hakuchou2"] = 4039289119] = "Hakuchou2";
  VehicleHash2[VehicleHash2["HalfTrack"] = 4262731174] = "HalfTrack";
  VehicleHash2[VehicleHash2["Handler"] = 444583674] = "Handler";
  VehicleHash2[VehicleHash2["Hauler"] = 1518533038] = "Hauler";
  VehicleHash2[VehicleHash2["Hauler2"] = 387748548] = "Hauler2";
  VehicleHash2[VehicleHash2["Hexer"] = 301427732] = "Hexer";
  VehicleHash2[VehicleHash2["Hotknife"] = 37348240] = "Hotknife";
  VehicleHash2[VehicleHash2["Huntley"] = 486987393] = "Huntley";
  VehicleHash2[VehicleHash2["Hydra"] = 970385471] = "Hydra";
  VehicleHash2[VehicleHash2["Infernus"] = 418536135] = "Infernus";
  VehicleHash2[VehicleHash2["Infernus2"] = 2889029532] = "Infernus2";
  VehicleHash2[VehicleHash2["Ingot"] = 3005245074] = "Ingot";
  VehicleHash2[VehicleHash2["Innovation"] = 4135840458] = "Innovation";
  VehicleHash2[VehicleHash2["Insurgent"] = 2434067162] = "Insurgent";
  VehicleHash2[VehicleHash2["Insurgent2"] = 2071877360] = "Insurgent2";
  VehicleHash2[VehicleHash2["Insurgent3"] = 2370534026] = "Insurgent3";
  VehicleHash2[VehicleHash2["Intruder"] = 886934177] = "Intruder";
  VehicleHash2[VehicleHash2["Issi2"] = 3117103977] = "Issi2";
  VehicleHash2[VehicleHash2["ItaliGTB"] = 2246633323] = "ItaliGTB";
  VehicleHash2[VehicleHash2["ItaliGTB2"] = 3812247419] = "ItaliGTB2";
  VehicleHash2[VehicleHash2["Jackal"] = 3670438162] = "Jackal";
  VehicleHash2[VehicleHash2["JB700"] = 1051415893] = "JB700";
  VehicleHash2[VehicleHash2["Jester"] = 2997294755] = "Jester";
  VehicleHash2[VehicleHash2["Jester2"] = 3188613414] = "Jester2";
  VehicleHash2[VehicleHash2["Jet"] = 1058115860] = "Jet";
  VehicleHash2[VehicleHash2["Jetmax"] = 861409633] = "Jetmax";
  VehicleHash2[VehicleHash2["Journey"] = 4174679674] = "Journey";
  VehicleHash2[VehicleHash2["Kalahari"] = 92612664] = "Kalahari";
  VehicleHash2[VehicleHash2["Khamelion"] = 544021352] = "Khamelion";
  VehicleHash2[VehicleHash2["Kuruma"] = 2922118804] = "Kuruma";
  VehicleHash2[VehicleHash2["Kuruma2"] = 410882957] = "Kuruma2";
  VehicleHash2[VehicleHash2["Landstalker"] = 1269098716] = "Landstalker";
  VehicleHash2[VehicleHash2["Lazer"] = 3013282534] = "Lazer";
  VehicleHash2[VehicleHash2["LE7B"] = 3062131285] = "LE7B";
  VehicleHash2[VehicleHash2["Lectro"] = 640818791] = "Lectro";
  VehicleHash2[VehicleHash2["Lguard"] = 469291905] = "Lguard";
  VehicleHash2[VehicleHash2["Limo2"] = 4180339789] = "Limo2";
  VehicleHash2[VehicleHash2["Lurcher"] = 2068293287] = "Lurcher";
  VehicleHash2[VehicleHash2["Luxor"] = 621481054] = "Luxor";
  VehicleHash2[VehicleHash2["Luxor2"] = 3080673438] = "Luxor2";
  VehicleHash2[VehicleHash2["Lynx"] = 482197771] = "Lynx";
  VehicleHash2[VehicleHash2["Mamba"] = 2634021974] = "Mamba";
  VehicleHash2[VehicleHash2["Mammatus"] = 2548391185] = "Mammatus";
  VehicleHash2[VehicleHash2["Manana"] = 2170765704] = "Manana";
  VehicleHash2[VehicleHash2["Manchez"] = 2771538552] = "Manchez";
  VehicleHash2[VehicleHash2["Marquis"] = 3251507587] = "Marquis";
  VehicleHash2[VehicleHash2["Marshall"] = 1233534620] = "Marshall";
  VehicleHash2[VehicleHash2["Massacro"] = 4152024626] = "Massacro";
  VehicleHash2[VehicleHash2["Massacro2"] = 3663206819] = "Massacro2";
  VehicleHash2[VehicleHash2["Maverick"] = 2634305738] = "Maverick";
  VehicleHash2[VehicleHash2["Mesa"] = 914654722] = "Mesa";
  VehicleHash2[VehicleHash2["Mesa2"] = 3546958660] = "Mesa2";
  VehicleHash2[VehicleHash2["Mesa3"] = 2230595153] = "Mesa3";
  VehicleHash2[VehicleHash2["MetroTrain"] = 868868440] = "MetroTrain";
  VehicleHash2[VehicleHash2["Miljet"] = 165154707] = "Miljet";
  VehicleHash2[VehicleHash2["Minivan"] = 3984502180] = "Minivan";
  VehicleHash2[VehicleHash2["Minivan2"] = 3168702960] = "Minivan2";
  VehicleHash2[VehicleHash2["Mixer"] = 3510150843] = "Mixer";
  VehicleHash2[VehicleHash2["Mixer2"] = 475220373] = "Mixer2";
  VehicleHash2[VehicleHash2["Monroe"] = 3861591579] = "Monroe";
  VehicleHash2[VehicleHash2["Monster"] = 3449006043] = "Monster";
  VehicleHash2[VehicleHash2["Moonbeam"] = 525509695] = "Moonbeam";
  VehicleHash2[VehicleHash2["Moonbeam2"] = 1896491931] = "Moonbeam2";
  VehicleHash2[VehicleHash2["Mower"] = 1783355638] = "Mower";
  VehicleHash2[VehicleHash2["Mule"] = 904750859] = "Mule";
  VehicleHash2[VehicleHash2["Mule2"] = 3244501995] = "Mule2";
  VehicleHash2[VehicleHash2["Mule3"] = 2242229361] = "Mule3";
  VehicleHash2[VehicleHash2["Nemesis"] = 3660088182] = "Nemesis";
  VehicleHash2[VehicleHash2["Nero"] = 1034187331] = "Nero";
  VehicleHash2[VehicleHash2["Nero2"] = 1093792632] = "Nero2";
  VehicleHash2[VehicleHash2["Nightblade"] = 2688780135] = "Nightblade";
  VehicleHash2[VehicleHash2["Nightshade"] = 2351681756] = "Nightshade";
  VehicleHash2[VehicleHash2["NightShark"] = 433954513] = "NightShark";
  VehicleHash2[VehicleHash2["Nimbus"] = 2999939664] = "Nimbus";
  VehicleHash2[VehicleHash2["Ninef"] = 1032823388] = "Ninef";
  VehicleHash2[VehicleHash2["Ninef2"] = 2833484545] = "Ninef2";
  VehicleHash2[VehicleHash2["Omnis"] = 3517794615] = "Omnis";
  VehicleHash2[VehicleHash2["Oppressor"] = 884483972] = "Oppressor";
  VehicleHash2[VehicleHash2["Oracle"] = 1348744438] = "Oracle";
  VehicleHash2[VehicleHash2["Oracle2"] = 3783366066] = "Oracle2";
  VehicleHash2[VehicleHash2["Osiris"] = 1987142870] = "Osiris";
  VehicleHash2[VehicleHash2["Packer"] = 569305213] = "Packer";
  VehicleHash2[VehicleHash2["Panto"] = 3863274624] = "Panto";
  VehicleHash2[VehicleHash2["Paradise"] = 1488164764] = "Paradise";
  VehicleHash2[VehicleHash2["Patriot"] = 3486509883] = "Patriot";
  VehicleHash2[VehicleHash2["PBus"] = 2287941233] = "PBus";
  VehicleHash2[VehicleHash2["PCJ"] = 3385765638] = "PCJ";
  VehicleHash2[VehicleHash2["Penetrator"] = 2536829930] = "Penetrator";
  VehicleHash2[VehicleHash2["Penumbra"] = 3917501776] = "Penumbra";
  VehicleHash2[VehicleHash2["Peyote"] = 1830407356] = "Peyote";
  VehicleHash2[VehicleHash2["Pfister811"] = 2465164804] = "Pfister811";
  VehicleHash2[VehicleHash2["Phantom"] = 2157618379] = "Phantom";
  VehicleHash2[VehicleHash2["Phantom2"] = 2645431192] = "Phantom2";
  VehicleHash2[VehicleHash2["Phantom3"] = 177270108] = "Phantom3";
  VehicleHash2[VehicleHash2["Phoenix"] = 2199527893] = "Phoenix";
  VehicleHash2[VehicleHash2["Picador"] = 1507916787] = "Picador";
  VehicleHash2[VehicleHash2["Pigalle"] = 1078682497] = "Pigalle";
  VehicleHash2[VehicleHash2["Police"] = 2046537925] = "Police";
  VehicleHash2[VehicleHash2["Police2"] = 2667966721] = "Police2";
  VehicleHash2[VehicleHash2["Police3"] = 1912215274] = "Police3";
  VehicleHash2[VehicleHash2["Police4"] = 2321795001] = "Police4";
  VehicleHash2[VehicleHash2["Policeb"] = 4260343491] = "Policeb";
  VehicleHash2[VehicleHash2["PoliceOld1"] = 2758042359] = "PoliceOld1";
  VehicleHash2[VehicleHash2["PoliceOld2"] = 2515846680] = "PoliceOld2";
  VehicleHash2[VehicleHash2["PoliceT"] = 456714581] = "PoliceT";
  VehicleHash2[VehicleHash2["Polmav"] = 353883353] = "Polmav";
  VehicleHash2[VehicleHash2["Pony"] = 4175309224] = "Pony";
  VehicleHash2[VehicleHash2["Pony2"] = 943752001] = "Pony2";
  VehicleHash2[VehicleHash2["Pounder"] = 2112052861] = "Pounder";
  VehicleHash2[VehicleHash2["Prairie"] = 2844316578] = "Prairie";
  VehicleHash2[VehicleHash2["Pranger"] = 741586030] = "Pranger";
  VehicleHash2[VehicleHash2["Predator"] = 3806844075] = "Predator";
  VehicleHash2[VehicleHash2["Premier"] = 2411098011] = "Premier";
  VehicleHash2[VehicleHash2["Primo"] = 3144368207] = "Primo";
  VehicleHash2[VehicleHash2["Primo2"] = 2254540506] = "Primo2";
  VehicleHash2[VehicleHash2["PropTrailer"] = 356391690] = "PropTrailer";
  VehicleHash2[VehicleHash2["Prototipo"] = 2123327359] = "Prototipo";
  VehicleHash2[VehicleHash2["Radi"] = 2643899483] = "Radi";
  VehicleHash2[VehicleHash2["RakeTrailer"] = 390902130] = "RakeTrailer";
  VehicleHash2[VehicleHash2["RancherXL"] = 1645267888] = "RancherXL";
  VehicleHash2[VehicleHash2["RancherXL2"] = 1933662059] = "RancherXL2";
  VehicleHash2[VehicleHash2["RallyTruck"] = 2191146052] = "RallyTruck";
  VehicleHash2[VehicleHash2["RapidGT"] = 2360515092] = "RapidGT";
  VehicleHash2[VehicleHash2["RapidGT2"] = 1737773231] = "RapidGT2";
  VehicleHash2[VehicleHash2["Raptor"] = 3620039993] = "Raptor";
  VehicleHash2[VehicleHash2["RatBike"] = 1873600305] = "RatBike";
  VehicleHash2[VehicleHash2["RatLoader"] = 3627815886] = "RatLoader";
  VehicleHash2[VehicleHash2["RatLoader2"] = 3705788919] = "RatLoader2";
  VehicleHash2[VehicleHash2["Reaper"] = 234062309] = "Reaper";
  VehicleHash2[VehicleHash2["Rebel"] = 3087195462] = "Rebel";
  VehicleHash2[VehicleHash2["Rebel2"] = 2249373259] = "Rebel2";
  VehicleHash2[VehicleHash2["Regina"] = 4280472072] = "Regina";
  VehicleHash2[VehicleHash2["RentalBus"] = 3196165219] = "RentalBus";
  VehicleHash2[VehicleHash2["Rhapsody"] = 841808271] = "Rhapsody";
  VehicleHash2[VehicleHash2["Rhino"] = 782665360] = "Rhino";
  VehicleHash2[VehicleHash2["Riot"] = 3089277354] = "Riot";
  VehicleHash2[VehicleHash2["Ripley"] = 3448987385] = "Ripley";
  VehicleHash2[VehicleHash2["Rocoto"] = 2136773105] = "Rocoto";
  VehicleHash2[VehicleHash2["Romero"] = 627094268] = "Romero";
  VehicleHash2[VehicleHash2["Rubble"] = 2589662668] = "Rubble";
  VehicleHash2[VehicleHash2["Ruffian"] = 3401388520] = "Ruffian";
  VehicleHash2[VehicleHash2["Ruiner"] = 4067225593] = "Ruiner";
  VehicleHash2[VehicleHash2["Ruiner2"] = 941494461] = "Ruiner2";
  VehicleHash2[VehicleHash2["Ruiner3"] = 777714999] = "Ruiner3";
  VehicleHash2[VehicleHash2["Rumpo"] = 1162065741] = "Rumpo";
  VehicleHash2[VehicleHash2["Rumpo2"] = 2518351607] = "Rumpo2";
  VehicleHash2[VehicleHash2["Rumpo3"] = 1475773103] = "Rumpo3";
  VehicleHash2[VehicleHash2["Ruston"] = 719660200] = "Ruston";
  VehicleHash2[VehicleHash2["SabreGT"] = 2609945748] = "SabreGT";
  VehicleHash2[VehicleHash2["SabreGT2"] = 223258115] = "SabreGT2";
  VehicleHash2[VehicleHash2["Sadler"] = 3695398481] = "Sadler";
  VehicleHash2[VehicleHash2["Sadler2"] = 734217681] = "Sadler2";
  VehicleHash2[VehicleHash2["Sanchez"] = 788045382] = "Sanchez";
  VehicleHash2[VehicleHash2["Sanchez2"] = 2841686334] = "Sanchez2";
  VehicleHash2[VehicleHash2["Sanctus"] = 1491277511] = "Sanctus";
  VehicleHash2[VehicleHash2["Sandking"] = 3105951696] = "Sandking";
  VehicleHash2[VehicleHash2["Sandking2"] = 989381445] = "Sandking2";
  VehicleHash2[VehicleHash2["Savage"] = 4212341271] = "Savage";
  VehicleHash2[VehicleHash2["Schafter2"] = 3039514899] = "Schafter2";
  VehicleHash2[VehicleHash2["Schafter3"] = 2809443750] = "Schafter3";
  VehicleHash2[VehicleHash2["Schafter4"] = 1489967196] = "Schafter4";
  VehicleHash2[VehicleHash2["Schafter5"] = 3406724313] = "Schafter5";
  VehicleHash2[VehicleHash2["Schafter6"] = 1922255844] = "Schafter6";
  VehicleHash2[VehicleHash2["Schwarzer"] = 3548084598] = "Schwarzer";
  VehicleHash2[VehicleHash2["Scorcher"] = 4108429845] = "Scorcher";
  VehicleHash2[VehicleHash2["Scrap"] = 2594165727] = "Scrap";
  VehicleHash2[VehicleHash2["Seashark"] = 3264692260] = "Seashark";
  VehicleHash2[VehicleHash2["Seashark2"] = 3678636260] = "Seashark2";
  VehicleHash2[VehicleHash2["Seashark3"] = 3983945033] = "Seashark3";
  VehicleHash2[VehicleHash2["Seminole"] = 1221512915] = "Seminole";
  VehicleHash2[VehicleHash2["Sentinel"] = 1349725314] = "Sentinel";
  VehicleHash2[VehicleHash2["Sentinel2"] = 873639469] = "Sentinel2";
  VehicleHash2[VehicleHash2["Serrano"] = 1337041428] = "Serrano";
  VehicleHash2[VehicleHash2["Seven70"] = 2537130571] = "Seven70";
  VehicleHash2[VehicleHash2["Shamal"] = 3080461301] = "Shamal";
  VehicleHash2[VehicleHash2["Sheava"] = 819197656] = "Sheava";
  VehicleHash2[VehicleHash2["Sheriff"] = 2611638396] = "Sheriff";
  VehicleHash2[VehicleHash2["Sheriff2"] = 1922257928] = "Sheriff2";
  VehicleHash2[VehicleHash2["Shotaro"] = 3889340782] = "Shotaro";
  VehicleHash2[VehicleHash2["Skylift"] = 1044954915] = "Skylift";
  VehicleHash2[VehicleHash2["SlamVan"] = 729783779] = "SlamVan";
  VehicleHash2[VehicleHash2["SlamVan2"] = 833469436] = "SlamVan2";
  VehicleHash2[VehicleHash2["SlamVan3"] = 1119641113] = "SlamVan3";
  VehicleHash2[VehicleHash2["Sovereign"] = 743478836] = "Sovereign";
  VehicleHash2[VehicleHash2["Specter"] = 1886268224] = "Specter";
  VehicleHash2[VehicleHash2["Specter2"] = 1074745671] = "Specter2";
  VehicleHash2[VehicleHash2["Speeder"] = 231083307] = "Speeder";
  VehicleHash2[VehicleHash2["Speeder2"] = 437538602] = "Speeder2";
  VehicleHash2[VehicleHash2["Speedo"] = 3484649228] = "Speedo";
  VehicleHash2[VehicleHash2["Speedo2"] = 728614474] = "Speedo2";
  VehicleHash2[VehicleHash2["Squalo"] = 400514754] = "Squalo";
  VehicleHash2[VehicleHash2["Stalion"] = 1923400478] = "Stalion";
  VehicleHash2[VehicleHash2["Stalion2"] = 3893323758] = "Stalion2";
  VehicleHash2[VehicleHash2["Stanier"] = 2817386317] = "Stanier";
  VehicleHash2[VehicleHash2["Stinger"] = 1545842587] = "Stinger";
  VehicleHash2[VehicleHash2["StingerGT"] = 2196019706] = "StingerGT";
  VehicleHash2[VehicleHash2["Stockade"] = 1747439474] = "Stockade";
  VehicleHash2[VehicleHash2["Stockade3"] = 4080511798] = "Stockade3";
  VehicleHash2[VehicleHash2["Stratum"] = 1723137093] = "Stratum";
  VehicleHash2[VehicleHash2["Stretch"] = 2333339779] = "Stretch";
  VehicleHash2[VehicleHash2["Stunt"] = 2172210288] = "Stunt";
  VehicleHash2[VehicleHash2["Submersible"] = 771711535] = "Submersible";
  VehicleHash2[VehicleHash2["Submersible2"] = 3228633070] = "Submersible2";
  VehicleHash2[VehicleHash2["Sultan"] = 970598228] = "Sultan";
  VehicleHash2[VehicleHash2["SultanRS"] = 3999278268] = "SultanRS";
  VehicleHash2[VehicleHash2["Suntrap"] = 4012021193] = "Suntrap";
  VehicleHash2[VehicleHash2["Superd"] = 1123216662] = "Superd";
  VehicleHash2[VehicleHash2["Supervolito"] = 710198397] = "Supervolito";
  VehicleHash2[VehicleHash2["Supervolito2"] = 2623428164] = "Supervolito2";
  VehicleHash2[VehicleHash2["Surano"] = 384071873] = "Surano";
  VehicleHash2[VehicleHash2["Surfer"] = 699456151] = "Surfer";
  VehicleHash2[VehicleHash2["Surfer2"] = 2983726598] = "Surfer2";
  VehicleHash2[VehicleHash2["Surge"] = 2400073108] = "Surge";
  VehicleHash2[VehicleHash2["Swift2"] = 1075432268] = "Swift2";
  VehicleHash2[VehicleHash2["Swift"] = 3955379698] = "Swift";
  VehicleHash2[VehicleHash2["T20"] = 1663218586] = "T20";
  VehicleHash2[VehicleHash2["Taco"] = 1951180813] = "Taco";
  VehicleHash2[VehicleHash2["Tailgater"] = 3286105550] = "Tailgater";
  VehicleHash2[VehicleHash2["Tampa"] = 972671128] = "Tampa";
  VehicleHash2[VehicleHash2["Tampa2"] = 3223586949] = "Tampa2";
  VehicleHash2[VehicleHash2["Tampa3"] = 3084515313] = "Tampa3";
  VehicleHash2[VehicleHash2["Tanker"] = 3564062519] = "Tanker";
  VehicleHash2[VehicleHash2["Tanker2"] = 1956216962] = "Tanker2";
  VehicleHash2[VehicleHash2["TankerCar"] = 586013744] = "TankerCar";
  VehicleHash2[VehicleHash2["Taxi"] = 3338918751] = "Taxi";
  VehicleHash2[VehicleHash2["Technical"] = 2198148358] = "Technical";
  VehicleHash2[VehicleHash2["Technical2"] = 1180875963] = "Technical2";
  VehicleHash2[VehicleHash2["Technical3"] = 1356124575] = "Technical3";
  VehicleHash2[VehicleHash2["Tempesta"] = 272929391] = "Tempesta";
  VehicleHash2[VehicleHash2["Thrust"] = 1836027715] = "Thrust";
  VehicleHash2[VehicleHash2["TipTruck"] = 48339065] = "TipTruck";
  VehicleHash2[VehicleHash2["TipTruck2"] = 3347205726] = "TipTruck2";
  VehicleHash2[VehicleHash2["Titan"] = 1981688531] = "Titan";
  VehicleHash2[VehicleHash2["Torero"] = 1504306544] = "Torero";
  VehicleHash2[VehicleHash2["Tornado"] = 464687292] = "Tornado";
  VehicleHash2[VehicleHash2["Tornado2"] = 1531094468] = "Tornado2";
  VehicleHash2[VehicleHash2["Tornado3"] = 1762279763] = "Tornado3";
  VehicleHash2[VehicleHash2["Tornado4"] = 2261744861] = "Tornado4";
  VehicleHash2[VehicleHash2["Tornado5"] = 2497353967] = "Tornado5";
  VehicleHash2[VehicleHash2["Tornado6"] = 2736567667] = "Tornado6";
  VehicleHash2[VehicleHash2["Toro"] = 1070967343] = "Toro";
  VehicleHash2[VehicleHash2["Toro2"] = 908897389] = "Toro2";
  VehicleHash2[VehicleHash2["Tourbus"] = 1941029835] = "Tourbus";
  VehicleHash2[VehicleHash2["TowTruck"] = 2971866336] = "TowTruck";
  VehicleHash2[VehicleHash2["TowTruck2"] = 3852654278] = "TowTruck2";
  VehicleHash2[VehicleHash2["TR2"] = 2078290630] = "TR2";
  VehicleHash2[VehicleHash2["TR3"] = 1784254509] = "TR3";
  VehicleHash2[VehicleHash2["TR4"] = 2091594960] = "TR4";
  VehicleHash2[VehicleHash2["Tractor"] = 1641462412] = "Tractor";
  VehicleHash2[VehicleHash2["Tractor2"] = 2218488798] = "Tractor2";
  VehicleHash2[VehicleHash2["Tractor3"] = 1445631933] = "Tractor3";
  VehicleHash2[VehicleHash2["TrailerLogs"] = 2016027501] = "TrailerLogs";
  VehicleHash2[VehicleHash2["TrailerLarge"] = 1502869817] = "TrailerLarge";
  VehicleHash2[VehicleHash2["Trailers"] = 3417488910] = "Trailers";
  VehicleHash2[VehicleHash2["Trailers2"] = 2715434129] = "Trailers2";
  VehicleHash2[VehicleHash2["Trailers3"] = 2236089197] = "Trailers3";
  VehicleHash2[VehicleHash2["Trailers4"] = 3194418602] = "Trailers4";
  VehicleHash2[VehicleHash2["TrailerSmall"] = 712162987] = "TrailerSmall";
  VehicleHash2[VehicleHash2["TrailerSmall2"] = 2413121211] = "TrailerSmall2";
  VehicleHash2[VehicleHash2["Trash"] = 1917016601] = "Trash";
  VehicleHash2[VehicleHash2["Trash2"] = 3039269212] = "Trash2";
  VehicleHash2[VehicleHash2["TRFlat"] = 2942498482] = "TRFlat";
  VehicleHash2[VehicleHash2["TriBike"] = 1127861609] = "TriBike";
  VehicleHash2[VehicleHash2["TriBike2"] = 3061159916] = "TriBike2";
  VehicleHash2[VehicleHash2["TriBike3"] = 3894672200] = "TriBike3";
  VehicleHash2[VehicleHash2["TrophyTruck"] = 101905590] = "TrophyTruck";
  VehicleHash2[VehicleHash2["TrophyTruck2"] = 3631668194] = "TrophyTruck2";
  VehicleHash2[VehicleHash2["Tropic"] = 290013743] = "Tropic";
  VehicleHash2[VehicleHash2["Tropic2"] = 1448677353] = "Tropic2";
  VehicleHash2[VehicleHash2["Tropos"] = 1887331236] = "Tropos";
  VehicleHash2[VehicleHash2["Tug"] = 2194326579] = "Tug";
  VehicleHash2[VehicleHash2["Turismor"] = 408192225] = "Turismor";
  VehicleHash2[VehicleHash2["Turismo2"] = 3312836369] = "Turismo2";
  VehicleHash2[VehicleHash2["TVTrailer"] = 2524324030] = "TVTrailer";
  VehicleHash2[VehicleHash2["Tyrus"] = 2067820283] = "Tyrus";
  VehicleHash2[VehicleHash2["UtilliTruck"] = 516990260] = "UtilliTruck";
  VehicleHash2[VehicleHash2["UtilliTruck2"] = 887537515] = "UtilliTruck2";
  VehicleHash2[VehicleHash2["UtilliTruck3"] = 2132890591] = "UtilliTruck3";
  VehicleHash2[VehicleHash2["Vacca"] = 338562499] = "Vacca";
  VehicleHash2[VehicleHash2["Vader"] = 4154065143] = "Vader";
  VehicleHash2[VehicleHash2["Vagner"] = 1939284556] = "Vagner";
  VehicleHash2[VehicleHash2["Valkyrie"] = 2694714877] = "Valkyrie";
  VehicleHash2[VehicleHash2["Valkyrie2"] = 1543134283] = "Valkyrie2";
  VehicleHash2[VehicleHash2["Velum"] = 2621610858] = "Velum";
  VehicleHash2[VehicleHash2["Velum2"] = 1077420264] = "Velum2";
  VehicleHash2[VehicleHash2["Verlierer2"] = 1102544804] = "Verlierer2";
  VehicleHash2[VehicleHash2["Vestra"] = 1341619767] = "Vestra";
  VehicleHash2[VehicleHash2["Vigero"] = 3469130167] = "Vigero";
  VehicleHash2[VehicleHash2["Vindicator"] = 2941886209] = "Vindicator";
  VehicleHash2[VehicleHash2["Virgo"] = 3796912450] = "Virgo";
  VehicleHash2[VehicleHash2["Virgo2"] = 3395457658] = "Virgo2";
  VehicleHash2[VehicleHash2["Virgo3"] = 16646064] = "Virgo3";
  VehicleHash2[VehicleHash2["Volatus"] = 2449479409] = "Volatus";
  VehicleHash2[VehicleHash2["Voltic"] = 2672523198] = "Voltic";
  VehicleHash2[VehicleHash2["Voltic2"] = 989294410] = "Voltic2";
  VehicleHash2[VehicleHash2["Voodoo"] = 2006667053] = "Voodoo";
  VehicleHash2[VehicleHash2["Voodoo2"] = 523724515] = "Voodoo2";
  VehicleHash2[VehicleHash2["Vortex"] = 3685342204] = "Vortex";
  VehicleHash2[VehicleHash2["Warrener"] = 1373123368] = "Warrener";
  VehicleHash2[VehicleHash2["Washington"] = 1777363799] = "Washington";
  VehicleHash2[VehicleHash2["Wastelander"] = 2382949506] = "Wastelander";
  VehicleHash2[VehicleHash2["Windsor"] = 1581459400] = "Windsor";
  VehicleHash2[VehicleHash2["Windsor2"] = 2364918497] = "Windsor2";
  VehicleHash2[VehicleHash2["Wolfsbane"] = 3676349299] = "Wolfsbane";
  VehicleHash2[VehicleHash2["XA21"] = 917809321] = "XA21";
  VehicleHash2[VehicleHash2["XLS"] = 1203490606] = "XLS";
  VehicleHash2[VehicleHash2["XLS2"] = 3862958888] = "XLS2";
  VehicleHash2[VehicleHash2["Youga"] = 65402552] = "Youga";
  VehicleHash2[VehicleHash2["Youga2"] = 1026149675] = "Youga2";
  VehicleHash2[VehicleHash2["Zentorno"] = 2891838741] = "Zentorno";
  VehicleHash2[VehicleHash2["Zion"] = 3172678083] = "Zion";
  VehicleHash2[VehicleHash2["Zion2"] = 3101863448] = "Zion2";
  VehicleHash2[VehicleHash2["ZombieA"] = 3285698347] = "ZombieA";
  VehicleHash2[VehicleHash2["ZombieB"] = 3724934023] = "ZombieB";
  VehicleHash2[VehicleHash2["ZType"] = 758895617] = "ZType";
  return VehicleHash2;
})(VehicleHash || {});
export {
  VehicleHash
};
