// @nativewrappers/fivem/hashes/WeaponHash.js
var WeaponHash = /* @__PURE__ */ ((WeaponHash2) => {
  WeaponHash2[WeaponHash2["Dagger"] = -1834847097] = "Dagger";
  WeaponHash2[WeaponHash2["Bat"] = -1786099057] = "Bat";
  WeaponHash2[WeaponHash2["Bottle"] = -102323637] = "Bottle";
  WeaponHash2[WeaponHash2["Crowbar"] = -2067956739] = "Crowbar";
  WeaponHash2[WeaponHash2["Unarmed"] = -1569615261] = "Unarmed";
  WeaponHash2[WeaponHash2["Flashlight"] = -1951375401] = "Flashlight";
  WeaponHash2[WeaponHash2["GolfClub"] = 1141786504] = "GolfClub";
  WeaponHash2[WeaponHash2["Hammer"] = 1317494643] = "Hammer";
  WeaponHash2[WeaponHash2["Hatchet"] = -102973651] = "Hatchet";
  WeaponHash2[WeaponHash2["KnuckleDuster"] = -656458692] = "KnuckleDuster";
  WeaponHash2[WeaponHash2["Knife"] = -1716189206] = "Knife";
  WeaponHash2[WeaponHash2["Machete"] = -581044007] = "Machete";
  WeaponHash2[WeaponHash2["SwitchBlade"] = -538741184] = "SwitchBlade";
  WeaponHash2[WeaponHash2["Nightstick"] = 1737195953] = "Nightstick";
  WeaponHash2[WeaponHash2["Wrench"] = 419712736] = "Wrench";
  WeaponHash2[WeaponHash2["BattleAxe"] = -853065399] = "BattleAxe";
  WeaponHash2[WeaponHash2["PoolCue"] = -1810795771] = "PoolCue";
  WeaponHash2[WeaponHash2["StoneHatchet"] = 940833800] = "StoneHatchet";
  WeaponHash2[WeaponHash2["Pistol"] = 453432689] = "Pistol";
  WeaponHash2[WeaponHash2["PistolMk2"] = 3219281620] = "PistolMk2";
  WeaponHash2[WeaponHash2["CombatPistol"] = 1593441988] = "CombatPistol";
  WeaponHash2[WeaponHash2["APPistol"] = 584646201] = "APPistol";
  WeaponHash2[WeaponHash2["StunGun"] = 911657153] = "StunGun";
  WeaponHash2[WeaponHash2["StunGunMp"] = 1171102963] = "StunGunMp";
  WeaponHash2[WeaponHash2["Pistol50"] = -1716589765] = "Pistol50";
  WeaponHash2[WeaponHash2["SNSPistol"] = -1076751822] = "SNSPistol";
  WeaponHash2[WeaponHash2["SNSPistolMk2"] = -2009644972] = "SNSPistolMk2";
  WeaponHash2[WeaponHash2["HeavyPistol"] = -771403250] = "HeavyPistol";
  WeaponHash2[WeaponHash2["VintagePistol"] = 137902532] = "VintagePistol";
  WeaponHash2[WeaponHash2["FlareGun"] = 1198879012] = "FlareGun";
  WeaponHash2[WeaponHash2["MarksmanPistol"] = -598887786] = "MarksmanPistol";
  WeaponHash2[WeaponHash2["Revolver"] = -1045183535] = "Revolver";
  WeaponHash2[WeaponHash2["RevolverMk2"] = -879347409] = "RevolverMk2";
  WeaponHash2[WeaponHash2["DoubleAction"] = -1746263880] = "DoubleAction";
  WeaponHash2[WeaponHash2["RayPistol"] = -1355376991] = "RayPistol";
  WeaponHash2[WeaponHash2["CeramicPistol"] = 727643628] = "CeramicPistol";
  WeaponHash2[WeaponHash2["NavyRevolver"] = -1853920116] = "NavyRevolver";
  WeaponHash2[WeaponHash2["GadgetPistol"] = 1470379660] = "GadgetPistol";
  WeaponHash2[WeaponHash2["MicroSMG"] = 324215364] = "MicroSMG";
  WeaponHash2[WeaponHash2["SMG"] = 736523883] = "SMG";
  WeaponHash2[WeaponHash2["SMGMk2"] = 2024373456] = "SMGMk2";
  WeaponHash2[WeaponHash2["AssaultSMG"] = -270015777] = "AssaultSMG";
  WeaponHash2[WeaponHash2["CombatPDW"] = 171789620] = "CombatPDW";
  WeaponHash2[WeaponHash2["MachinePistol"] = -619010992] = "MachinePistol";
  WeaponHash2[WeaponHash2["MiniSMG"] = -1121678507] = "MiniSMG";
  WeaponHash2[WeaponHash2["RayCarbine"] = 1198256469] = "RayCarbine";
  WeaponHash2[WeaponHash2["MG"] = -1660422300] = "MG";
  WeaponHash2[WeaponHash2["CombatMG"] = 2144741730] = "CombatMG";
  WeaponHash2[WeaponHash2["CombatMGMk2"] = 3686625920] = "CombatMGMk2";
  WeaponHash2[WeaponHash2["Gusenberg"] = 1627465347] = "Gusenberg";
  WeaponHash2[WeaponHash2["PumpShotgun"] = 487013001] = "PumpShotgun";
  WeaponHash2[WeaponHash2["PumpShotgunMk2"] = 1432025498] = "PumpShotgunMk2";
  WeaponHash2[WeaponHash2["SawnOffShotgun"] = 2017895192] = "SawnOffShotgun";
  WeaponHash2[WeaponHash2["AssaultShotgun"] = -494615257] = "AssaultShotgun";
  WeaponHash2[WeaponHash2["BullpupShotgun"] = -1654528753] = "BullpupShotgun";
  WeaponHash2[WeaponHash2["Musket"] = -1466123874] = "Musket";
  WeaponHash2[WeaponHash2["HeavyShotgun"] = 984333226] = "HeavyShotgun";
  WeaponHash2[WeaponHash2["DoubleBarrelShotgun"] = -275439685] = "DoubleBarrelShotgun";
  WeaponHash2[WeaponHash2["SweeperShotgun"] = 317205821] = "SweeperShotgun";
  WeaponHash2[WeaponHash2["CombatShotgun"] = 94989220] = "CombatShotgun";
  WeaponHash2[WeaponHash2["AssaultRifle"] = -1074790547] = "AssaultRifle";
  WeaponHash2[WeaponHash2["AssaultRifleMk2"] = 961495388] = "AssaultRifleMk2";
  WeaponHash2[WeaponHash2["CarbineRifle"] = -2084633992] = "CarbineRifle";
  WeaponHash2[WeaponHash2["CarbineRifleMk2"] = 4208062921] = "CarbineRifleMk2";
  WeaponHash2[WeaponHash2["AdvancedRifle"] = -1357824103] = "AdvancedRifle";
  WeaponHash2[WeaponHash2["SpecialCarbine"] = -1063057011] = "SpecialCarbine";
  WeaponHash2[WeaponHash2["SpecialCarbineMk2"] = -1768145561] = "SpecialCarbineMk2";
  WeaponHash2[WeaponHash2["BullpupRifle"] = 2132975508] = "BullpupRifle";
  WeaponHash2[WeaponHash2["BullpupRifleMk2"] = -2066285827] = "BullpupRifleMk2";
  WeaponHash2[WeaponHash2["CompactRifle"] = 1649403952] = "CompactRifle";
  WeaponHash2[WeaponHash2["MilitaryRifle"] = -1658906650] = "MilitaryRifle";
  WeaponHash2[WeaponHash2["HeavyRifle"] = -947031628] = "HeavyRifle";
  WeaponHash2[WeaponHash2["SniperRifle"] = 100416529] = "SniperRifle";
  WeaponHash2[WeaponHash2["HeavySniper"] = 205991906] = "HeavySniper";
  WeaponHash2[WeaponHash2["HeavySniperMk2"] = 177293209] = "HeavySniperMk2";
  WeaponHash2[WeaponHash2["MarksmanRifle"] = -952879014] = "MarksmanRifle";
  WeaponHash2[WeaponHash2["MarksmanRifleMk2"] = 1785463520] = "MarksmanRifleMk2";
  WeaponHash2[WeaponHash2["RPG"] = -1312131151] = "RPG";
  WeaponHash2[WeaponHash2["GrenadeLauncher"] = -1568386805] = "GrenadeLauncher";
  WeaponHash2[WeaponHash2["GrenadeLauncherSmoke"] = 1305664598] = "GrenadeLauncherSmoke";
  WeaponHash2[WeaponHash2["Minigun"] = 1119849093] = "Minigun";
  WeaponHash2[WeaponHash2["Firework"] = 2138347493] = "Firework";
  WeaponHash2[WeaponHash2["Railgun"] = 1834241177] = "Railgun";
  WeaponHash2[WeaponHash2["HomingLauncher"] = 1672152130] = "HomingLauncher";
  WeaponHash2[WeaponHash2["CompactGrenadeLauncher"] = 125959754] = "CompactGrenadeLauncher";
  WeaponHash2[WeaponHash2["RayMinigun"] = -1238556825] = "RayMinigun";
  WeaponHash2[WeaponHash2["EmpLauncher"] = -618237638] = "EmpLauncher";
  WeaponHash2[WeaponHash2["Grenade"] = -1813897027] = "Grenade";
  WeaponHash2[WeaponHash2["BZGas"] = -1600701090] = "BZGas";
  WeaponHash2[WeaponHash2["Molotov"] = 615608432] = "Molotov";
  WeaponHash2[WeaponHash2["StickyBomb"] = 741814745] = "StickyBomb";
  WeaponHash2[WeaponHash2["ProximityMine"] = -1420407917] = "ProximityMine";
  WeaponHash2[WeaponHash2["Snowball"] = 126349499] = "Snowball";
  WeaponHash2[WeaponHash2["PipeBomb"] = -1169823560] = "PipeBomb";
  WeaponHash2[WeaponHash2["Ball"] = 600439132] = "Ball";
  WeaponHash2[WeaponHash2["SmokeGrenade"] = -37975472] = "SmokeGrenade";
  WeaponHash2[WeaponHash2["Flare"] = 1233104067] = "Flare";
  WeaponHash2[WeaponHash2["PetrolCan"] = 883325847] = "PetrolCan";
  WeaponHash2[WeaponHash2["Parachute"] = -72657034] = "Parachute";
  WeaponHash2[WeaponHash2["FireExtinguisher"] = 101631238] = "FireExtinguisher";
  WeaponHash2[WeaponHash2["HazardCan"] = -1168940174] = "HazardCan";
  WeaponHash2[WeaponHash2["FertilizerCan"] = 406929569] = "FertilizerCan";
  return WeaponHash2;
})(WeaponHash || {});
var VehicleWeaponHash = /* @__PURE__ */ ((VehicleWeaponHash2) => {
  VehicleWeaponHash2[VehicleWeaponHash2["Invalid"] = -1] = "Invalid";
  VehicleWeaponHash2[VehicleWeaponHash2["Tank"] = 1945616459] = "Tank";
  VehicleWeaponHash2[VehicleWeaponHash2["SpaceRocket"] = -123497569] = "SpaceRocket";
  VehicleWeaponHash2[VehicleWeaponHash2["PlaneRocket"] = -821520672] = "PlaneRocket";
  VehicleWeaponHash2[VehicleWeaponHash2["PlayerLaser"] = -268631733] = "PlayerLaser";
  VehicleWeaponHash2[VehicleWeaponHash2["PlayerBullet"] = 1259576109] = "PlayerBullet";
  VehicleWeaponHash2[VehicleWeaponHash2["PlayerBuzzard"] = 1186503822] = "PlayerBuzzard";
  VehicleWeaponHash2[VehicleWeaponHash2["PlayerHunter"] = -1625648674] = "PlayerHunter";
  VehicleWeaponHash2[VehicleWeaponHash2["PlayerLazer"] = -494786007] = "PlayerLazer";
  VehicleWeaponHash2[VehicleWeaponHash2["EnemyLaser"] = 1566990507] = "EnemyLaser";
  VehicleWeaponHash2[VehicleWeaponHash2["SearchLight"] = -844344963] = "SearchLight";
  VehicleWeaponHash2[VehicleWeaponHash2["Radar"] = -764006018] = "Radar";
  return VehicleWeaponHash2;
})(VehicleWeaponHash || {});
var AmmoType = /* @__PURE__ */ ((AmmoType2) => {
  AmmoType2[AmmoType2["Melee"] = 0] = "Melee";
  AmmoType2[AmmoType2["FireExtinguisher"] = 1359393852] = "FireExtinguisher";
  AmmoType2[AmmoType2["Flare"] = 1808594799] = "Flare";
  AmmoType2[AmmoType2["FlareGun"] = 1173416293] = "FlareGun";
  AmmoType2[AmmoType2["PetrolCan"] = 3395492001] = "PetrolCan";
  AmmoType2[AmmoType2["Shotgun"] = 2416459067] = "Shotgun";
  AmmoType2[AmmoType2["Pistol"] = 1950175060] = "Pistol";
  AmmoType2[AmmoType2["Ball"] = 4287981158] = "Ball";
  AmmoType2[AmmoType2["Snowball"] = 2182627693] = "Snowball";
  AmmoType2[AmmoType2["Sniper"] = 1285032059] = "Sniper";
  AmmoType2[AmmoType2["AssaultRifle"] = 218444191] = "AssaultRifle";
  AmmoType2[AmmoType2["SMG"] = 1820140472] = "SMG";
  AmmoType2[AmmoType2["Molotov"] = 1446246869] = "Molotov";
  AmmoType2[AmmoType2["StunGun"] = 2955849184] = "StunGun";
  AmmoType2[AmmoType2["MG"] = 1788949567] = "MG";
  AmmoType2[AmmoType2["GrenadeLauncher"] = 1003267566] = "GrenadeLauncher";
  AmmoType2[AmmoType2["RPG"] = 1742569970] = "RPG";
  AmmoType2[AmmoType2["Minigun"] = 2680539266] = "Minigun";
  AmmoType2[AmmoType2["Firework"] = 2938367503] = "Firework";
  AmmoType2[AmmoType2["Railgun"] = 2034517757] = "Railgun";
  AmmoType2[AmmoType2["HomingLauncher"] = 2568293933] = "HomingLauncher";
  AmmoType2[AmmoType2["Grenade"] = 1003688881] = "Grenade";
  AmmoType2[AmmoType2["StickyBomb"] = 1411692055] = "StickyBomb";
  AmmoType2[AmmoType2["ProximityMine"] = 2938243239] = "ProximityMine";
  AmmoType2[AmmoType2["PipeBomb"] = 357983224] = "PipeBomb";
  AmmoType2[AmmoType2["SmokeGrenade"] = 3859679398] = "SmokeGrenade";
  AmmoType2[AmmoType2["BZGas"] = 2608103076] = "BZGas";
  return AmmoType2;
})(AmmoType || {});
export {
  AmmoType,
  VehicleWeaponHash,
  WeaponHash
};
