// @nativewrappers/fivem/hashes/WeatherTypeHash.js
var WeatherTypeHash = /* @__PURE__ */ ((WeatherTypeHash2) => {
  WeatherTypeHash2[WeatherTypeHash2["Unknown"] = -1] = "Unknown";
  WeatherTypeHash2[WeatherTypeHash2["ExtraSunny"] = -1750463879] = "ExtraSunny";
  WeatherTypeHash2[WeatherTypeHash2["Clear"] = 916995460] = "Clear";
  WeatherTypeHash2[WeatherTypeHash2["Neutral"] = -1530260698] = "Neutral";
  WeatherTypeHash2[WeatherTypeHash2["Smog"] = 282916021] = "Smog";
  WeatherTypeHash2[WeatherTypeHash2["Foggy"] = -1368164796] = "Foggy";
  WeatherTypeHash2[WeatherTypeHash2["Clouds"] = 821931868] = "Clouds";
  WeatherTypeHash2[WeatherTypeHash2["Overcast"] = -1148613331] = "Overcast";
  WeatherTypeHash2[WeatherTypeHash2["Clearing"] = 1840358669] = "Clearing";
  WeatherTypeHash2[WeatherTypeHash2["Raining"] = 1420204096] = "Raining";
  WeatherTypeHash2[WeatherTypeHash2["ThunderStorm"] = -1233681761] = "ThunderStorm";
  WeatherTypeHash2[WeatherTypeHash2["Blizzard"] = 669657108] = "Blizzard";
  WeatherTypeHash2[WeatherTypeHash2["Snowing"] = -273223690] = "Snowing";
  WeatherTypeHash2[WeatherTypeHash2["Snowlight"] = 603685163] = "Snowlight";
  WeatherTypeHash2[WeatherTypeHash2["Christmas"] = -1429616491] = "Christmas";
  WeatherTypeHash2[WeatherTypeHash2["Halloween"] = -921030142] = "Halloween";
  return WeatherTypeHash2;
})(WeatherTypeHash || {});
export {
  WeatherTypeHash
};
